--- CMakeLists.txt.orig	2018-11-21 14:42:49.703161700 +0000
+++ CMakeLists.txt	2018-11-21 14:42:58.941231900 +0000
@@ -394,9 +394,9 @@
 run_vk_xml_generate(helper_file_generator.py vk_object_types.h)
 run_vk_xml_generate(helper_file_generator.py vk_extension_helper.h)
 run_vk_xml_generate(helper_file_generator.py vk_typemap_helper.h)
-if(BUILD_LAYERS)
-    run_external_revision_generate(${EXTERNAL_SOURCE_ROOT}/glslang/External/spirv-tools SPIRV_TOOLS_COMMIT_ID spirv_tools_commit_id.h)
-endif()
+#if(BUILD_LAYERS)
+#    run_external_revision_generate(${EXTERNAL_SOURCE_ROOT}/glslang/External/spirv-tools SPIRV_TOOLS_COMMIT_ID spirv_tools_commit_id.h)
+#endif()
 
 
 
