--- crt/_mingw.h.in.orig	2020-04-30 23:14:15.165495331 +0100
+++ crt/_mingw.h.in	2020-04-30 23:17:28.568056762 +0100
@@ -418,16 +418,14 @@
 #endif
 
 /* We are activating __USE_MINGW_ANSI_STDIO for various define indicators.
- * printf ll modifier (unsupported by msvcrt.dll) is required by C99 and C++11 standards. */
+   Note that we enable it also for _GNU_SOURCE in C++, but not for C case. */
 #if (defined (_POSIX) || defined (_POSIX_SOURCE) || defined (_POSIX_C_SOURCE) \
      || defined (_ISOC99_SOURCE) \
-     || (defined (__STDC_VERSION__) && __STDC_VERSION__ >= 199901L && __MSVCRT_VERSION__ < 0xE00) \
-     || (defined (__cplusplus) && __cplusplus >= 201103L && __MSVCRT_VERSION__ < 0xE00) \
      || defined (_XOPEN_SOURCE) || defined (_XOPEN_SOURCE_EXTENDED) \
-     || defined (_GNU_SOURCE) \
+     || (defined (_GNU_SOURCE) && defined (__cplusplus)) \
      || defined (_SVID_SOURCE)) \
     && !defined(__USE_MINGW_ANSI_STDIO)
-/* Enable __USE_MINGW_ANSI_STDIO if user did _not_ specify it explicitly... */
+/* Enable __USE_MINGW_ANSI_STDIO if _POSIX defined and if user did _not_ specify it explicitly... */
 #  define __USE_MINGW_ANSI_STDIO			1
 #endif
 
