--- src/common/string_utils.cpp.orig	2016-09-27 16:30:16.856000000 +0000
+++ src/common/string_utils.cpp	2016-09-27 16:32:56.788000000 +0000
@@ -11,6 +11,7 @@
 
 #include <algorithm>
 #include <stdlib.h>
+#include <cstdlib>
 #include <string.h>
 #include <fstream>
 #include <sstream>
@@ -141,20 +142,13 @@
 Optional<std::vector<wchar_t>> WidenString(size_t length, const char *cString)
 {
     std::vector<wchar_t> wcstring(length + 1);
-#if !defined(ANGLE_PLATFORM_WINDOWS)
+
     size_t written = mbstowcs(wcstring.data(), cString, length + 1);
     if (written == 0)
     {
         return Optional<std::vector<wchar_t>>::Invalid();
     }
-#else
-    size_t convertedChars = 0;
-    errno_t err = mbstowcs_s(&convertedChars, wcstring.data(), length + 1, cString, _TRUNCATE);
-    if (err != 0)
-    {
-        return Optional<std::vector<wchar_t>>::Invalid();
-    }
-#endif
+
     return Optional<std::vector<wchar_t>>(wcstring);
 }
 
