--- src/libANGLE/renderer/d3d/RendererD3D.cpp.orig	2017-05-22 14:25:08.100833500 +0100
+++ src/libANGLE/renderer/d3d/RendererD3D.cpp	2017-05-22 14:26:05.832148400 +0100
@@ -313,8 +313,7 @@
     if (getLUID(&adapterLuid))
     {
         char adapterLuidString[64];
-        sprintf_s(adapterLuidString, sizeof(adapterLuidString), "(adapter LUID: %08x%08x)",
-                  adapterLuid.HighPart, adapterLuid.LowPart);
+        sprintf(adapterLuidString, "(adapter LUID: %08x%08x)", adapterLuid.HighPart, adapterLuid.LowPart);
         return std::string(adapterLuidString);
     }
 
