--- build/cmake/pkg_config.cmake.orig	2018-01-21 19:02:31.544323989 +0000
+++ build/cmake/pkg_config.cmake	2018-01-21 19:02:52.640335580 +0000
@@ -48,7 +48,7 @@
 file(APPEND "${pkgconfig_file}" "Version: ${package_version}\n")
 file(APPEND "${pkgconfig_file}" "Requires:\n")
 file(APPEND "${pkgconfig_file}" "Conflicts:\n")
-file(APPEND "${pkgconfig_file}" "Libs: -L${prefix}/lib -l${pkg_name} -lm\n")
+file(APPEND "${pkgconfig_file}" "Libs: -L${prefix}/lib -l${pkg_name} -lm -lpthread\n")
 if (CONFIG_MULTITHREAD AND HAVE_PTHREAD_H)
   file(APPEND "${pkgconfig_file}" "Libs.private: -lm -lpthread\n")
 else ()
