--- av1/encoder/pass2_strategy.c.orig	2021-05-21 10:32:49.433322500 +0100
+++ av1/encoder/pass2_strategy.c	2021-05-21 10:33:21.253322500 +0100
@@ -3396,12 +3396,12 @@
   if (current_frame->frame_number == 0) {
     const GFConfig *const gf_cfg = &cpi->oxcf.gf_cfg;
     const RateControlCfg *const rc_cfg = &cpi->oxcf.rc_cfg;
-    if (use_ml_model_to_decide_flat_gop(rc_cfg) && can_disable_altref(gf_cfg) &&
-        IS_FP_STATS_TO_PREDICT_FLAT_GOP_INVALID(total_stats))
-      warn(
-          "First pass stats required in the ML model to predict a flat GOP "
-          "structure is invalid. Continuing encoding by disabling the ML "
-          "model.\n");
+//    if (use_ml_model_to_decide_flat_gop(rc_cfg) && can_disable_altref(gf_cfg) &&
+//        IS_FP_STATS_TO_PREDICT_FLAT_GOP_INVALID(total_stats))
+//      warn(
+//          "First pass stats required in the ML model to predict a flat GOP "
+//          "structure is invalid. Continuing encoding by disabling the ML "
+//          "model.\n");
   }
 
   if (cpi->oxcf.rc_cfg.mode != AOM_Q && current_frame->frame_number == 0 &&
