--- wscript.orig	2022-02-25 20:10:59.750128500 +0000
+++ wscript	2022-02-25 20:12:16.710128500 +0000
@@ -75,7 +75,7 @@
     			    int main() { boost::filesystem::copy_file ("a", "b"); }\n
 			    """,
                    msg='Checking for boost filesystem library',
-                   libpath='/usr/local/lib',
+                   libpath='lib',
                    lib=['boost_filesystem%s' % boost_lib_suffix, 'boost_system%s' % boost_lib_suffix],
                    uselib_store='BOOST_FILESYSTEM')
 
--- cscript.orig	2022-02-25 20:12:21.690128500 +0000
+++ cscript	2022-02-25 20:12:53.340128500 +0000
@@ -6,7 +6,7 @@
     elif target.platform == 'windows':
         cmd += f' --target-windows-{target.bits}'
 
-    target.append_with_space('LIBS', '-lboost_system')
-    target.append_with_space('LIBS', '-lboost_filesystem')
+    target.append_with_space('LIBS', '-lboost_system-mt-x64')
+    target.append_with_space('LIBS', '-lboost_filesystem-mt-x64')
     target.command(cmd)
     target.command('./waf build install')
--- wscript.orig	2022-02-25 21:09:10.520128500 +0000
+++ wscript	2022-02-25 21:09:32.130128500 +0000
@@ -9,15 +9,7 @@
 
 APPNAME = 'libasdcp-carl'
 
-if os.path.exists('.git'):
-    this_version = subprocess.Popen(shlex.split('git tag -l --points-at HEAD'), stdout=subprocess.PIPE).communicate()[0].decode('utf-8')
-    last_version = subprocess.Popen(shlex.split('git describe --tags --abbrev=0'), stdout=subprocess.PIPE).communicate()[0].decode('utf-8')
-    if this_version == '':
-        VERSION = '%sdevel' % last_version[1:].strip()
-    else:
-        VERSION = this_version[1:].strip()
-else:
-    VERSION = open('VERSION').read().strip()
+VERSION = '2.0.14'
 
 def options(opt):
     opt.load('compiler_cxx')
