--- wscript.orig	2017-09-06 19:52:46.970416700 +0100
+++ wscript	2017-09-06 19:54:20.586301300 +0100
@@ -34,7 +34,7 @@
     conf.check_cfg(package='openssl', args='--cflags --libs', uselib_store='OPENSSL', mandatory=True)
 
     if conf.options.target_windows:
-        boost_lib_suffix = ''
+        boost_lib_suffix = '-mt'
     else:
         boost_lib_suffix = ''
 
@@ -60,7 +60,7 @@
     			    int main() { boost::filesystem::copy_file ("a", "b"); }\n
 			    """,
                    msg='Checking for boost filesystem library',
-                   libpath='/usr/local/lib',
+                   libpath='lib',
                    lib=['boost_filesystem%s' % boost_lib_suffix, 'boost_system%s' % boost_lib_suffix],
                    uselib_store='BOOST_FILESYSTEM')
 
