--- src/Makefile.am.orig	2015-12-28 10:28:28.532332384 +0000
+++ src/Makefile.am	2015-12-28 10:30:43.648332384 +0000
@@ -84,7 +84,8 @@
 		KM_util.h KM_xml.cpp KM_xml.h KM_tai.h KM_tai.cpp
 
 # linker flags (*not* including libraries to link against) for a library
-libkumu_la_LDFLAGS = -release @VERSION@
+libkumu_la_LDFLAGS = -no-undefined -release @VERSION@
+libkumu_la_LIBADD = $(ASDCPLIB_LIBS)
 
 
 # sources for asdcp library that don't get added to a distribution
@@ -113,7 +114,7 @@
 nodist_libasdcp_la_SOURCES += TimedText_Transform.h TimedText_Transform.cpp
 endif
 
-libasdcp_la_LDFLAGS = -release @VERSION@
+libasdcp_la_LDFLAGS = -no-undefined -release @VERSION@
 # additional libraries to link against for a library
 libasdcp_la_LIBADD = libkumu.la
 libasdcp_la_CPPFLAGS = -DASDCP_PLATFORM=\"@host@\"
--- src/Makefile.am.orig	2020-04-12 16:33:46.769367990 +0100
+++ src/Makefile.am	2020-04-12 16:34:21.865021734 +0100
@@ -156,7 +156,7 @@
 	ACES_Sequence_Parser.cpp \
 	AS_02_ACES.cpp
 
-libas02_la_LDFLAGS = -release @VERSION@
+libas02_la_LDFLAGS = -no-undefined -release @VERSION@
 libas02_la_LIBADD = libasdcp.la libkumu.la
 libas02_la_CPPFLAGS = -DASDCP_PLATFORM=\"@host@\"
 endif
@@ -169,7 +169,7 @@
 	PHDR_Sequence_Parser.cpp \
 	AS_02_PHDR.cpp
 
-libphdr_la_LDFLAGS = -release @VERSION@
+libphdr_la_LDFLAGS = -no-undefined -release @VERSION@
 libphdr_la_LIBADD = libasdcp.la libkumu.la
 libphdr_la_CPPFLAGS = -DASDCP_PLATFORM=\"@host@\"
 endif
--- src/Makefile.am.orig	2020-04-12 16:42:56.427937366 +0100
+++ src/Makefile.am	2020-04-12 16:44:40.582907070 +0100
@@ -170,7 +170,7 @@
 	AS_02_PHDR.cpp
 
 libphdr_la_LDFLAGS = -no-undefined -release @VERSION@
-libphdr_la_LIBADD = libasdcp.la libkumu.la
+libphdr_la_LIBADD = libas02.la libasdcp.la libkumu.la
 libphdr_la_CPPFLAGS = -DASDCP_PLATFORM=\"@host@\"
 endif
 
