--- CMakeLists.txt.orig	2025-04-05 08:57:09.835438474 +0100
+++ CMakeLists.txt	2025-04-05 08:57:27.955168916 +0100
@@ -70,12 +70,12 @@
   )
 endif()
 
-if(WIN32)
-  list(APPEND sources
-    src/extras/getopt.c
-    src/extras/getopt1.c
-  )
-endif()
+#if(WIN32)
+#  list(APPEND sources
+#    src/extras/getopt.c
+#    src/extras/getopt1.c
+#  )
+#endif()
 
 add_executable(
   AtomicParsley
--- src/AtomicParsley.h.orig	2025-04-05 09:01:34.131507475 +0100
+++ src/AtomicParsley.h	2025-04-05 09:02:00.021122320 +0100
@@ -84,11 +84,11 @@
 #endif
 
 #include <signal.h>
-#ifndef _WIN32
+/* #ifndef _WIN32 */
 #include <getopt.h>
-#else
+/* #else
 #include "extras/getopt.h"
-#endif
+#endif */
 
 #ifndef PRIu64
 #ifdef _WIN32
