--- avs_core/filters/AviSource/avi_source.cpp.orig	2021-03-06 17:46:15.340000000 +0000
+++ avs_core/filters/AviSource/avi_source.cpp	2021-03-06 17:48:47.580000000 +0000
@@ -294,7 +294,7 @@
     return result;
 }
 
-#ifndef MSVC
+#if !defined MSVC && !defined _WIN32
 static __inline LRESULT
 ICDecompressEx(HIC hic,DWORD dwFlags,LPBITMAPINFOHEADER lpbiSrc,LPVOID lpSrc,INT xSrc,INT ySrc,INT dxSrc,INT dySrc,LPBITMAPINFOHEADER lpbiDst,LPVOID lpDst,INT xDst,INT yDst,INT dxDst,INT dyDst)
 {
@@ -334,7 +334,7 @@
 	ic.dyDst = dyDst;
 	return ICSendMessage(hic,ICM_DECOMPRESSEX_BEGIN,(DWORD_PTR)&ic,sizeof(ic));
 }
-#endif // MSVC
+#endif // MSVC or _WIN32
 
 LRESULT AVISource::DecompressBegin(LPBITMAPINFOHEADER lpbiSrc, LPBITMAPINFOHEADER lpbiDst) {
   if (!ex) {
