--- avs_core/core/BufferPool.h.orig	2021-03-18 17:15:21.680000000 +0000
+++ avs_core/core/BufferPool.h	2021-03-18 17:15:36.150000000 +0000
@@ -10,7 +10,7 @@
 private:
 
   struct BufferDesc;
-  typedef std::multimap<size_t, BufferDesc*> MapType;
+  typedef std::multimap<std::size_t, BufferDesc*> MapType;
 
   InternalEnvironment* Env;
   MapType Map;
--- avs_core/core/BufferPool.h.orig	2021-03-18 17:20:55.680000000 +0000
+++ avs_core/core/BufferPool.h	2021-03-18 17:21:17.120000000 +0000
@@ -15,7 +15,7 @@
   InternalEnvironment* Env;
   MapType Map;
 
-  void* PrivateAlloc(size_t nBytes, size_t alignment, void* user);
+  void* PrivateAlloc(std::size_t nBytes, std::size_t alignment, void* user);
   void PrivateFree(void* buffer);
 
 public:
@@ -23,7 +23,7 @@
   BufferPool(InternalEnvironment* env);
   ~BufferPool();
 
-  void* Allocate(size_t nBytes, size_t alignment, bool pool);
+  void* Allocate(std::size_t nBytes, std::size_t alignment, bool pool);
   void Free(void* ptr);
 
 };
