--- boost/system/detail/system_category_win32.hpp.orig	2019-02-10 18:37:32.634894190 +0000
+++ boost/system/detail/system_category_win32.hpp	2019-02-10 18:46:37.485414846 +0000
@@ -26,7 +26,7 @@
 namespace detail
 {
 
-#if ( defined(_MSC_VER) && _MSC_VER < 1900 ) || ( defined(__MINGW32__) && !defined(__MINGW64_VERSION_MAJOR) )
+#if ( defined(_MSC_VER) && _MSC_VER < 1900 )
 
 inline char const * unknown_message_win32( int ev, char * buffer, std::size_t len )
 {
