--- mingw-w64-build.orig	2018-09-11 21:56:02.489912900 +0100
+++ mingw-w64-build	2018-09-11 21:57:13.362719700 +0100
@@ -159,7 +159,7 @@
   echo "configuring gcc" >&3
   "../../src/gcc-$v_gcc/configure" --target="$host" --disable-shared \
     --enable-static --disable-multilib --prefix="$prefix" \
-    --enable-languages=c,c++ --disable-nls || error_exit
+    --enable-shared --enable-languages=c,c++,fortran --disable-nls || error_exit
   echo "running 'make-gcc' for gcc" >&3
   make -j $cpus all-gcc || error_exit
   echo "running 'install-gcc' for gcc" >&3
--- mingw-w64-build.orig	2018-09-18 15:09:28.486392400 +0100
+++ mingw-w64-build	2018-09-18 15:12:04.986413200 +0100
@@ -17,7 +17,7 @@
 #
 
 v_script="4.0.1"
-v_mingww64="6git"
+v_mingww64="git"
 v_binutils="2.32git"
 v_gcc="8svn"
 v_gmp="6.1.2"
@@ -83,7 +83,7 @@
   cd "$src" || error_exit
 
   echo "downloading mingw-w64" >&3
-  git clone --depth 1 -b v6.x git://git.code.sf.net/p/mingw-w64/mingw-w64 mingw-w64-$v_mingww64 || error_exit
+  git clone --depth 1 git://git.code.sf.net/p/mingw-w64/mingw-w64 mingw-w64-$v_mingww64 || error_exit
 
   echo "downloading binutils" >&3
   git clone --depth 1 -b binutils-2_31-branch --single-branch git://sourceware.org/git/binutils-gdb.git binutils-$v_binutils || error_exit
--- mingw-w64-build.orig	2018-09-28 14:53:27.452643200 +0100
+++ mingw-w64-build	2018-09-28 14:55:32.780794000 +0100
@@ -149,7 +149,7 @@
   clean_build "$bld/mingw-w64"
   echo "configuring mingw-w64-headers" >&3
   "../../src/mingw-w64-$v_mingww64/mingw-w64-headers/configure" --build="$build" \
-    --host="$host" --prefix="$prefix/$host" --enable-secure-api || error_exit
+    --host="$host" --prefix="$prefix/$host" --enable-sdk=all --enable-secure-api || error_exit
   echo "installing mingw-w64-headers" >&3
   make install || error_exit
   cd "$prefix" || error_exit
@@ -168,7 +168,7 @@
   clean_build "$bld/mingw-w64"
   echo "configuring mingw-w64-crt" >&3
   "../../src/mingw-w64-$v_mingww64/mingw-w64-crt/configure" --build="$build" --host="$host" \
-    --prefix="$prefix/$host" --with-sysroot="$prefix/$host"
+    --prefix="$prefix/$host" --with-sysroot="$prefix/$host" --enable-lib64 --disable-lib32 --disable-libarm32 --disable-libarm64
   echo "building mingw-w64-crt" >&3
   make -j $cpus || error_exit
   echo "installing mingw-w64-crt" >&3
@@--- mingw-w64-build.orig	2018-11-23 10:47:34.778519900 +0000
@@+++ mingw-w64-build	2018-11-23 10:50:02.977004000 +0000
@@ -89,7 +89,7 @@
@@   git clone --depth 1 -b binutils-2_31-branch --single-branch git://sourceware.org/git/binutils-gdb.git binutils-$v_binutils || error_exit
@@ 
@@   echo "downloading gcc" >&3
@@-  svn co svn://gcc.gnu.org/svn/gcc/branches/gcc-8-branch gcc-$v_gcc || error_exit
@@+  svn co svn://gcc.gnu.org/svn/gcc/trunk gcc-$v_gcc || error_exit
@@ 
@@   local urls=(
@@     "https://ftp.gnu.org/gnu/gmp/gmp-$v_gmp.tar.xz"
@@--- mingw-w64-build.orig	2018-11-23 13:10:34.495530200 +0000
@@+++ mingw-w64-build	2018-11-23 13:11:07.913024600 +0000
@@ -19,7 +19,7 @@
@@ v_script="4.0.1"
@@ v_mingww64="git"
@@ v_binutils="2.31git"
@@-v_gcc="8svn"
@@+v_gcc="trunk"
@@ v_gmp="6.1.2"
@@ v_mpfr="4.0.1"
@@ v_mpc="1.1.0"
@@--- mingw-w64-build.orig	2018-11-23 14:10:13.808222600 +0000
@@+++ mingw-w64-build	2018-11-23 14:10:27.317552300 +0000
@@ -161,7 +161,7 @@
@@     --enable-static --disable-multilib --prefix="$prefix" \
@@     --enable-shared --enable-languages=c,c++,fortran --disable-nls || error_exit
@@   echo "running 'make-gcc' for gcc" >&3
@@-  make -j $cpus all-gcc || error_exit
@@+  make -j 1 all-gcc || error_exit
@@   echo "running 'install-gcc' for gcc" >&3
@@   make install-gcc || error_exit
@@ 
@@ -176,7 +176,7 @@
@@ 
@@   cd "$bld/gcc" || error_exit
@@   echo "building gcc" >&3
@@-  make -j $cpus || error_exit
@@+  make -j 1 || error_exit
@@   echo "installing gcc" >&3
@@   make install || error_exit
@@ 
@@--- mingw-w64-build.orig	2018-11-29 22:22:23.658557100 +0000
@@+++ mingw-w64-build	2018-11-29 22:24:25.257201600 +0000
@@ -84,6 +84,9 @@
@@ 
@@   echo "downloading mingw-w64" >&3
@@   git clone --depth 1 git://git.code.sf.net/p/mingw-w64/mingw-w64 mingw-w64-$v_mingww64 || error_exit
@@+  cd mingw-w64-git
@@+  git checkout 172cf5 || error_exit
@@+  cd ..
@@ 
@@   echo "downloading binutils" >&3
@@   git clone --depth 1 -b binutils-2_31-branch --single-branch git://sourceware.org/git/binutils-gdb.git binutils-$v_binutils || error_exit
@@--- mingw-w64-build.orig	2018-11-29 22:40:47.092815700 +0000
@@+++ mingw-w64-build	2018-11-29 22:40:57.511903500 +0000
@@ -83,7 +83,7 @@
@@   cd "$src" || error_exit
@@ 
@@   echo "downloading mingw-w64" >&3
@@-  git clone --depth 1 git://git.code.sf.net/p/mingw-w64/mingw-w64 mingw-w64-$v_mingww64 || error_exit
@@+  git clone git://git.code.sf.net/p/mingw-w64/mingw-w64 mingw-w64-$v_mingww64 || error_exit
@@   cd mingw-w64-git
@@   git checkout 172cf5 || error_exit
@@   cd ..
@@--- mingw-w64-build.orig	2018-11-30 21:13:08.497817100 +0000
@@+++ mingw-w64-build	2018-11-30 21:15:35.418608100 +0000
@@ -92,7 +92,10 @@
@@   git clone --depth 1 -b binutils-2_31-branch --single-branch git://sourceware.org/git/binutils-gdb.git binutils-$v_binutils || error_exit
@@ 
@@   echo "downloading gcc" >&3
@@-  svn co svn://gcc.gnu.org/svn/gcc/branches/gcc-8-branch gcc-$v_gcc || error_exit
@@+  git clone https://github.com/gcc-mirror/gcc.git gcc-$v_gcc || error_exit
@@+  cd gcc-8svn
@@+    git checkout b96a695f2bffb7a25831c931f6228078e978cea9 || error_exit
@@+  cd ..
@@ 
@@   local urls=(
@@     "https://ftp.gnu.org/gnu/gmp/gmp-$v_gmp.tar.xz"
@@--- mingw-w64-build.orig	2018-11-30 22:23:38.582110300 +0000
@@+++ mingw-w64-build	2018-11-30 22:24:52.537316500 +0000
@@ -92,10 +92,7 @@
@@   git clone --depth 1 -b binutils-2_31-branch --single-branch git://sourceware.org/git/binutils-gdb.git binutils-$v_binutils || error_exit
@@ 
@@   echo "downloading gcc" >&3
@@-  git clone https://github.com/gcc-mirror/gcc.git gcc-$v_gcc || error_exit
@@-  cd gcc-8svn
@@-    git checkout b96a695f2bffb7a25831c931f6228078e978cea9 || error_exit
@@-  cd ..
@@+  git clone --depth 1 -b gcc-8_2_0-release --single-branch https://github.com/gcc-mirror/gcc.git gcc-$v_gcc || error_exit
@@ 
@@   local urls=(
@@     "https://ftp.gnu.org/gnu/gmp/gmp-$v_gmp.tar.xz"
@@--- mingw-w64-build.orig	2018-12-01 05:59:42.638981700 +0000
@@+++ mingw-w64-build	2018-12-01 06:01:25.880160300 +0000
@@ -23,7 +23,7 @@
@@ v_gmp="6.1.2"
@@ v_mpfr="4.0.1"
@@ v_mpc="1.1.0"
@@-v_isl="0.20"
@@+v_isl="0.19"
@@ 
@@ show_help()
@@ {
--- mingw-w64-build.orig	2018-12-22 22:00:57.746578632 +0000
+++ mingw-w64-build	2018-12-22 22:02:17.426116911 +0000
@@ -89,7 +89,8 @@
   git clone --depth 1 -b binutils-2_31-branch --single-branch git://sourceware.org/git/binutils-gdb.git binutils-$v_binutils || error_exit
 
   echo "downloading gcc" >&3
-  svn co svn://gcc.gnu.org/svn/gcc/branches/gcc-8-branch gcc-$v_gcc || error_exit
+  git clone --depth 1 -b master --single-branch git@github.com:Warblefly/gcc-working-temp.git gcc-$v_gcc || error_exit
+#  svn co svn://gcc.gnu.org/svn/gcc/branches/gcc-8-branch gcc-$v_gcc || error_exit
 
   local urls=(
     "https://ftp.gnu.org/gnu/gmp/gmp-$v_gmp.tar.xz"
