--- mingw-w64-build.orig	2018-09-11 21:56:02.489912900 +0100
+++ mingw-w64-build	2018-09-11 21:57:13.362719700 +0100
@@ -159,7 +159,7 @@
   echo "configuring gcc" >&3
   "../../src/gcc-$v_gcc/configure" --target="$host" --disable-shared \
     --enable-static --disable-multilib --prefix="$prefix" \
-    --enable-languages=c,c++ --disable-nls || error_exit
+    --enable-languages=c,c++,fortran --disable-nls || error_exit
   echo "running 'make-gcc' for gcc" >&3
   make -j $cpus all-gcc || error_exit
   echo "running 'install-gcc' for gcc" >&3
--- mingw-w64-build.orig	2018-09-18 15:09:28.486392400 +0100
+++ mingw-w64-build	2018-09-18 15:12:04.986413200 +0100
@@ -17,7 +17,7 @@
 #
 
 v_script="4.0.1"
-v_mingww64="5git"
+v_mingww64="git"
 v_binutils="2.31git"
 v_gcc="8svn"
 v_gmp="6.1.2"
@@ -83,7 +83,7 @@
   cd "$src" || error_exit
 
   echo "downloading mingw-w64" >&3
-  git clone --depth 1 -b v5.x git://git.code.sf.net/p/mingw-w64/mingw-w64 mingw-w64-$v_mingww64 || error_exit
+  git clone --depth 1 git://git.code.sf.net/p/mingw-w64/mingw-w64 mingw-w64-$v_mingww64 || error_exit
 
   echo "downloading binutils" >&3
   git clone --depth 1 -b binutils-2_31-branch --single-branch git://sourceware.org/git/binutils-gdb.git binutils-$v_binutils || error_exit
