--- mingw-w64-build.orig	2017-12-21 22:17:10.067247266 +0000
+++ mingw-w64-build	2017-12-21 22:28:21.512080095 +0000
@@ -17,8 +17,8 @@
 #
 
 v_script="4.0.1"
-v_binutils="2.29.1"
-v_gcc="7.2.0"
+v_binutils="2.29.1"
+v_gcc="7.3.0"
 v_gmp="6.1.2"
 v_mpfr="3.1.666666"
 v_mpc="1.0.3"
@@ -86,7 +86,7 @@
 
   local urls=(
     "http://ftp.gnu.org/gnu/binutils/binutils-$v_binutils.tar.bz2"
-    "http://ftp.gnu.org/gnu/gcc/gcc-$v_gcc/gcc-$v_gcc.tar.xz"
+    "http://ftp.gnu.org/gnu/gcc/gcc-$v_gcc/gcc-$v_gcc.tar.xz"
     "http://ftp.gnu.org/gnu/gmp/gmp-$v_gmp.tar.xz"
     "http://ftp.gnu.org/gnu/mpfr/mpfr-$v_mpfr.tar.xz"
     "http://ftp.gnu.org/gnu/mpc/mpc-$v_mpc.tar.gz"
@@ -97,7 +97,7 @@
     local archive="${url##*/}"
     local c
     echo "downloading $archive" >&3
-    wget -nv "$url" || error_exit
+    wget "$url" || error_exit
     echo "extracting $archive" >&3
     case "${archive##*.}" in
       gz) c=z ;;
@@ -139,18 +139,18 @@
 
   clean_build "$bld/binutils"
   echo "configuring binutils" >&3
-  "../../src/binutils-$v_binutils/configure" --prefix="$prefix" --disable-shared \
-    --enable-static --with-sysroot="$prefix" --target="$host" \
+  "../../src/binutils-$v_binutils/configure" --prefix="$prefix" --enable-shared \
+    --disable-static --with-sysroot="$prefix" --target="$host" \
     --disable-multilib --disable-nls || error_exit
   echo "building binutils" >&3
   make -j $cpus || error_exit
   echo "installing binutils" >&3
-  make install || error_exit
+  make install-strip || error_exit
 
   clean_build "$bld/mingw-w64"
   echo "configuring mingw-w64-headers" >&3
   "../../src/mingw-w64-git/mingw-w64-headers/configure" --build="$build" \
-    --host="$host" --prefix="$prefix/$host" --enable-secure-api || error_exit
+    --host="$host" --prefix="$prefix/$host" --enable-secure-api --enable-idl || error_exit
   echo "installing mingw-w64-headers" >&3
   make install || error_exit
   cd "$prefix" || error_exit
@@ -158,28 +158,78 @@
 
   clean_build "$bld/gcc"
   echo "configuring gcc" >&3
-  "../../src/gcc-$v_gcc/configure" --target="$host" --disable-shared \
-    --enable-static --disable-multilib --prefix="$prefix" \
-    --enable-languages=c,c++ --disable-nls || error_exit
+  "../../src/gcc-$v_gcc/configure" --target="$host" --enable-shared \
+    --disable-static --disable-multilib --prefix="$prefix" \
+    --enable-languages=c,c++,fortran --disable-nls || error_exit
   echo "running 'make-gcc' for gcc" >&3
   make -j $cpus all-gcc || error_exit
   echo "running 'install-gcc' for gcc" >&3
-  make install-gcc || error_exit
+  make install-strip-gcc || error_exit
 
   clean_build "$bld/mingw-w64"
   echo "configuring mingw-w64-crt" >&3
   "../../src/mingw-w64-git/mingw-w64-crt/configure" --build="$build" --host="$host" \
-    --prefix="$prefix/$host" --with-sysroot="$prefix/$host"
+    --prefix="$prefix/$host" --disable-lib32 --enable-lib64 --disable-libarm32 \
+    --disable-libarm64 --enable-experimental --with-sysroot="$prefix/$host"
   echo "building mingw-w64-crt" >&3
   make -j $cpus || error_exit
   echo "installing mingw-w64-crt" >&3
-  make install || error_exit
+  make install-strip || error_exit
 
   cd "$bld/gcc" || error_exit
   echo "building gcc" >&3
   make -j $cpus || error_exit
   echo "installing gcc" >&3
-  make install || error_exit
+  make install-strip || error_exit
+
+  clean_build "$bld/mingw-w64"
+  echo "configuring mingw-w64-tools/gendef"
+  "../../src/mingw-w64-git/mingw-w64-tools/gendef/configure" --build="$build" --host="$host" \
+  --prefix="$prefix/$host" --with-sysroot="$prefix/$host" || error_exit
+  echo "building mingw-w64-tools/gendef"
+  make -j $cpus || error_exit
+  echo "installing mingw-w64-tools/gendef"
+  make install-strip || error_exit
+
+  clean_build "$bld/mingw-w64"
+  echo "configuring mingw-w64-tools/genidl"
+  "../../src/mingw-w64-git/mingw-w64-tools/genidl/configure" --build="$build" --host="$host" \
+  --prefix="$prefix/$host" --with-sysroot="$prefix/$host" || error_exit
+  echo "building mingw-w64-tools/genidl"
+  make -j $cpus || error_exit
+  echo "installing mingw-w64-tools/genidl"
+  make install-strip || error_exit
+
+  clean_build "$bld/mingw-w64"
+  echo "configuring mingw-w64-tools/genlib"
+  "../../src/mingw-w64-git/mingw-w64-tools/genlib/configure" --build="$build" --host="$host" \
+  --prefix="$prefix/$host" --with-sysroot="$prefix/$host" || error_exit
+  echo "building mingw-w64-tools/genlib"
+  make -j $cpus || error_exit
+  echo "installing mingw-w64-tools/genlib"
+  make install-strip || error_exit
+
+  clean_build "$bld/mingw-w64"
+  echo "configuring mingw-w64-tools/genpeimg"
+  "../../src/mingw-w64-git/mingw-w64-tools/genpeimg/configure" --build="$build" --host="$host" \
+  --prefix="$prefix/$host" --with-sysroot="$prefix/$host" || error_exit
+  echo "building mingw-w64-tools/genpeimg"
+  make -j $cpus || error_exit
+  echo "installing mingw-w64-tools/genpeimg"
+  make install-strip || error_exit
+
+  clean_build "$bld/mingw-w64"
+  echo "configuring mingw-w64-tools/widl"
+  export ac_cv_func_malloc_0_nonnull=yes
+  export ac_cv_func_realloc_0_nonnull=yes
+  "../../src/mingw-w64-git/mingw-w64-tools/widl/configure" --build="$build" --host="$host" \
+   --prefix="$prefix/$host" --with-sysroot="$prefix/$host" || error_exit
+  echo "building mingw-w64-tools/widl"
+  make -j $cpus || error_exit
+  echo "installing mingw-w64-tools/widl"
+  make install-strip || error_exit
+  unset ac_cv_func_malloc_0_nonnull
+  unset ac_cv_func_realloc_0_nonnull
 
   return 0
 }
