--- src/cairo-pdf-interchange.c.orig	2016-10-02 12:23:27.419000000 +0000
+++ src/cairo-pdf-interchange.c	2016-10-02 12:24:38.428000000 +0000
@@ -52,6 +52,8 @@
 #include "cairo-error-private.h"
 #include "cairo-output-stream-private.h"
 
+#include <private-strndup.h>
+
 static void
 write_rect_to_pdf_quad_points (cairo_output_stream_t   *stream,
 			       const cairo_rectangle_t *rect,
--- test/pdf-tagged-text.c.orig	2016-10-02 12:51:35.623000000 +0000
+++ test/pdf-tagged-text.c	2016-10-02 12:52:11.092000000 +0000
@@ -33,6 +33,8 @@
 #include <cairo.h>
 #include <cairo-pdf.h>
 
+#include <private-strndup.h>
+
 /* This test checks PDF with
  * - tagged text
  * - hyperlinks
