--- configure.ac.orig	2020-05-13 09:03:45.651925359 +0100
+++ configure.ac	2020-05-13 09:04:09.939672497 +0100
@@ -4727,14 +4727,6 @@
 ENABLE_STATIC="$enable_static"
 AC_SUBST(ENABLE_STATIC)
 
-dnl merge the pkg-config Libs.private field into Libs when static-only
-if test "x$enable_shared" = "xno"; then
-  LIBCURL_NO_SHARED=$LIBCURL_LIBS
-else
-  LIBCURL_NO_SHARED=
-fi
-AC_SUBST(LIBCURL_NO_SHARED)
-
 dnl
 dnl For keeping supported features and protocols also in pkg-config file
 dnl since it is more cross-compile friendly than curl-config
--- libcurl.pc.in.orig	2020-05-16 11:48:09.022998421 +0100
+++ libcurl.pc.in	2020-05-16 11:48:26.118825445 +0100
@@ -34,6 +34,6 @@
 URL: https://curl.haxx.se/
 Description: Library to transfer files with ftp, http, etc.
 Version: @CURLVERSION@
-Libs: -L${libdir} -lcurl @LIBCURL_NO_SHARED@
+Libs: -L${libdir} -lcurl
 Libs.private: @LIBCURL_LIBS@
 Cflags: -I${includedir} @CPPFLAG_CURL_STATICLIB@
--- curl-config.in.orig	2020-08-24 20:13:54.500000000 +0100
+++ curl-config.in	2020-08-24 20:14:16.380000000 +0100
@@ -160,12 +160,8 @@
         else
            CURLLIBDIR=""
         fi
-        if test "X@ENABLE_SHARED@" = "Xno"; then
-          echo ${CURLLIBDIR}-lcurl @LIBCURL_LIBS@
-        else
-          echo ${CURLLIBDIR}-lcurl
-        fi
         ;;
+
     --ssl-backends)
         echo "@SSL_BACKENDS@"
         ;;
