--- src/lib/magick_image_proxy.cc.orig	2017-07-05 17:12:25.886506300 +0100
+++ src/lib/magick_image_proxy.cc	2017-07-05 17:13:07.435953600 +0100
@@ -102,10 +102,6 @@
 		throw DecodeError (String::compose (_("Could not decode image file (%1)"), error));
 	}
 
-	if (magick_image->colorSpace() == Magick::LogColorspace) {
-		magick_image->colorSpace(Magick::RGBColorspace);
-	}
-
 	dcp::Size size (magick_image->columns(), magick_image->rows());
 
 	_image.reset (new Image (AV_PIX_FMT_RGB24, size, true));
