--- wscript.orig	2021-03-29 15:00:28.750000000 +0100
+++ wscript	2021-03-29 15:01:28.370000000 +0100
@@ -231,7 +231,7 @@
                        mandatory=True,
                        msg='Checking for libicu',
                        okmsg='yes',
-                       libpath=['/usr/lib', '/usr/lib/x86_64-linux-gnu'],
+                       #libpath=['/usr/local/lib', '/usr/lib', '/usr/lib/x86_64-linux-gnu'],
                        lib=['icuio', 'icui18n', 'icudata', 'icuuc'],
                        uselib_store='ICU')
 
