--- wscript.orig	2021-09-26 21:16:21.154601700 +0100
+++ wscript	2021-09-26 21:16:46.424601700 +0100
@@ -151,6 +151,8 @@
     if conf.env.TARGET_WINDOWS:
         conf.env.append_value('CXXFLAGS', '-DDCPOMATIC_WINDOWS')
         conf.env.append_value('CXXFLAGS', '-DWIN32_LEAN_AND_MEAN')
+        conf.env.append_value('CXXFLAGS', '-DUNICODE')
+        conf.env.append_value('CXXFLAGS', '-D_UNICODE')
         conf.env.append_value('CXXFLAGS', '-DBOOST_THREAD_PROVIDES_GENERIC_SHARED_MUTEX_ON_WIN')
         conf.env.append_value('CXXFLAGS', '-mfpmath=sse')
         conf.env.append_value('CXXFLAGS', '-Wcast-align')
