--- src/lib/text_subtitle.h.orig	2017-05-17 19:18:33.724952300 +0100
+++ src/lib/text_subtitle.h	2017-05-17 19:19:31.372570200 +0100
@@ -22,7 +22,7 @@
 #define DCPOMATIC_TEXT_SUBTITLE_H
 
 #include "dcpomatic_time.h"
-#include <sub/subtitle.h>
+#include <libsub/subtitle.h>
 #include <boost/shared_ptr.hpp>
 #include <vector>
 
--- src/lib/job.cc.orig	2017-05-17 19:42:47.749531700 +0100
+++ src/lib/job.cc	2017-05-17 19:43:06.791159400 +0100
@@ -30,7 +30,7 @@
 #include "log.h"
 #include "compose.hpp"
 #include <dcp/exceptions.h>
-#include <sub/exceptions.h>
+#include <libsub/exceptions.h>
 #include <boost/thread.hpp>
 #include <boost/filesystem.hpp>
 #include <boost/foreach.hpp>
--- src/lib/text_subtitle.cc.orig	2017-05-17 19:43:45.885394300 +0100
+++ src/lib/text_subtitle.cc	2017-05-17 19:44:01.079348300 +0100
@@ -22,9 +22,9 @@
 #include "cross.h"
 #include "exceptions.h"
 #include "text_subtitle_content.h"
-#include <sub/subrip_reader.h>
-#include <sub/ssa_reader.h>
-#include <sub/collect.h>
+#include <libsub/subrip_reader.h>
+#include <libsub/ssa_reader.h>
+#include <libsub/collect.h>
 #include <unicode/ucsdet.h>
 #include <unicode/ucnv.h>
 #include <iostream>
--- src/lib/ffmpeg_decoder.cc.orig	2017-05-17 19:41:25.788726300 +0100
+++ src/lib/ffmpeg_decoder.cc	2017-05-17 20:01:22.065209500 +0100
@@ -42,9 +42,9 @@
 #include "subtitle_content.h"
 #include "audio_content.h"
 #include <dcp/subtitle_string.h>
-#include <sub/ssa_reader.h>
-#include <sub/subtitle.h>
-#include <sub/collect.h>
+/* #include <libsub/ssa_reader.h> */
+#include <libsub/subtitle.h>
+#include <libsub/collect.h>
 extern "C" {
 #include <libavcodec/avcodec.h>
 #include <libavformat/avformat.h>
