--- wscript.orig	2016-04-28 20:15:06.832000000 +0000
+++ wscript	2016-04-28 20:15:27.945000000 +0000
@@ -75,6 +75,7 @@
                                        '-D__STDC_FORMAT_MACROS',
                                        '-msse',
                                        '-fno-strict-aliasing',
+                                       '-fpermissive', '-DBOOST_ASIO_DISABLE_STD_FUTURE=1',
                                        '-Wall',
                                        '-Wno-attributes',
                                        '-Wextra',
