--- src/wx/wscript.orig	2022-03-07 11:31:51.879802800 +0000
+++ src/wx/wscript	2022-03-07 11:36:14.599802800 +0000
@@ -277,14 +277,17 @@
         conf.env.LIB_RTAUDIO.append('mfuuid')
         conf.env.LIB_RTAUDIO.append('wmcodecdspuuid')
 
-    if conf.env.TARGET_WINDOWS_64 or conf.env.TARGET_WINDOWS_32 or conf.env.TARGET_LINUX:
-        conf.check_cfg(package='gl', args='--cflags --libs', uselib_store='GL', mandatory=True)
-        conf.check_cfg(package='glu', args='--cflags --libs', uselib_store='GLU', mandatory=True)
-        if conf.env.TARGET_WINDOWS_64 or conf.env.TARGET_WINDOWS_32:
-            conf.check_cfg(package='glew', args='--cflags --libs', uselib_store='GLEW', mandatory=True)
-    else:
-        conf.env.STLIB_GL = 'gl'
-        conf.env.STLIB_GLU = 'glu'
+#    if conf.env.TARGET_WINDOWS_64 or conf.env.TARGET_WINDOWS_32 or conf.env.TARGET_LINUX:
+#        conf.check_cfg(package='gl', args='--cflags --libs', uselib_store='GL', mandatory=True)
+#        conf.check_cfg(package='glu', args='--cflags --libs', uselib_store='GLU', mandatory=True)
+#        if conf.env.TARGET_WINDOWS_64 or conf.env.TARGET_WINDOWS_32:
+#            conf.check_cfg(package='glew', args='--cflags --libs', uselib_store='GLEW', mandatory=True)
+#    else:
+#        conf.env.STLIB_GL = 'gl'
+#        conf.env.STLIB_GLU = 'glu'
+    conf.env.LIB_GL = ['opengl32']
+    conf.env.LIB_GLU = ['glu32']
+    conf.env.LIB_GLEW = ['glew32']
 
     if conf.env.TARGET_LINUX:
         conf.check_cxx(fragment="""
