--- wscript.orig	2023-04-11 14:11:07.917207514 +0100
+++ wscript	2023-04-11 14:11:36.367207347 +0100
@@ -153,8 +153,8 @@
         conf.check(lib='dsound', uselib_store='DSOUND', msg="Checking for library dsound")
         conf.check(lib='winmm', uselib_store='WINMM', msg="Checking for library winmm")
         conf.check(lib='ksuser', uselib_store='KSUSER', msg="Checking for library ksuser")
-        boost_lib_suffix = '-mt'
-        boost_thread = 'boost_thread_win32-mt'
+        boost_lib_suffix = '-mt-x64'
+        boost_thread = 'boost_thread-mt-x64'
         conf.check_cxx(fragment="""
                                #include <boost/locale.hpp>\n
                                int main() { std::locale::global (boost::locale::generator().generate ("")); }\n
