--- plugins/djvFFmpegPlugin/djvFFmpegSave.cpp.orig	2017-11-06 22:03:33.820494000 +0000
+++ plugins/djvFFmpegPlugin/djvFFmpegSave.cpp	2017-11-06 22:04:47.568695800 +0000
@@ -232,7 +232,10 @@
     avCodecContext->height        = info.size.y;
     avCodecContext->time_base.den = info.sequence.speed.scale();
     avCodecContext->time_base.num = info.sequence.speed.duration();
-    
+
+#define CODEC_FLAG_GLOBAL_HEADER 0x00400000
+#define CODEC_FLAG_QSCALE        0x0002
+ 
     if (avFormat->flags & AVFMT_GLOBALHEADER)
         avCodecContext->flags |= CODEC_FLAG_GLOBAL_HEADER;
 
