--- cmake/Modules/FindOCIO.cmake.orig	2020-01-24 15:12:03.165692835 +0000
+++ cmake/Modules/FindOCIO.cmake	2020-01-24 15:12:36.445381375 +0000
@@ -43,7 +43,7 @@
     set_target_properties(OCIO::OCIO PROPERTIES
         IMPORTED_LOCATION "${OCIO_LIBRARY}"
         INTERFACE_INCLUDE_DIRECTORIES "${OCIO_INCLUDE_DIRS}"
-		INTERFACE_COMPILE_DEFINITIONS ${OCIO_DEFINES})
+		INTERFACE_COMPILE_DEFINITIONS "${OCIO_DEFINES}")
 endif()
 if(OCIO_FOUND AND NOT TARGET OCIO)
     add_library(OCIO INTERFACE)
--- lib/djvCore/OSWin32.cpp.orig	2020-01-27 12:49:11.266413690 +0000
+++ lib/djvCore/OSWin32.cpp	2020-01-27 12:51:55.864802866 +0000
@@ -31,7 +31,8 @@
 
 #include <djvCore/FileInfo.h>
 #include <djvCore/Memory.h>
-
+#define NTDDI_VERSION 0x06000000
+#define _WIN32_WINNT 0x0600
 #ifndef WIN32_LEAN_AND_MEAN
 #define WIN32_LEAN_AND_MEAN
 #endif // WIN32_LEAN_AND_MEAN
--- lib/djvCore/CMakeLists.txt.orig	2020-01-27 12:58:37.528880701 +0000
+++ lib/djvCore/CMakeLists.txt	2020-01-27 12:58:58.916672088 +0000
@@ -139,7 +139,7 @@
     Threads::Threads
     ${CMAKE_DL_LIBS})
 if (WIN32)
-    set(LIBRARIES ${LIBRARIES} Netapi32.lib mpr.lib)
+    set(LIBRARIES ${LIBRARIES} netapi32.lib mpr.lib)
 elseif (APPLE)
     find_library(CORE_SERVICES CoreServices)
     set(LIBRARIES ${LIBRARIES} ${CORE_SERVICES})
--- lib/djvCore/DirectoryWatcherWin32.cpp.orig	2020-02-04 22:29:09.429981659 +0000
+++ lib/djvCore/DirectoryWatcherWin32.cpp	2020-02-04 22:29:34.849713850 +0000
@@ -62,7 +62,7 @@
                 std::condition_variable changedCV;
                 std::mutex mutex;
                 std::thread thread;
-                std::atomic<bool> running = true;
+		std::atomic<bool> running = { true };
                 std::function<void(void)> callback;
                 std::shared_ptr<Time::Timer> timer;
             };
--- lib/djvCore/OSWin32.cpp.orig	2020-02-04 22:28:40.446287015 +0000
+++ lib/djvCore/OSWin32.cpp	2020-02-04 22:30:31.009122182 +0000
@@ -40,10 +40,10 @@
 #define NOMINMAX
 #endif // NOMINMAX
 #include <windows.h>
-#include <Shlobj.h>
+#include <shlobj.h>
 #include <shellapi.h>
 #include <stdlib.h>
-#include <VersionHelpers.h>
+#include <versionhelpers.h>
 
 #include <codecvt>
 #include <locale>
--- lib/djvCore/CMakeLists.txt.orig	2020-02-04 22:38:04.132348218 +0000
+++ lib/djvCore/CMakeLists.txt	2020-02-04 22:38:44.903918657 +0000
@@ -139,7 +139,7 @@
     Threads::Threads
     ${CMAKE_DL_LIBS})
 if (WIN32)
-    set(LIBRARIES ${LIBRARIES} netapi32.lib mpr.lib)
+    set(LIBRARIES ${LIBRARIES} msvcr120.lib netapi32.lib mpr.lib)
 elseif (APPLE)
     find_library(CORE_SERVICES CoreServices)
     set(LIBRARIES ${LIBRARIES} ${CORE_SERVICES})
--- lib/djvAV/CMakeLists.txt.orig	2020-02-04 22:55:02.643479825 +0000
+++ lib/djvAV/CMakeLists.txt	2020-02-04 22:55:21.079341269 +0000
@@ -151,7 +151,7 @@
 	OCIO
     IlmBase
     #OpenAL
-    RtAudio
+    rtaudio
     OpenGL::GL
     djvCore)
 if(FFmpeg_FOUND)
--- bin/djv/win.rc.orig	2020-02-06 18:36:03.506931711 +0000
+++ bin/djv/win.rc	2020-02-06 18:36:17.118797808 +0000
@@ -1,3 +1,3 @@
 #include "winrc.h"
 
-DJV_ICON ICON "..\\..\\etc\\Windows\\djv.ico"
+DJV_ICON ICON "../../etc/Windows/djv.ico"
--- lib/djvCore/Core.h.orig	2020-02-08 11:30:55.806774567 +0000
+++ lib/djvCore/Core.h	2020-02-08 11:31:10.978629111 +0000
@@ -77,7 +77,7 @@
 
 #if defined(DJV_PLATFORM_WINDOWS)
 //! \bug https://social.msdn.microsoft.com/Forums/vstudio/en-US/8f40dcd8-c67f-4eba-9134-a19b9178e481/vs-2015-rc-linker-stdcodecvt-error?forum=vcgeneral
-typedef unsigned int djv_char_t;
+typedef char16_t djv_char_t;
 #define DJV_SNPRINTF sprintf_s
 #define DJV_STRNCPY strncpy_s
 #else // DJV_PLATFORM_WINDOWS
--- lib/djvAV/Audio.h.orig	2020-04-04 21:54:48.715943723 +0100
+++ lib/djvAV/Audio.h	2020-04-04 21:55:06.291765310 +0100
@@ -37,7 +37,7 @@
 //#include <AL/al.h>
 //#include <AL/alc.h>
 //#include <AL/alext.h>
-#include <RtAudio.h>
+#include <rtaudio/RtAudio.h>
 
 #include <limits>
 
--- lib/djvAV/AudioSystem.cpp.orig	2020-04-04 23:00:06.524020720 +0100
+++ lib/djvAV/AudioSystem.cpp	2020-04-04 23:00:21.919864050 +0100
@@ -34,7 +34,7 @@
 #include <djvCore/Error.h>
 #include <djvCore/OS.h>
 
-#include <RtAudio.h>
+#include <rtaudio/RtAudio.h>
 
 using namespace djv::Core;
 
--- lib/djvViewApp/Media.h.orig	2020-04-04 23:06:40.396012651 +0100
+++ lib/djvViewApp/Media.h	2020-04-04 23:06:55.683857081 +0100
@@ -36,7 +36,7 @@
 #include <djvCore/ListObserver.h>
 #include <djvCore/ValueObserver.h>
 
-#include <RtAudio.h>
+#include <rtaudio/RtAudio.h>
 
 namespace djv
 {
--- lib/djvAV/PNGRead.cpp.orig	2020-04-13 21:28:28.509935857 +0100
+++ lib/djvAV/PNGRead.cpp	2020-04-13 21:29:29.404968651 +0100
@@ -135,7 +135,7 @@
                         png_infop *  pngInfo,
                         png_infop *  pngInfoEnd)
                     {
-                        if (setjmp(png_jmpbuf(png)))
+                        if (__builtin_setjmp(png_jmpbuf(png)))
                         {
                             return false;
                         }
@@ -184,7 +184,7 @@
                 {
                     bool pngScanline(png_structp png, uint8_t * out)
                     {
-                        if (setjmp(png_jmpbuf(png)))
+                        if (__builtin_setjmp(png_jmpbuf(png)))
                         {
                             return false;
                         }
@@ -194,7 +194,7 @@
 
                     bool pngEnd(png_structp png, png_infop pngInfo)
                     {
-                        if (setjmp(png_jmpbuf(png)))
+                        if (__builtin_setjmp(png_jmpbuf(png)))
                         {
                             return false;
                         }
--- lib/djvAV/JPEGWrite.cpp.orig	2020-04-13 21:31:49.810822144 +0100
+++ lib/djvAV/JPEGWrite.cpp	2020-04-13 21:32:23.258325555 +0100
@@ -102,7 +102,7 @@
 
                     bool jpegInit(jpeg_compress_struct* jpeg, JPEGErrorStruct* error)
                     {
-                        if (setjmp(error->jump))
+                        if (__builtin_setjmp(error->jump))
                         {
                             return false;
                         }
@@ -118,7 +118,7 @@
                         const Options& options,
                         JPEGErrorStruct* error)
                     {
-                        if (setjmp(error->jump))
+                        if (__builtin_setjmp(error->jump))
                         {
                             return false;
                         }
@@ -155,7 +155,7 @@
                         const uint8_t* in,
                         JPEGErrorStruct* error)
                     {
-                        if (::setjmp(error->jump))
+                        if (::__builtin_setjmp(error->jump))
                         {
                             return false;
                         }
@@ -169,7 +169,7 @@
 
                     bool jpeg_end(jpeg_compress_struct* jpeg, JPEGErrorStruct* error)
                     {
-                        if (::setjmp(error->jump))
+                        if (::__builtin_setjmp(error->jump))
                         {
                             return false;
                         }
--- lib/djvAV/PNGWrite.cpp.orig	2020-04-13 21:34:46.648248971 +0100
+++ lib/djvAV/PNGWrite.cpp	2020-04-13 21:35:24.335715421 +0100
@@ -136,7 +136,7 @@
                         png_infop *         pngInfo,
                         const Image::Info & info)
                     {
-                        if (setjmp(png_jmpbuf(png)))
+                        if (__builtin_setjmp(png_jmpbuf(png)))
                         {
                             return false;
                         }
@@ -178,7 +178,7 @@
 
                     bool pngScanline(png_structp png, const uint8_t * in)
                     {
-                        if (setjmp(png_jmpbuf(png)))
+                        if (__builtin_setjmp(png_jmpbuf(png)))
                             return false;
                         png_write_row(png, reinterpret_cast<const png_byte*>(in));
                         return true;
@@ -186,7 +186,7 @@
 
                     bool pngEnd(png_structp png, png_infop pngInfo)
                     {
-                        if (setjmp(png_jmpbuf(png)))
+                        if (__builtin_setjmp(png_jmpbuf(png)))
                             return false;
                         png_write_end(png, pngInfo);
                         return true;
--- CMakeLists.txt.orig	2020-08-09 13:48:59.510000000 +0100
+++ CMakeLists.txt	2020-08-09 13:49:11.050000000 +0100
@@ -303,7 +303,7 @@
 include(djvPackage)
 
 add_subdirectory(lib)
-add_subdirectory(tests)
+#add_subdirectory(tests)
 if(DJV_BUILD_TINY)
 elseif(DJV_BUILD_MINIMAL)
     add_subdirectory(examples)
--- CMakeLists.txt.orig	2020-08-09 13:59:35.350000000 +0100
+++ CMakeLists.txt	2020-08-09 14:00:01.820000000 +0100
@@ -68,8 +68,8 @@
 elseif(WIN32)
     set(DJV_PLATFORM_WINDOWS TRUE)
     add_definitions(-DDJV_PLATFORM_WINDOWS)
-    set(CMAKE_C_FLAGS "/MP /EHsc")
-    set(CMAKE_CXX_FLAGS "/MP /EHsc")
+    #set(CMAKE_C_FLAGS "/MP /EHsc")
+    #set(CMAKE_CXX_FLAGS "/MP /EHsc")
 endif()
 
 # Set the output directories for the build.
--- lib/djvAV/PNG.cpp.orig	2020-08-09 14:16:05.570000000 +0100
+++ lib/djvAV/PNG.cpp	2020-08-09 14:16:24.640000000 +0100
@@ -51,7 +51,7 @@
     {
         auto error = reinterpret_cast<djv::AV::IO::PNG::ErrorStruct *>(png_get_error_ptr(in));
         error->messages.push_back(msg);
-        longjmp(png_jmpbuf(in), 1);
+        __builtin_longjmp(png_jmpbuf(in), 1);
     }
 
     void djvPngWarning(png_structp in, png_const_charp msg)
--- lib/djvAV/JPEG.cpp.orig	2020-08-09 14:25:43.160000000 +0100
+++ lib/djvAV/JPEG.cpp	2020-08-09 14:26:15.600000000 +0100
@@ -68,7 +68,7 @@
                         char message[JMSG_LENGTH_MAX] = "";
                         in->err->format_message(in, message);
                         error->messages.push_back(message);
-                        ::longjmp(error->jump, 1);
+                        ::__builtin_longjmp(error->jump, 1);
                     }
 
                     void djvJPEGWarning(j_common_ptr in, int level)
--- lib/djvCore/RapidJSONTemplatesInline.h.orig	2020-08-09 17:44:47.620000000 +0100
+++ lib/djvCore/RapidJSONTemplatesInline.h	2020-08-09 17:45:20.150000000 +0100
@@ -1,7 +1,7 @@
 // SPDX-License-Identifier: BSD-3-Clause
 // Copyright (c) 2004-2020 Darby Johnston
 // All rights reserved.
-
+#undef GetObject
 namespace djv
 {
     template<typename T>
