--- CMakeLists.txt.orig	2017-09-04 16:11:59.629455900 +0100
+++ CMakeLists.txt	2017-09-04 16:12:55.181493200 +0100
@@ -46,7 +46,7 @@
     set(EXTRA_LIBS ${CORESERVICES_LIB})
 endif()
 
-if (WIN32)
+if (WIN32 AND MSVC)
     if(NOT ICONV_DIR)
       set(ICONV_DIR "${CMAKE_SOURCE_DIR}/winbuild")
     endif()
@@ -114,7 +114,7 @@
    link_directories(${CUSTOM_LINK_DIR})
 endif()
 
-if (win_static)
+if (win_static AND MSVC)
     set(CompilerFlags
         CMAKE_CXX_FLAGS
         CMAKE_CXX_FLAGS_DEBUG
--- addon/doxysearch/CMakeLists.txt.orig	2017-09-04 16:13:41.332477000 +0100
+++ addon/doxysearch/CMakeLists.txt	2017-09-04 16:14:02.149359500 +0100
@@ -3,7 +3,7 @@
 find_package(Xapian REQUIRED)
 find_package(ZLIB REQUIRED)
 
-if (WIN32)
+if (WIN32 AND NOT(MINGW))
   set(WIN_EXTRA_LIBS "uuid.lib rpcrt4.lib ws2_32.lib")
 endif()
 
