--- dxgi1_3.orig	2017-09-09 18:49:42.090710700 +0100
+++ dxgi1_3.h	2017-09-09 19:01:47.860110600 +0100
@@ -132,10 +132,19 @@
         virtual void STDMETHODCALLTYPE Trim( void) = 0;
         
     };
-    
-    
+
 #else 	/* C style interface */
 
+    typedef struct _DXGI_MODE_DESC1 {
+      UINT                     Width;
+      UINT                     Height;
+      DXGI_RATIONAL            RefreshRate;
+      DXGI_FORMAT              Format;
+      DXGI_MODE_SCANLINE_ORDER ScanlineOrdering;
+      DXGI_MODE_SCALING        Scaling;
+      BOOL                     Stereo;
+    } DXGI_MODE_DESC1;
+
     typedef struct IDXGIDevice3Vtbl
     {
         BEGIN_INTERFACE
