--- CMakeLists.txt.orig	2021-05-23 19:14:52.933322500 +0100
+++ CMakeLists.txt	2021-05-23 19:15:25.593322500 +0100
@@ -314,11 +314,11 @@
     # Suppress compliler warnings - see ECC-850
     # Suppress warnings about using 'insecure' functions. Fixing this would require changes all over
     # the codebase which would damage portability.
-    ecbuild_add_c_flags("/D_CRT_SECURE_NO_WARNINGS")
+    # ecbuild_add_c_flags("/D_CRT_SECURE_NO_WARNINGS")
     # Suppress warnings about using well-known C functions.
-    ecbuild_add_c_flags("/D_CRT_NONSTDC_NO_DEPRECATE")
+    # ecbuild_add_c_flags("/D_CRT_NONSTDC_NO_DEPRECATE")
     # Suppress C4267: warns about possible loss of data when converting 'size_t' to 'int'.
-    ecbuild_add_c_flags("/wd4267")
+    # ecbuild_add_c_flags("/wd4267")
 endif()
 
 ###############################################################################
--- CMakeLists.txt.orig	2021-05-23 19:22:59.093322500 +0100
+++ CMakeLists.txt	2021-05-23 19:24:09.453322500 +0100
@@ -297,13 +297,11 @@
 
 # fix for #if IEEE_LE or IEE_BE instead of #ifdef
 
-if( IEEE_BE )
-    set( IEEE_LE 0 )
-endif()
+# Absolute JW hack. Cross-compiling. Can't run tests
+set( IEEE_LE 1 )
+
+set( IEEE_BE 0 )
 
-if( IEEE_LE )
-    set( IEEE_BE 0 )
-endif()
 
 set( ECCODES_ON_WINDOWS 0 )
 if( EC_OS_NAME MATCHES "windows" )
--- CMakeLists.txt.orig	2021-05-23 19:38:34.483322500 +0100
+++ CMakeLists.txt	2021-05-23 19:39:36.723322500 +0100
@@ -51,6 +51,9 @@
       " ECCODES_HAVE_C_INLINE
 )
 
+set( IEEE_LE 1 )
+set( IEEE_BE 0 )
+
 include(eccodes_test_endiness)
 if( EC_OS_NAME MATCHES "windows" )
     include(eccodes_find_linux_utils)
--- CMakeLists.txt.orig	2021-05-23 19:49:13.883322500 +0100
+++ CMakeLists.txt	2021-05-23 19:49:30.493322500 +0100
@@ -353,9 +353,9 @@
 
 add_definitions( -DHAVE_ECCODES_CONFIG_H )
 
-if( CMAKE_COMPILER_IS_GNUCC )
-    ecbuild_add_c_flags("-pedantic")
-endif()
+if( CMAKE_COMPILER_IS_GNUCC )
+    ecbuild_add_c_flags("-Wno-div-by-zero -DMS_WIN64")
+endif()
 
 # gfortran 10 has become stricter with argument matching
 if( HAVE_FORTRAN AND CMAKE_Fortran_COMPILER_ID MATCHES "GNU" AND NOT CMAKE_Fortran_COMPILER_VERSION VERSION_LESS 10 )
--- tools/grib_tools.c.orig	2021-05-25 19:51:43.093322500 +0100
+++ tools/grib_tools.c	2021-05-25 19:52:11.163322500 +0100
@@ -625,11 +625,7 @@
     struct stat s;
     int stat_val = 0;
 
-#ifndef ECCODES_ON_WINDOWS
-    stat_val = lstat(path, &s);
-#else
     stat_val = stat(path, &s);
-#endif
 
     if (stat_val != 0) {
         grib_context_log(c, (GRIB_LOG_ERROR) | (GRIB_LOG_PERROR), "Cannot stat %s", path);
