--- src/makernote.cpp.orig	2016-04-06 22:53:08.686000000 +0000
+++ src/makernote.cpp	2016-04-06 22:54:11.917000000 +0000
@@ -42,7 +42,7 @@
 #include <string>
 #include <cstring>
 
-#ifdef _MSC_VER
+#if defined(_MSC_VER) || defined(_WIN32)
 # include <windows.h>
 # include <shlobj.h>
   /* older SDKs not have these */
@@ -92,7 +92,7 @@
 		{
     		std::string homedir;
     		std::string inifile;
-#ifdef _MSC_VER
+#if defined(_MSC_VER) || defined(_WIN32)
 			char path[MAX_PATH];
 			if (SUCCEEDED(SHGetFolderPathA(NULL, CSIDL_PROFILE, NULL, 0, path))) {
   				homedir = std::string(path);
