--- configure.orig	2021-04-14 19:30:29.650000000 +0100
+++ configure	2021-04-14 19:30:53.920000000 +0100
@@ -4351,10 +4351,10 @@
 
 if enabled cuda_nvcc; then
     nvcc_default="nvcc"
-    nvccflags_default="-gencode arch=compute_30,code=sm_30 -O2"
+    nvccflags_default="-gencode arch=compute_35,code=sm_35 -O2 -allow-unsupported-compiler"
 else
     nvcc_default="clang"
-    nvccflags_default="--cuda-gpu-arch=sm_30 -O2"
+    nvccflags_default="--cuda-gpu-arch=sm_35 -O2 -allow-unsupported-compiler"
     NVCC_C=""
 fi
 
--- configure.orig	2021-12-03 09:42:34.094574900 +0000
+++ configure	2021-12-03 09:43:24.654574900 +0000
@@ -6370,7 +6370,7 @@
 if enabled cuda_nvcc; then
     nvccflags="$nvccflags -ptx"
 else
-    nvccflags="$nvccflags -S -nocudalib -nocudainc --cuda-device-only -Wno-c++11-narrowing -include ${source_link}/compat/cuda/cuda_runtime.h"
+    nvccflags="$nvccflags -S -nocudalib -nocudainc --cuda-device-only -Wno-c++11-narrowing -include ${source_link}/compat/cuda/cuda_runtime.h -allow-unsupported-compiler"
     check_nvcc cuda_llvm
 fi
 
