--- data/Makefile.am.orig	2018-06-23 13:09:05.978149000 +0100
+++ data/Makefile.am	2018-06-23 13:09:45.484393100 +0100
@@ -22,11 +22,11 @@
 	@rm -f dll_uninstall.nsh
 	@rm -rf dlls
 	@mkdir -p dlls
-	"$(srcdir)/dllcopy.sh" "$(top_builddir)/src/interface" filezilla.exe $(OBJDUMP) $(CXX) $$PATH
-	"$(srcdir)/dllcopy.sh" "$(top_builddir)/src/putty" fzsftp.exe $(OBJDUMP) $(CXX) $$PATH
-	"$(srcdir)/dllcopy.sh" "$(top_builddir)/src/putty" fzputtygen.exe $(OBJDUMP) $(CXX) $$PATH
+	"$(srcdir)/dllcopy.sh" "$(top_builddir)/src/interface" filezilla.exe $(OBJDUMP) $(CXX) "$$PATH"
+	"$(srcdir)/dllcopy.sh" "$(top_builddir)/src/putty" fzsftp.exe $(OBJDUMP) $(CXX) "$$PATH"
+	"$(srcdir)/dllcopy.sh" "$(top_builddir)/src/putty" fzputtygen.exe $(OBJDUMP) $(CXX) "$$PATH"
 	if test -f "${top_builddir}/src/storj/fzstorj.exe"; then \
-		"$(srcdir)/dllcopy.sh" "$(top_builddir)/src/storj" fzstorj.exe $(OBJDUMP) $(CXX) $$PATH; \
+		"$(srcdir)/dllcopy.sh" "$(top_builddir)/src/storj" fzstorj.exe $(OBJDUMP) $(CXX) "$$PATH"; \
 	fi
 
 installlangfiles.nsh: Makefile
