--- src/file.c.orig	2022-06-01 22:01:36.443295200 +0100
+++ src/file.c	2022-06-01 22:02:02.613295200 +0100
@@ -110,6 +110,7 @@
 #define OPT_LONGONLY(longname, opt, def, doc, id)	\
     {longname, opt, NULL, id},
 #include "file_opts.h"
+#include <ctype.h>
 #undef OPT
 #undef OPT_LONGONLY
     {0, 0, NULL, 0}
--- src/file.c.orig	2022-06-01 22:12:57.843295200 +0100
+++ src/file.c	2022-06-01 22:13:31.273295200 +0100
@@ -40,6 +40,7 @@
 #include <stdlib.h>
 #include <unistd.h>
 #include <string.h>
+#include <ctype.h>
 #ifdef RESTORE_TIME
 # if (__COHERENT__ >= 0x420)
 #  include <sys/utime.h>
