--- src/interface/volume_enumerator.cpp.orig	2020-01-10 23:21:50.813042437 +0000
+++ src/interface/volume_enumerator.cpp	2020-01-10 23:22:54.688558159 +0000
@@ -5,8 +5,8 @@
 
 #include <wx/msw/registry.h>
 
-DEFINE_EVENT_TYPE(fzEVT_VOLUMEENUMERATED)
-DEFINE_EVENT_TYPE(fzEVT_VOLUMESENUMERATED)
+wxDEFINE_EVENT(fzEVT_VOLUMEENUMERATED, wxCommandEvent);
+wxDEFINE_EVENT(fzEVT_VOLUMESENUMERATED, wxCommandEvent);
 
 CVolumeDescriptionEnumeratorThread::CVolumeDescriptionEnumeratorThread(wxEvtHandler* pEvtHandler, fz::thread_pool & pool)
 	: m_pEvtHandler(pEvtHandler)
--- src/interface/volume_enumerator.h.orig	2020-01-10 23:30:28.711770314 +0000
+++ src/interface/volume_enumerator.h	2020-01-10 23:32:52.524633315 +0000
@@ -20,8 +20,8 @@
 // startup, use a background thread to obtain the labels.
 #ifdef FZ_WINDOWS
 
-DECLARE_EVENT_TYPE(fzEVT_VOLUMEENUMERATED, -1)
-DECLARE_EVENT_TYPE(fzEVT_VOLUMESENUMERATED, -1)
+wxDECLARE_EVENT(fzEVT_VOLUMEENUMERATED, wxCommandEvent);
+wxDECLARE_EVENT(fzEVT_VOLUMESENUMERATED, wxCommandEvent);
 
 class CVolumeDescriptionEnumeratorThread final
 {
--- src/interface/led.h.orig	2020-01-10 23:44:27.077807678 +0000
+++ src/interface/led.h	2020-01-10 23:45:14.133219489 +0000
@@ -4,7 +4,7 @@
 #include <wx/event.h>
 #include <wx/timer.h>
 
-DECLARE_EVENT_TYPE(fzEVT_UPDATE_LED_TOOLTIP, -1)
+wxDECLARE_EVENT(fzEVT_UPDATE_LED_TOOLTIP, wxCommandEvent);
 
 class CLed final : public wxWindow
 {
--- src/interface/led.cpp.orig	2020-01-10 23:44:41.141630733 +0000
+++ src/interface/led.cpp	2020-01-10 23:45:39.340908738 +0000
@@ -5,7 +5,7 @@
 
 #include <wx/dcclient.h>
 
-DEFINE_EVENT_TYPE(fzEVT_UPDATE_LED_TOOLTIP)
+wxDEFINE_EVENT(fzEVT_UPDATE_LED_TOOLTIP, wxCommandEvent);
 
 BEGIN_EVENT_TABLE(CLed, wxWindow)
 	EVT_PAINT(CLed::OnPaint)
--- src/interface/commandqueue.h.orig	2020-01-10 23:47:59.963223684 +0000
+++ src/interface/commandqueue.h	2020-01-10 23:48:23.726946109 +0000
@@ -6,7 +6,7 @@
 class CState;
 class CMainFrame;
 
-DECLARE_EVENT_TYPE(fzEVT_GRANTEXCLUSIVEENGINEACCESS, -1)
+wxDECLARE_EVENT(fzEVT_GRANTEXCLUSIVEENGINEACCESS, wxCommandEvent);
 
 class CCommandQueue final
 {
--- src/interface/commandqueue.cpp.orig	2020-01-10 23:47:48.883353755 +0000
+++ src/interface/commandqueue.cpp	2020-01-10 23:48:41.386741022 +0000
@@ -9,7 +9,7 @@
 
 #include <algorithm>
 
-DEFINE_EVENT_TYPE(fzEVT_GRANTEXCLUSIVEENGINEACCESS)
+wxDEFINE_EVENT(fzEVT_GRANTEXCLUSIVEENGINEACCESS, wxCommandEvent);
 
 int CCommandQueue::m_requestIdCounter = 0;
 
