--- src/utils/cst_tokenstream.c.orig	2017-10-21 22:35:14.470043300 +0100
+++ src/utils/cst_tokenstream.c	2017-10-21 22:35:35.051100800 +0100
@@ -266,7 +266,7 @@
     (*buffer)[p] = '\0';
 }
 
-inline int ts_utf8_sequence_length(char c0)
+int ts_utf8_sequence_length(char c0)
 {
     // Get the expected length of UTF8 sequence given its most
     // significant byte
--- src/utils/cst_val.c.orig	2017-10-21 22:36:19.741888700 +0100
+++ src/utils/cst_val.c	2017-10-21 22:36:38.211224500 +0100
@@ -475,7 +475,7 @@
     }
 }
 
-inline int utf8_sequence_length(char c0)
+int utf8_sequence_length(char c0)
 {
     // Get the expected length of UTF8 sequence given its most
     // significant byte
