--- CMakeLists.txt.orig	2020-05-10 21:31:55.998300839 +0100
+++ CMakeLists.txt	2020-05-10 21:32:15.007144014 +0100
@@ -69,17 +69,13 @@
       string (REGEX MATCH "[3-9]\\.[0-9]\\.[0-9]" _gcc_version "${_gcc_version_info}")
 
       # gcc <4.1 had poor support for symbol visibility
-      if ((${_gcc_version} VERSION_GREATER "4.1") OR (${_gcc_version} VERSION_EQUAL "4.1"))
          set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fvisibility=hidden")
          set (ENABLE_VISIBILITY ON)
          add_definitions (-DLIBGME_VISIBILITY)
 
          # GCC >= 4.2 also correctly supports making inline members have hidden
          # visibility by default.
-         if ((${_gcc_version} VERSION_GREATER "4.2") OR (${_gcc_version} VERSION_EQUAL "4.2"))
             set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fvisibility-inlines-hidden")
-         endif()
-      endif()
    endif() # test visibility
 
     # Cache this result
