--- config/override.m4.orig	2021-09-17 23:23:17.145044600 +0100
+++ config/override.m4	2021-09-17 23:23:29.925044600 +0100
@@ -29,7 +29,7 @@
 
 dnl Ensure exactly this Autoconf version is used
 m4_ifndef([_GCC_AUTOCONF_VERSION],
-  [m4_define([_GCC_AUTOCONF_VERSION], [2.69])])
+  [m4_define([_GCC_AUTOCONF_VERSION], [2.71])])
 
 dnl Test for the exact version when AC_INIT is expanded.
 dnl This allows to update the tree in steps (for testing)
