--- gettext-runtime/man/Makefile.am.orig	2017-05-05 15:34:26.470738200 +0100
+++ gettext-runtime/man/Makefile.am	2017-05-05 15:35:25.953103700 +0100
@@ -108,7 +108,7 @@
 $(man_MAN1IN) $(man_MAN1OTHER): help2man $(top_srcdir)/../.version
 	progname=`echo $@ | sed -e 's/\.in$$//' -e 's/\.1$$//'`; \
 	IN_HELP2MAN=1 $(SHELL) x-to-1 $(UPDATEMODE) "$(PERL)" "$(HELP2MAN)" \
-	  ../src/$${progname}$(EXEEXT) $(srcdir)/$${progname}.x $@
+	  ${progname} $(srcdir)/$${progname}.x $@
 
 gettext.1.in: gettext.x ../src/gettext.c
 ngettext.1.in: ngettext.x ../src/ngettext.c
--- gettext-tools/man/Makefile.am.orig	2017-05-05 16:08:58.852815200 +0100
+++ gettext-tools/man/Makefile.am	2017-05-05 16:10:00.230247800 +0100
@@ -95,7 +95,7 @@
 $(man_MAN1SRC): help2man $(top_srcdir)/../.version
 	progname=`echo $@ | sed -e 's/\.in$$//' -e 's/\.1$$//'`; \
 	IN_HELP2MAN=1 $(SHELL) x-to-1 $(UPDATEMODE) "$(PERL)" "$(HELP2MAN)" \
-	  ../src/$${progname}$(EXEEXT) $(srcdir)/$${progname}.x $@
+	  ${progname} $(srcdir)/$${progname}.x $@
 
 msgcmp.1: msgcmp.x ../src/msgcmp.c
 msgfmt.1: msgfmt.x ../src/msgfmt.c
@@ -117,7 +117,7 @@
 $(man_MAN1MISC): help2man $(top_srcdir)/../.version
 	progname=`echo $@ | sed -e 's/\.in$$//' -e 's/\.1$$//'`; \
 	IN_HELP2MAN=1 $(SHELL) x-to-1 $(UPDATEMODE) "$(PERL)" "$(HELP2MAN)" \
-	  ../misc/$${progname}$(EXEEXT) $(srcdir)/$${progname}.x $@
+	  ${progname} $(srcdir)/$${progname}.x $@
 
 gettextize.1: gettextize.x ../misc/gettextize.in
 autopoint.1: autopoint.x ../misc/autopoint.in
