--- glib/glibmm/objectbase.h.orig	2016-05-09 22:00:14.635000000 +0000
+++ glib/glibmm/objectbase.h	2016-05-09 22:01:55.286000000 +0000
@@ -30,7 +30,16 @@
 #include <typeinfo>
 #include <map> // Needed until the next ABI break.
 #include <memory> // Not used by ObjectBase any more, but user code may rely on it being here.
-#include <mutex>
+#include <boost/thread.hpp>
+
+namespace std {
+  using boost::mutex;
+  using boost::recursive_mutex;
+  using boost::lock_guard;
+  using boost::condition_variable;
+  using boost::unique_lock;
+  using boost::thread;
+}
 
 #ifndef DOXYGEN_SHOULD_SKIP_THIS
 extern "C" {
--- glib/glibmm/dispatcher.cc.orig	2016-05-09 22:21:36.059000000 +0000
+++ glib/glibmm/dispatcher.cc	2016-05-09 22:22:34.010000000 +0000
@@ -35,7 +35,16 @@
 #include <io.h>
 #include <direct.h>
 #include <list>
-#include <mutex>
+#include <boost/thread.hpp>
+
+namespace std {
+  using boost::mutex;
+  using boost::recursive_mutex;
+  using boost::lock_guard;
+  using boost::condition_variable;
+  using boost::unique_lock;
+  using boost::thread;
+}
 #else
 #include <unistd.h>
 #endif
