--- gio/giomm/application.cc.orig	2020-01-05 10:35:42.098141541 +0000
+++ gio/giomm/application.cc	2020-01-05 10:45:24.971744987 +0000
@@ -29,7 +29,7 @@
 #include <giomm/init.h>
 #include <cstring> // std::memset()
 #include <map>
-#include <mutex>
+#include <boost/thread/mutex.hpp>
 #include <vector>
 
 using Flags = Gio::Application::Flags;
@@ -197,7 +197,7 @@
 
 // Gio::Application instances may be used in different threads.
 // Accesses to option_arg_callback_data must be thread-safe.
-std::mutex option_arg_callback_data_mutex;
+boost::mutex option_arg_callback_data_mutex;
 
 gboolean
 Application_option_arg_callback(
@@ -207,7 +207,7 @@
 
   // option_name is either a single dash followed by a single letter (for a
   // short name) or two dashes followed by a long option name.
-  std::unique_lock<std::mutex> lock(option_arg_callback_data_mutex);
+  boost::unique_lock<boost::mutex> lock(option_arg_callback_data_mutex);
   OptionArgCallbackDataMap::const_iterator iterFind = option_arg_callback_data.end();
   if (option_name[1] == '-')
   {
@@ -292,7 +292,7 @@
 Application::~Application() noexcept
 {
   // Delete all OptionArgCallbackData instances that belong to this application.
-  std::lock_guard<std::mutex> lock(option_arg_callback_data_mutex);
+  boost::lock_guard<boost::mutex> lock(option_arg_callback_data_mutex);
   OptionArgCallbackDataMap::iterator iter = option_arg_callback_data.begin();
   while (iter != option_arg_callback_data.end())
   {
@@ -439,7 +439,7 @@
 {
   OptionArgCallbackData* callback_data = nullptr;
   {
-    std::lock_guard<std::mutex> lock(option_arg_callback_data_mutex);
+    boost::lock_guard<boost::mutex> lock(option_arg_callback_data_mutex);
     OptionArgCallbackDataMap::iterator iterFind = option_arg_callback_data.find(long_name);
     if (iterFind != option_arg_callback_data.end())
       return; // Ignore duplicates
--- gio/giomm/application.cc.orig	2020-01-05 10:49:13.337321165 +0000
+++ gio/giomm/application.cc	2020-01-05 10:49:40.673032738 +0000
@@ -30,6 +30,7 @@
 #include <cstring> // std::memset()
 #include <map>
 #include <boost/thread/mutex.hpp>
+#include <boost/thread/lock_guard.hpp>
 #include <vector>
 
 using Flags = Gio::Application::Flags;
--- gio/giomm/application.cc.orig	2020-01-05 10:51:41.799758115 +0000
+++ gio/giomm/application.cc	2020-01-05 10:51:59.583571397 +0000
@@ -418,7 +418,7 @@
 {
   OptionArgCallbackData* callback_data = nullptr;
   {
-    std::lock_guard<std::mutex> lock(option_arg_callback_data_mutex);
+    boost::lock_guard<boost::mutex> lock(option_arg_callback_data_mutex);
     OptionArgCallbackDataMap::iterator iterFind = option_arg_callback_data.find(long_name);
     if (iterFind != option_arg_callback_data.end())
       return; // Ignore duplicates
