--- glib/glibmm/dispatcher.cc.orig	2017-05-11 11:15:50.478230900 +0100
+++ glib/glibmm/dispatcher.cc	2017-05-11 11:16:41.730817400 +0100
@@ -32,7 +32,15 @@
 #include <io.h>
 #include <direct.h>
 #include <list>
-#include <mutex>
+#include <boost/thread.hpp>
+namespace std {
+  using boost::mutex;
+  using boost::recursive_mutex;
+  using boost::lock_guard;
+  using boost::condition_variable;
+  using boost::unique_lock;
+  using boost::thread;
+}
 #else
 #include <unistd.h>
 #endif
