--- gobject/Makefile.in.orig	2016-11-24 20:05:09.215926100 +0000
+++ gobject/Makefile.in	2016-11-24 20:05:32.328226800 +0000
@@ -893,7 +893,7 @@
 # GObject - GLib Type, Object, Parameter and Signal Library
 # Copyright (C) 1997,98,99,2000 Tim Janik and Red Hat, Inc.
 #
-SUBDIRS = . tests
+SUBDIRS = . 
 AM_CPPFLAGS = \
 	-DG_LOG_DOMAIN=\"GLib-GObject\"		\
 	$(glib_INCLUDES)			\
--- Makefile.in.orig	2016-11-24 20:45:20.078945400 +0000
+++ Makefile.in	2016-11-24 20:45:47.758654600 +0000
@@ -835,7 +835,7 @@
 
 @ENABLE_INSTALLED_TESTS_TRUE@installed_test_meta_DATA = $(installed_testcases:=.test)
 ACLOCAL_AMFLAGS = -I m4macros ${ACLOCAL_FLAGS}
-SUBDIRS = . m4macros glib gmodule gthread gobject gio po docs tests
+SUBDIRS = . m4macros glib gmodule gthread gobject gio po docs
 DIST_SUBDIRS = $(SUBDIRS) build
 bin_SCRIPTS = glib-gettextize
 AM_CPPFLAGS = \
--- Makefile.am.orig	2016-11-24 20:58:29.035384000 +0000
+++ Makefile.am	2016-11-24 20:58:42.299464100 +0000
@@ -6,7 +6,7 @@
 
 ACLOCAL_AMFLAGS = -I m4macros ${ACLOCAL_FLAGS}
 
-SUBDIRS = . m4macros glib gmodule gthread gobject gio po docs tests
+SUBDIRS = . m4macros glib gmodule gthread gobject gio po docs
 DIST_SUBDIRS = $(SUBDIRS) build
 
 bin_SCRIPTS = glib-gettextize
