--- glib/glibmm/dispatcher.cc.orig	2016-05-09 22:21:36.059000000 +0000
+++ glib/glibmm/dispatcher.cc	2016-05-09 22:22:34.010000000 +0000
@@ -35,7 +35,16 @@
 #include <io.h>
 #include <direct.h>
 #include <list>
-#include <mutex>
+#include <boost/thread.hpp>
+
+namespace std {
+  using boost::mutex;
+  using boost::recursive_mutex;
+  using boost::lock_guard;
+  using boost::condition_variable;
+  using boost::unique_lock;
+  using boost::thread;
+}
 #else
 #include <unistd.h>
 #endif
