--- gmp-h.in.orig	2025-04-18 16:03:31.455093780 +0100
+++ gmp-h.in	2025-04-18 16:04:14.454620710 +0100
@@ -102,7 +102,7 @@
 
 #if defined (__GNUC__)
 #define __GMP_DECLSPEC_EXPORT  __declspec(__dllexport__)
-#define __GMP_DECLSPEC_IMPORT  __declspec(__dllimport__)
+#define __GMP_DECLSPEC_IMPORT
 #endif
 #if defined (_MSC_VER) || defined (__BORLANDC__)
 #define __GMP_DECLSPEC_EXPORT  __declspec(dllexport)
