--- src/socket.c.orig	2021-01-12 13:21:47.240000000 +0000
+++ src/socket.c	2021-01-12 13:23:19.320000000 +0000
@@ -523,19 +523,6 @@
 		}
 
 		if (hints.ai_socktype == SOCK_DGRAM) {
-#if defined(IP_DONTFRAG)
-			int yes = 1;
-			if (setsockopt(sd, IPPROTO_IP, IP_DONTFRAG,
-				       (const void *) &yes,
-				       sizeof(yes)) < 0)
-				perror("setsockopt(IP_DF) failed");
-#elif defined(IP_MTU_DISCOVER)
-			int yes = IP_PMTUDISC_DO;
-			if (setsockopt(sd, IPPROTO_IP, IP_MTU_DISCOVER,
-				       (const void *) &yes,
-				       sizeof(yes)) < 0)
-				perror("setsockopt(IP_DF) failed");
-#endif
 		}
 
 		if (fastopen && ptr->ai_socktype == SOCK_STREAM
--- src/serv.c.orig	2021-01-12 13:47:59.740000000 +0000
+++ src/serv.c	2021-01-12 13:48:26.390000000 +0000
@@ -989,19 +989,6 @@
 				continue;
 			}
 		} else {
-#if defined(IP_DONTFRAG)
-			yes = 1;
-			if (setsockopt(s, IPPROTO_IP, IP_DONTFRAG,
-				       (const void *) &yes,
-				       sizeof(yes)) < 0)
-				perror("setsockopt(IP_DF) failed");
-#elif defined(IP_MTU_DISCOVER)
-			yes = IP_PMTUDISC_DO;
-			if (setsockopt(s, IPPROTO_IP, IP_MTU_DISCOVER,
-				       (const void *) &yes,
-				       sizeof(yes)) < 0)
-				perror("setsockopt(IP_DF) failed");
-#endif
 		}
 
 		if (bind(s, ptr->ai_addr, ptr->ai_addrlen) < 0) {
