--- makefile.orig	2020-02-19 12:56:36.346915289 +0000
+++ makefile	2020-02-19 13:00:21.084120400 +0000
@@ -101,8 +101,8 @@
 # export COMP_SYS=intel_linux
 #
 # can uncomment following two lines to use gcc/gfortran
-#export CC=gcc
-#export FC=gfortran
+export CC=x86_64-w64-mingw32-gcc
+export FC=x86_64-w64-mingw32-gfortran
 #
 # for OS-X: uncomment line for makefile -f scripts/makefile.darwin
 #
@@ -115,8 +115,8 @@
 
 
 # Warning do not set both USE_NETCDF3 and USE_NETCDF4 to one
-USE_NETCDF3=1
-USE_NETCDF4=0
+USE_NETCDF3=0
+USE_NETCDF4=1
 USE_REGEX=1
 USE_TIGGE=1
 USE_MYSQL=0
--- makefile.orig	2020-02-20 14:29:11.682080785 +0000
+++ makefile	2020-02-20 14:29:25.073963004 +0000
@@ -613,7 +613,7 @@
 w=wgrib2
 prog=$w/wgrib2
 
-all:	${netcdf4src} ${hdf5src} ${prog} aux_progs/gmerge aux_progs/smallest_grib2 aux_progs/smallest_4
+all:	${prog} aux_progs/gmerge aux_progs/smallest_grib2 aux_progs/smallest_4
 
 
 ${prog}:        $w/*.c $w/*.h ${jlib} ${aeclib} ${netcdf3lib} ${pnglib} ${hdf5lib} ${g2clib} ${netcdf4lib} ${iplib} ${spectrallib} ${gctpclib} ${proj4lib}
--- makefile.orig	2020-02-20 14:34:16.350844371 +0000
+++ makefile	2020-02-20 14:56:12.928767798 +0000
@@ -120,7 +120,7 @@
 USE_REGEX=1
 USE_TIGGE=1
 USE_MYSQL=0
-USE_IPOLATES=3
+USE_IPOLATES=0
 USE_SPECTRAL=0
 USE_UDF=0
 USE_OPENMP=1
--- makefile.orig	2020-02-20 21:06:53.975325527 +0000
+++ makefile	2020-02-20 21:46:05.791243794 +0000
@@ -116,7 +116,7 @@
 
 # Warning do not set both USE_NETCDF3 and USE_NETCDF4 to one
 USE_NETCDF3=0
-USE_NETCDF4=1
+USE_NETCDF4=0
 USE_REGEX=1
 USE_TIGGE=1
 USE_MYSQL=0
@@ -141,13 +141,13 @@
 # often enviroment variable FC=fortran compiler, is set to f77, needs f95+ compiler
 #
 cwd:=${CURDIR}
-lib:=${cwd}/lib
+lib:=${mingw_w64_x86_64_prefix}/lib
 tmp:=${cwd}/tmp
 export TMPDIR=${tmp}
 wLDFLAGS:=-L${lib}
 a:=$(shell mkdir -p ${lib})
 a:=$(shell mkdir -p ${tmp})
-wCPPFLAGS:=-I${cwd}/include
+wCPPFLAGS:=-I${mingw_w64_x86_64_prefix}/include
 # netcdf4 should be compiled without OpenMP
 netcdf3CPPFLAGS:=-I${cwd}/include
 netcdf4CPPFLAGS:=-I${cwd}/include
@@ -466,10 +466,10 @@
 
 jasperdir=${cwd}/jasper-1.900.1
 ifeq ($(USE_JASPER),1)
-   jsrc=jasper-1.900.1-14ubuntu3.2.debian.tgz
+#   jsrc=jasper-1.900.1-14ubuntu3.2.debian.tgz
    jlib=${lib}/libjasper.a
    wLDFLAGS+=-ljasper
-   wCPPFLAGS+=-I${jasperdir}/src/libjasper/include
+   wCPPFLAGS+=-I${mingw_w64_x86_64_prefix}/include/jasper
    a:=$(shell echo "\#define USE_JASPER" >> ${CONFIG_H})
 else
    a:=$(shell echo "//\#define USE_JASPER" >> ${CONFIG_H})
@@ -479,7 +479,7 @@
 
 aecdir=${cwd}/libaec-1.0.2
 ifeq ($(USE_AEC),1)
-   aecsrc=libaec-1.0.2.tar.gz
+#   aecsrc=libaec-1.0.2.tar.gz
    aeclib=${lib}/libaec.a
    wLDFLAGS+=-laec
    a:=$(shell echo "\#define USE_AEC \"${aecsrc}\"" >> ${CONFIG_H})
@@ -509,9 +509,9 @@
 hdf5dir:=${cwd}/hdf5-1.10.4
 netcdf4dir:=${cwd}/netcdf-4.6.1
 ifeq ($(USE_NETCDF4),1)
-   netcdf4src=netcdf-4.6.1.tar.gz
+#   netcdf4src=netcdf-4.6.1.tar.gz
    netcdf4lib:=${lib}/libnetcdf.a
-   hdf5src:=hdf5-1.10.4.tar.gz
+#   hdf5src:=hdf5-1.10.4.tar.gz
    hdf5lib:=${lib}/libhdf5.a
    wLDFLAGS+=-lnetcdf -lhdf5_hl -lhdf5 -ldl
 #   wCPPFLAGS+=-I${netcdf4dir}/include -I${hdf5dir}/src -I${hdf5dir}/hl/src
@@ -573,7 +573,7 @@
 
 pngdir=${cwd}/libpng-1.2.57
 ifeq ($(USE_PNG),1)
-   pngsrc=${cwd}/libpng-1.2.57.tar.gz
+#   pngsrc=${cwd}/libpng-1.2.57.tar.gz
    pnglib=${lib}/libpng.a
    wLDFLAGS+=-lpng
 # wCPPFLAGS+=-I$p
@@ -616,7 +616,7 @@
 all:	${prog} aux_progs/gmerge aux_progs/smallest_grib2 aux_progs/smallest_4
 
 
-${prog}:        $w/*.c $w/*.h ${jlib} ${aeclib} ${netcdf3lib} ${pnglib} ${hdf5lib} ${g2clib} ${netcdf4lib} ${iplib} ${spectrallib} ${gctpclib} ${proj4lib}
+${prog}:        $w/*.c $w/*.h 
 	cd "$w" && export LDFLAGS="${wLDFLAGS}" && export CPPFLAGS="${wCPPFLAGS}" && ${MAKE}
 
 fast:        $w/*.c $w/*.h ${jlib} ${aeclib} ${netcdf3lib} ${pnglib} ${hdf5lib} ${g2clib} ${netcdf4lib} ${iplib} ${spectrallib} ${gctpclib} ${proj4lib}
--- makefile.orig	2020-02-20 21:55:31.689291890 +0000
+++ makefile	2020-02-20 21:55:56.832935577 +0000
@@ -597,7 +597,7 @@
 endif
 
 wLDFLAGS+=-lm
-wCPPFLAGS+=-I/usr/include ${CPPFLAGS}
+wCPPFLAGS+=-I${mingw_w64_x86_64_prefix}/include ${CPPFLAGS}
 
 # -----------------------------------------------------
 
--- makefile.orig	2020-02-20 22:01:02.002546691 +0000
+++ makefile	2020-02-20 22:01:08.130432285 +0000
@@ -123,7 +123,7 @@
 USE_IPOLATES=0
 USE_SPECTRAL=0
 USE_UDF=0
-USE_OPENMP=1
+USE_OPENMP=0
 USE_PROJ4=0
 USE_WMO_VALIDATION=0
 DISABLE_TIMEZONE=0
--- wgrib2/enc_jpeg2000_clone.c.orig	2020-02-20 22:39:23.386386072 +0000
+++ wgrib2/enc_jpeg2000_clone.c	2020-02-20 22:39:57.529952216 +0000
@@ -126,13 +126,14 @@
    is only used for japser version 1.x up to 1.9.24
    for newer code the above line is removed
  */
+#if 0
 #if JAS_VERSION_MAJOR == 1
    image.inmem_=1;
 #endif
 #ifndef JAS_VERSION_MAJOR
    image.inmem_=1;
 #endif
-
+#endif
 
 
 
--- makefile.orig	2020-02-20 22:56:54.793509477 +0000
+++ makefile	2020-02-20 22:57:06.217515679 +0000
@@ -126,7 +126,7 @@
 USE_OPENMP=0
 USE_PROJ4=0
 USE_WMO_VALIDATION=0
-DISABLE_TIMEZONE=0
+DISABLE_TIMEZONE=1
 MAKE_FTN_API=1
 DISABLE_ALARM=0
 
--- makefile.orig	2020-02-20 23:05:05.840301359 +0000
+++ makefile	2020-02-20 23:06:45.055803495 +0000
@@ -617,7 +617,7 @@
 
 
 ${prog}:        $w/*.c $w/*.h 
-	cd "$w" && export LDFLAGS="${wLDFLAGS}" && export CPPFLAGS="${wCPPFLAGS}" && ${MAKE}
+	cd "$w" && export LDFLAGS="${wLDFLAGS} -lgeo" && export CPPFLAGS="${wCPPFLAGS}" && ${MAKE}
 
 fast:        $w/*.c $w/*.h ${jlib} ${aeclib} ${netcdf3lib} ${pnglib} ${hdf5lib} ${g2clib} ${netcdf4lib} ${iplib} ${spectrallib} ${gctpclib} ${proj4lib}
 	cd "$w" && export LDFLAGS="${wLDFLAGS}" && export CPPFLAGS="${wCPPFLAGS}" && ${MAKE} fast
--- makefile.orig	2020-02-20 23:33:33.036217713 +0000
+++ makefile	2020-02-20 23:33:53.492055652 +0000
@@ -617,7 +617,7 @@
 
 
 ${prog}:        $w/*.c $w/*.h 
-	cd "$w" && export LDFLAGS="${wLDFLAGS} -lgeo" && export CPPFLAGS="${wCPPFLAGS}" && ${MAKE}
+	cd "$w" && export LDFLAGS="${wLDFLAGS} -lgeo -lregex" && export CPPFLAGS="${wCPPFLAGS}" && ${MAKE}
 
 fast:        $w/*.c $w/*.h ${jlib} ${aeclib} ${netcdf3lib} ${pnglib} ${hdf5lib} ${g2clib} ${netcdf4lib} ${iplib} ${spectrallib} ${gctpclib} ${proj4lib}
 	cd "$w" && export LDFLAGS="${wLDFLAGS}" && export CPPFLAGS="${wCPPFLAGS}" && ${MAKE} fast
--- makefile.orig	2020-10-18 18:13:53.170000000 +0100
+++ makefile	2020-10-18 18:17:09.760000000 +0100
@@ -182,27 +182,27 @@
 endif
 
 ifeq ($(USE_REGEX),1)
-   a:=$(shell echo "\#define USE_REGEX" >> ${CONFIG_H})
+   a:=$(shell echo "#define USE_REGEX" >> ${CONFIG_H})
 else
-   a:=$(shell echo "//\#define USE_REGEX" >> ${CONFIG_H})
+   a:=$(shell echo "//#define USE_REGEX" >> ${CONFIG_H})
 endif
 
 ifeq ($(USE_TIGGE),1)
-   a:=$(shell echo "\#define USE_TIGGE" >> ${CONFIG_H})
+   a:=$(shell echo "#define USE_TIGGE" >> ${CONFIG_H})
 else
-   a:=$(shell echo "//\#define USE_TIGGE" >> ${CONFIG_H})
+   a:=$(shell echo "//#define USE_TIGGE" >> ${CONFIG_H})
 endif
 
 ifeq ($(DISABLE_ALARM),1)
    a:=$(shell echo "#define DISABLE_ALARM" >> ${CONFIG_H})
 else
-   a:=$(shell echo "//\#define DISABLE_ALARM" >> ${CONFIG_H})
+   a:=$(shell echo "//#define DISABLE_ALARM" >> ${CONFIG_H})
 endif
 
 ifeq ($(DISABLE_TIMEZONE),1)
-   a:=$(shell echo "\#define DISABLE_TIMEZONE" >> ${CONFIG_H})
+   a:=$(shell echo "#define DISABLE_TIMEZONE" >> ${CONFIG_H})
 else
-   a:=$(shell echo "//\#define DISABLE_TIMEZONE" >> ${CONFIG_H})
+   a:=$(shell echo "//#define DISABLE_TIMEZONE" >> ${CONFIG_H})
 endif
 
 need_ftn=0
@@ -214,9 +214,9 @@
 endif
 
 ifeq ($(USE_UDF),1)
-   a:=$(shell echo "\#define USE_UDF" >> ${CONFIG_H})
+   a:=$(shell echo "#define USE_UDF" >> ${CONFIG_H})
 else
-   a:=$(shell echo "//\#define USE_UDF" >> ${CONFIG_H})
+   a:=$(shell echo "//#define USE_UDF" >> ${CONFIG_H})
 endif
 
 ifeq ($(USE_IPOLATES),1)
@@ -229,13 +229,13 @@
    ipdir:=${cwd}/ip2lib_d
    iplib=${lib}/libip2_d.a
    wLDFLAGS+=-lip2_d
-   a:=$(shell echo "\#define USE_SPECTRAL ${USE_SPECTRAL}" > ${ipdir}/config.h)
+   a:=$(shell echo "#define USE_SPECTRAL ${USE_SPECTRAL}" > ${ipdir}/config.h)
 endif
 
 ifneq ($(USE_IPOLATES),0)
-a:=$(shell echo "\#define IPOLATES_LIB \"`basename ${ipdir}`\"" >> ${CONFIG_H})
+a:=$(shell echo "#define IPOLATES_LIB \"`basename ${ipdir}`\"" >> ${CONFIG_H})
 endif
-a:=$(shell echo "\#define USE_IPOLATES ${USE_IPOLATES}" >> ${CONFIG_H})
+a:=$(shell echo "#define USE_IPOLATES ${USE_IPOLATES}" >> ${CONFIG_H})
 
 # C compile and load commmands
 # wCPPFLAGS has the directory of the includes 
@@ -308,7 +308,7 @@
       FTN_REAL8:="-fdefault-real-8 -fdefault-double-8"
    endif
    ifeq ($(USE_OPENMP),1)
-      a:=$(shell echo "\#define USE_OPENMP" >> ${CONFIG_H})
+      a:=$(shell echo "#define USE_OPENMP" >> ${CONFIG_H})
       wCPPFLAGS+=-fopenmp
       wFFLAGS+=-fopenmp
    endif
@@ -347,7 +347,7 @@
    hdf5CFLAGS+=-O2
    ifeq ($(USE_JASPER),1)
       CCjasper:=gcc
-      a:=$(shell echo "\#define CC_jasper \"${CCjasper}\"" >> ${CONFIG_H})
+      a:=$(shell echo "#define CC_jasper \"${CCjasper}\"" >> ${CONFIG_H})
    endif
    ifeq ($(need_ftn),1)
       wCPPFLAGS+=-DIFORT -cxxlib
@@ -356,7 +356,7 @@
       FTN_REAL8:="-r8"
    endif
    ifeq ($(USE_OPENMP),1)
-      a:=$(shell echo "\#define USE_OPENMP" >> ${CONFIG_H})
+      a:=$(shell echo "#define USE_OPENMP" >> ${CONFIG_H})
       wCPPFLAGS+=-qopenmp
       wFFLAGS+=-qopenmp
    endif
@@ -374,7 +374,7 @@
       FTN_REAL8:=???
    endif
    ifeq ($(USE_OPENMP),1)
-      a:=$(shell echo "\#define USE_OPENMP" >> ${CONFIG_H})
+      a:=$(shell echo "#define USE_OPENMP" >> ${CONFIG_H})
       wCPPFLAGS+=-xopenmp
       wFFLAGS+=-xopenmp
    endif
@@ -438,9 +438,9 @@
    g2clib:=${lib}/libgrib2c.a
    wLDFLAGS+=-lgrib2c
    wCPPFLAGS+=-I$g
-   a:=$(shell echo "\#define USE_G2CLIB" >> ${CONFIG_H})
+   a:=$(shell echo "#define USE_G2CLIB" >> ${CONFIG_H})
 else
-   a:=$(shell echo "//\#define USE_G2CLIB" >> ${CONFIG_H})
+   a:=$(shell echo "//#define USE_G2CLIB" >> ${CONFIG_H})
 endif
 
 # gctpc library
@@ -457,9 +457,9 @@
    proj4lib:=${lib}/libproj.a
    wLDFLAGS+=-lproj
 #   wCPPFLAGS+=-I${proj4dir}/src
-   a:=$(shell echo "\#define USE_PROJ4" >> ${CONFIG_H})
+   a:=$(shell echo "#define USE_PROJ4" >> ${CONFIG_H})
 else
-   a:=$(shell echo "//\#define USE_PROJ4" >> ${CONFIG_H})
+   a:=$(shell echo "//#define USE_PROJ4" >> ${CONFIG_H})
 endif
 
 # Jasper
@@ -470,9 +470,9 @@
    jlib=${lib}/libjasper.a
    wLDFLAGS+=-ljasper
    wCPPFLAGS+=-I${mingw_w64_x86_64_prefix}/include/jasper
-   a:=$(shell echo "\#define USE_JASPER" >> ${CONFIG_H})
+   a:=$(shell echo "#define USE_JASPER" >> ${CONFIG_H})
 else
-   a:=$(shell echo "//\#define USE_JASPER" >> ${CONFIG_H})
+   a:=$(shell echo "//#define USE_JASPER" >> ${CONFIG_H})
 endif
 
 # AEC
@@ -482,9 +482,9 @@
 #   aecsrc=libaec-1.0.2.tar.gz
    aeclib=${lib}/libaec.a
    wLDFLAGS+=-laec
-   a:=$(shell echo "\#define USE_AEC \"${aecsrc}\"" >> ${CONFIG_H})
+   a:=$(shell echo "#define USE_AEC \"${aecsrc}\"" >> ${CONFIG_H})
 else
-   a:=$(shell echo "//\#define USE_AEC" >> ${CONFIG_H})
+   a:=$(shell echo "//#define USE_AEC" >> ${CONFIG_H})
 endif
 
 # SPECTRAL (optional used by ip2_d)
@@ -492,7 +492,7 @@
 ifeq ($(USE_SPECTRAL),1)
    spectrallib:=${lib}/libsp_v2.0.2_d.a
    wLDFLAGS+=-lsp_v2.0.2_d
-   a:=$(shell echo "\#define USE_SPECTRAL 1" >> ${CONFIG_H})
+   a:=$(shell echo "#define USE_SPECTRAL 1" >> ${CONFIG_H})
 endif
 
 netcdf3dir:=${cwd}/netcdf-3.6.3
@@ -501,9 +501,9 @@
    netcdf3lib:=${lib}/libnetcdf.a
    wLDFLAGS+=-lnetcdf
 #   wCPPFLAGS+=-I$n/libsrc
-   a:=$(shell echo "\#define USE_NETCDF3" >> ${CONFIG_H})
+   a:=$(shell echo "#define USE_NETCDF3" >> ${CONFIG_H})
 else
-   a:=$(shell echo "//\#define USE_NETCDF3" >> ${CONFIG_H})
+   a:=$(shell echo "//#define USE_NETCDF3" >> ${CONFIG_H})
 endif
 
 hdf5dir:=${cwd}/hdf5-1.10.4
@@ -515,18 +515,18 @@
    hdf5lib:=${lib}/libhdf5.a
    wLDFLAGS+=-lnetcdf -lhdf5_hl -lhdf5 -ldl
 #   wCPPFLAGS+=-I${netcdf4dir}/include -I${hdf5dir}/src -I${hdf5dir}/hl/src
-   a:=$(shell echo "\#define USE_NETCDF4" >> ${CONFIG_H})
-   a:=$(shell echo "\#define HDF5 \"${hdf5src}\"" >> ${CONFIG_H})
+   a:=$(shell echo "#define USE_NETCDF4" >> ${CONFIG_H})
+   a:=$(shell echo "#define HDF5 \"${hdf5src}\"" >> ${CONFIG_H})
 else
-   a:=$(shell echo "//\#define USE_NETCDF4" >> ${CONFIG_H})
+   a:=$(shell echo "//#define USE_NETCDF4" >> ${CONFIG_H})
 endif
 
 ifeq ($(USE_MYSQL),1)
    wCPPFLAGS+=`mysql_config --cflags`
    wLDFLAGS+=`mysql_config --libs`
-   a:=$(shell echo "\#define USE_MYSQL" >> ${CONFIG_H})
+   a:=$(shell echo "#define USE_MYSQL" >> ${CONFIG_H})
 else
-   a:=$(shell echo "//\#define USE_MYSQL" >> ${CONFIG_H})
+   a:=$(shell echo "//#define USE_MYSQL" >> ${CONFIG_H})
 endif
 
 # OPENMP .. only select configurations
@@ -534,14 +534,14 @@
 ifeq ($(USE_OPENMP),1)
    ifeq ($(findstring opencc,$(notdir $(CC))),opencc)
       ifeq ($(findstring openf95,$(notdir $(FC))),openf95)
-	 a:=$(shell echo "\#define USE_OPENMP" >> ${CONFIG_H})
+	 a:=$(shell echo "#define USE_OPENMP" >> ${CONFIG_H})
 	 wCPPFLAGS+=-fopenmp
 	 wFFLAGS+=-fopenmp
       endif
    endif
    ifeq ($(findstring xlc_r,$(notdir $(CC))),xlc_r)
       ifeq ($(findstring xlf_r,$(notdir $(FC))),xlf_r)
-	 a:=$(shell echo "\#define USE_OPENMP" >> ${CONFIG_H})
+	 a:=$(shell echo "#define USE_OPENMP" >> ${CONFIG_H})
 	 wCPPFLAGS+=-qsmp=omp
 	 wFFLAGS+=-qsmp=omp
       endif
@@ -552,22 +552,22 @@
 # save fortran and C compiler names in config.h file
 
 ifeq ($(findstring gcc,$(notdir $(CC))),gcc)
-   a:=$(shell echo "\#define CC \"`${CC} --version | head -n 1`\"" >> ${CONFIG_H})
+   a:=$(shell echo "#define CC \"`${CC} --version | head -n 1`\"" >> ${CONFIG_H})
 else ifeq ($(findstring icc,$(notdir $(CC))),icc)
-   a:=$(shell echo "\#define CC \"`${CC} --version | head -n 1`\"" >> ${CONFIG_H})
+   a:=$(shell echo "#define CC \"`${CC} --version | head -n 1`\"" >> ${CONFIG_H})
 else
-   a:=$(shell echo "\#define CC \"${CC}\"" >> ${CONFIG_H})
+   a:=$(shell echo "#define CC \"${CC}\"" >> ${CONFIG_H})
 endif
 
 ifeq ($(findstring gfortran,$(notdir $(FC))),gfortran)
-   a:=$(shell echo "\#define FORTRAN \"`${FC} --version | head -n 1`\"" >> ${CONFIG_H})
+   a:=$(shell echo "#define FORTRAN \"`${FC} --version | head -n 1`\"" >> ${CONFIG_H})
 else ifeq ($(findstring ifort,$(notdir $(FC))),ifort)
-   a:=$(shell echo "\#define FORTRAN \"`${FC} --version | head -n 1`\"" >> ${CONFIG_H})
+   a:=$(shell echo "#define FORTRAN \"`${FC} --version | head -n 1`\"" >> ${CONFIG_H})
 else
-   a:=$(shell echo "\#define FORTRAN \"${FC}\"" >> ${CONFIG_H})
+   a:=$(shell echo "#define FORTRAN \"${FC}\"" >> ${CONFIG_H})
 endif
 
-a:=$(shell echo "\#define BUILD_COMMENTS \"${BUILD_COMMENTS}\"" >> ${CONFIG_H})
+a:=$(shell echo "#define BUILD_COMMENTS \"${BUILD_COMMENTS}\"" >> ${CONFIG_H})
 
 # png 
 
@@ -577,7 +577,7 @@
    pnglib=${lib}/libpng.a
    wLDFLAGS+=-lpng
 # wCPPFLAGS+=-I$p
-   a:=$(shell echo "\#define USE_PNG" >> ${CONFIG_H})
+   a:=$(shell echo "#define USE_PNG" >> ${CONFIG_H})
 
 zdir=${cwd}/zlib-1.2.11
 # z
@@ -586,14 +586,14 @@
    wLDFLAGS+=-lz
    # wCPPFLAGS+=-I$z
 else
-   a:=$(shell echo "//\#define USE_PNG" >> ${CONFIG_H})
+   a:=$(shell echo "//#define USE_PNG" >> ${CONFIG_H})
 endif
 
 # WMO Validation testing mode
 ifeq ($(USE_WMO_VALIDATION),1)
-   a:=$(shell echo "\#define WMO_VALIDATION" >> ${CONFIG_H})
+   a:=$(shell echo "#define WMO_VALIDATION" >> ${CONFIG_H})
 else
-   a:=$(shell echo "//\#define WMO_VALIDATION" >> ${CONFIG_H})
+   a:=$(shell echo "//#define WMO_VALIDATION" >> ${CONFIG_H})
 endif
 
 wLDFLAGS+=-lm
