--- demos/gtk-demo/Makefile.am.orig	2017-11-18 16:07:40.417430400 +0000
+++ demos/gtk-demo/Makefile.am	2017-11-18 16:09:02.013132500 +0000
@@ -175,7 +175,7 @@
 dist_appsicon48_DATA = data/48x48/gtk3-demo.png data/48x48/gtk3-demo-symbolic.symbolic.png
 dist_appsicon256_DATA = data/256x256/gtk3-demo.png data/256x256/gtk3-demo-symbolic.symbolic.png
 
-update_icon_cache = $(top_builddir)/gtk/gtk-update-icon-cache$(EXEEXT) --ignore-theme-index --force
+update_icon_cache = /usr/bin/gtk-update-icon-cache --ignore-theme-index --force
 
 install-data-hook: install-update-icon-cache
 uninstall-hook: uninstall-update-icon-cache
--- demos/widget-factory/Makefile.am.orig	2017-11-18 23:33:43.814273000 +0000
+++ demos/widget-factory/Makefile.am	2017-11-18 23:34:06.471530100 +0000
@@ -47,7 +47,7 @@
 dist_appsicon48_DATA = data/48x48/gtk3-widget-factory.png data/48x48/gtk3-widget-factory-symbolic.symbolic.png
 dist_appsicon256_DATA = data/256x256/gtk3-widget-factory.png data/256x256/gtk3-widget-factory-symbolic.symbolic.png
 
-update_icon_cache = $(top_builddir)/gtk/gtk-update-icon-cache$(EXEEXT) --ignore-theme-index --force
+update_icon_cache = /usr/bin/gtk-update-icon-cache --ignore-theme-index --force
 
 install-data-hook: install-update-icon-cache
 uninstall-hook: uninstall-update-icon-cache
