--- demos/widget-factory/Makefile.am.orig	2016-07-06 20:28:09.237000000 +0000
+++ demos/widget-factory/Makefile.am	2016-07-06 20:28:33.858000000 +0000
@@ -49,8 +49,8 @@
 
 update_icon_cache = $(top_builddir)/gtk/gtk-update-icon-cache$(EXEEXT) --ignore-theme-index --force
 
-install-data-hook: install-update-icon-cache
-uninstall-hook: uninstall-update-icon-cache
+install-data-hook: #install-update-icon-cache
+uninstall-hook: #uninstall-update-icon-cache
 
 install-update-icon-cache:
 	$(AM_V_at)$(POST_INSTALL)
