--- source/i18n/digitlst.cpp.orig	2017-12-09 22:08:39.285560880 +0000
+++ source/i18n/digitlst.cpp	2017-12-09 22:09:58.193006142 +0000
@@ -61,11 +61,7 @@
 #endif
 
 #if U_USE_STRTOD_L
-# if U_PLATFORM_USES_ONLY_WIN32_API || U_PLATFORM == U_PF_CYGWIN
-#   include <locale.h>
-# else
-#   include <xlocale.h>
-# endif
+# include <locale.h>
 #endif
 
 // ***************************************************************************
