--- liblash/lash_config.c.orig	2020-03-15 11:55:50.850766795 +0000
+++ liblash/lash_config.c	2020-03-15 11:56:58.434095510 +0000
@@ -20,7 +20,7 @@
  */
 
 #include <string.h>
-#include <arpa/inet.h>
+#include <winsock2.h>
 #include <rpc/xdr.h>
 
 #include "common/debug.h"
--- liblash/lash_config.c.orig	2020-03-15 12:03:10.557560357 +0000
+++ liblash/lash_config.c	2020-03-15 12:03:26.649367760 +0000
@@ -21,6 +21,7 @@
 
 #include <string.h>
 #include <winsock2.h>
+#include <rpc/types.h>
 #include <rpc/xdr.h>
 
 #include "common/debug.h"
--- liblash/lash.c.orig	2020-03-15 12:28:00.877375617 +0000
+++ liblash/lash.c	2020-03-15 12:28:54.772815048 +0000
@@ -24,15 +24,17 @@
 #include <stdlib.h>
 #include <string.h>
 #include <sys/types.h>
-#include <sys/wait.h>
-#include <netdb.h>
+/* #include <sys/wait.h>*/
+/* #include <netdb.h> */
+#include <winsock2.h>
 #include <dbus/dbus.h>
 
 #include <errno.h>
-#include <netinet/in.h>
-#include <arpa/inet.h>
+/* #include <netinet/in.h> */
+/* #include <arpa/inet.h> */
 #include <unistd.h>
 #include <sys/param.h>
+#include <rpc/types.h>
 #include <rpc/xdr.h>
 
 #include "common/safety.h"
--- dbus/method.h.orig	2020-03-18 22:05:15.994195471 +0000
+++ dbus/method.h	2020-03-18 22:05:52.181845943 +0000
@@ -24,7 +24,7 @@
 
 #include <stdbool.h>
 #include <dbus/dbus.h>
-
+#include "dbus/interface.h"
 #include "dbus/types.h"
 
 #define DIRECTION_OUT (0)
--- lashd/file.c.orig	2020-03-18 22:24:42.616428884 +0000
+++ lashd/file.c	2020-03-18 22:25:45.303555087 +0000
@@ -145,7 +145,7 @@
 	if (stat(parent, &parentstat) == -1) {
 		lash_error("Cannot stat parent %s to create directory %s: %s",
 		           parent, dir, strerror(errno));
-	} else if (mkdir(dir, parentstat.st_mode) == -1) {
+	} else if (_mkdir(dir) == -1) {
 		lash_error("Cannot create directory %s: %s",
 		           dir, strerror(errno));
 	}
--- liblash/lash_config.c.orig	2020-03-19 14:27:56.401466011 +0000
+++ liblash/lash_config.c	2020-03-19 14:28:28.541261490 +0000
@@ -30,6 +30,7 @@
 
 #include "lash_config.h"
 
+#include "dbus/interface.h"
 #include "dbus/method.h"
 
 bool
--- lashd/file.c.orig	2020-03-19 14:56:06.029303930 +0000
+++ lashd/file.c	2020-03-19 14:56:33.441024296 +0000
@@ -24,6 +24,7 @@
 #include <string.h>
 #include <sys/types.h>
 #include <dirent.h>
+#include <direct.h>
 #include <sys/stat.h>
 #include <unistd.h>
 #include <errno.h>
--- liblash/lash.c.orig	2020-03-19 21:32:24.508063805 +0000
+++ liblash/lash.c	2020-03-19 21:32:57.947782098 +0000
@@ -26,7 +26,7 @@
 #include <sys/types.h>
 /* #include <sys/wait.h>*/
 /* #include <netdb.h> */
-#include <winsock2.h>
+/* #include <winsock2.h> */
 #include <dbus/dbus.h>
 
 #include <errno.h>
--- liblash/lash_config.c.orig	2020-03-19 21:36:47.321748191 +0000
+++ liblash/lash_config.c	2020-03-19 21:36:57.665652975 +0000
@@ -20,7 +20,7 @@
  */
 
 #include <string.h>
-#include <winsock2.h>
+/* #include <winsock2.h> */
 #include <rpc/types.h>
 #include <rpc/xdr.h>
 
--- liblash/Makefile.am.orig	2020-03-19 21:40:05.431884975 +0000
+++ liblash/Makefile.am	2020-03-19 21:40:41.255540337 +0000
@@ -37,7 +37,7 @@
 	$(DBUS_LIBS)
 
 liblash_la_LDFLAGS = \
-	-export-dynamic -version-info 2:1:1
+	-export-dynamic -version-info 2:1:1 -no-undefined
 
 # Be sure to read before updating version info:
 #   http://www.gnu.org/software/libtool/manual/html_node/Updating-version-info.html
--- liblash/Makefile.am.orig	2020-03-19 21:52:09.520655742 +0000
+++ liblash/Makefile.am	2020-03-19 21:52:29.888447290 +0000
@@ -34,7 +34,8 @@
 
 liblash_la_LIBADD = \
 	$(UUID_LIBS) \
-	$(DBUS_LIBS)
+	$(DBUS_LIBS) \
+	-lportablexdr 
 
 liblash_la_LDFLAGS = \
 	-export-dynamic -version-info 2:1:1 -no-undefined
--- liblash/Makefile.am.orig	2020-03-19 21:55:55.254337554 +0000
+++ liblash/Makefile.am	2020-03-19 21:56:19.294089789 +0000
@@ -35,7 +35,7 @@
 liblash_la_LIBADD = \
 	$(UUID_LIBS) \
 	$(DBUS_LIBS) \
-	-lportablexdr 
+	-lportablexdr -lws2_32 
 
 liblash_la_LDFLAGS = \
 	-export-dynamic -version-info 2:1:1 -no-undefined
--- dbus/interface.h.orig	2020-03-20 12:57:47.332247892 +0000
+++ dbus/interface.h	2020-03-20 12:58:01.260097449 +0000
@@ -27,7 +27,7 @@
 #include "dbus/types.h"
 #include "dbus/method.h"
 #include "dbus/signal.h"
-
+#undef interface
 struct _interface
 {
 	const char                *name;
--- dbus/method.h.orig	2020-03-20 14:07:34.304875895 +0000
+++ dbus/method.h	2020-03-20 14:08:43.212096036 +0000
@@ -27,6 +27,11 @@
 #include "dbus/interface.h"
 #include "dbus/types.h"
 
+/* Conflict with rpc.h */
+#ifdef interface
+#undef interface
+#endif
+
 #define DIRECTION_OUT (0)
 #define DIRECTION_IN  (1)
 
--- lashd/main.c.orig	2020-03-20 22:07:41.374823202 +0000
+++ lashd/main.c	2020-03-20 23:18:52.788742836 +0000
@@ -49,6 +49,9 @@
 #include "sigsegv.h"
 #include "proctitle.h"
 
+typedef void (*__sighandler_t) (int);
+typedef __sighandler_t sig_t;
+
 void
 term_handler(int signum)
 {
--- lashd/main.c.orig	2020-03-20 23:29:55.869285448 +0000
+++ lashd/main.c	2020-03-20 23:30:21.101004339 +0000
@@ -35,6 +35,7 @@
 
 #include <jack/jack.h>
 #include <libxml/tree.h>
+extern const char *strsignal (int);
 
 #include "common/debug.h"
 
--- lashd/main.c.orig	2020-03-20 23:38:18.859698370 +0000
+++ lashd/main.c	2020-03-20 23:38:36.279505336 +0000
@@ -30,6 +30,7 @@
 #include <stdlib.h>
 #include <signal.h>
 #include <limits.h>
+#include <time.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 
--- lashd/main.c.orig	2020-03-22 21:16:05.334763865 +0000
+++ lashd/main.c	2020-03-22 21:16:40.134260446 +0000
@@ -30,6 +30,7 @@
 #include <stdlib.h>
 #include <signal.h>
 #include <limits.h>
+#define _POSIX_C_SOURCE 1
 #include <time.h>
 #include <sys/types.h>
 #include <sys/stat.h>
--- lashd/main.c.orig	2020-03-22 21:26:44.382121540 +0000
+++ lashd/main.c	2020-03-22 21:27:33.357493909 +0000
@@ -161,11 +161,11 @@
 	if (sigh == SIG_IGN)
 		signal(SIGINT, SIG_IGN);
 
-	sigh = signal(SIGHUP, term_handler);
+	/* sigh = signal(SIGHUP, term_handler);
 	if (sigh == SIG_IGN)
-		signal(SIGHUP, SIG_IGN);
+		signal(SIGHUP, SIG_IGN); */
 
-	signal(SIGPIPE, SIG_IGN);
+	/* signal(SIGPIPE, SIG_IGN); */
 
 	/* setup our SIGSEGV magic that prints nice stack in our logfile */ 
 	setup_sigsegv();
--- lashd/log.c.orig	2020-03-22 21:30:47.063037612 +0000
+++ lashd/log.c	2020-03-22 21:31:14.402693739 +0000
@@ -49,7 +49,7 @@
 		if (errno == ENOENT)
 		{
 			lash_info("Directory \"%s\" does not exist. Creating...", dirname);
-			if (mkdir(dirname, mode) != 0)
+			if (mkdir(dirname) != 0)
 			{
 				lash_error("Failed to create \"%s\" directory: %d (%s)", dirname, errno, strerror(errno));
 				return false;
