--- src/cxml.cc.orig	2017-11-04 14:02:24.018937700 +0000
+++ src/cxml.cc	2017-11-04 14:04:02.846367100 +0000
@@ -46,10 +46,10 @@
 	return _node->get_name ();
 }
 
-shared_ptr<cxml::Node>
+boost::shared_ptr<cxml::Node>
 cxml::Node::node_child (string name) const
 {
-	list<shared_ptr<cxml::Node> > n = node_children (name);
+	list<boost::shared_ptr<cxml::Node> > n = node_children (name);
 	if (n.size() > 1) {
 		throw cxml::Error ("duplicate XML tag " + name);
 	} else if (n.empty ()) {
@@ -59,20 +59,20 @@
 	return n.front ();
 }
 
-shared_ptr<cxml::Node>
+boost::shared_ptr<cxml::Node>
 cxml::Node::optional_node_child (string name) const
 {
-	list<shared_ptr<cxml::Node> > n = node_children (name);
+	list<boost::shared_ptr<cxml::Node> > n = node_children (name);
 	if (n.size() > 1) {
 		throw cxml::Error ("duplicate XML tag " + name);
 	} else if (n.empty ()) {
-		return shared_ptr<cxml::Node> ();
+		return boost::shared_ptr<cxml::Node> ();
 	}
 	
 	return n.front ();
 }
 
-list<shared_ptr<cxml::Node> >
+list<boost::shared_ptr<cxml::Node> >
 cxml::Node::node_children (string name) const
 {
 	/* XXX: using find / get_path should work here, but I can't follow
@@ -81,10 +81,10 @@
 
 	xmlpp::Node::NodeList c = _node->get_children ();
 	
-	list<shared_ptr<cxml::Node> > n;
+	list<boost::shared_ptr<cxml::Node> > n;
 	for (xmlpp::Node::NodeList::iterator i = c.begin (); i != c.end(); ++i) {
 		if ((*i)->get_name() == name) {
-			n.push_back (shared_ptr<Node> (new Node (*i)));
+			n.push_back (boost::shared_ptr<Node> (new Node (*i)));
 		}
 	}
 	
@@ -101,7 +101,7 @@
 optional<string>
 cxml::Node::optional_string_child (string c) const
 {
-	list<shared_ptr<Node> > nodes = node_children (c);
+	list<boost::shared_ptr<Node> > nodes = node_children (c);
 	if (nodes.size() > 1) {
 		throw cxml::Error ("duplicate XML tag " + c);
 	}
