--- meson.build.orig	2020-07-04 15:32:33.030000000 +0100
+++ meson.build	2020-07-04 15:34:12.110000000 +0100
@@ -33,6 +33,7 @@
 
 dep_tasn1 = dependency('libtasn1')
 dep_thread = dependency('threads')
+dep_winsock2 = cc.find_library('ws2_32', required:true)
 dep_rt = []
 
 optional_arguments = [
@@ -232,7 +233,7 @@
   version: dsm_soname_version,
   soversion: dsm_soversion,
   include_directories: includes,
-  dependencies: [dep_tasn1, dep_thread, dep_iconv, dep_log],
+  dependencies: [dep_tasn1, dep_thread, dep_iconv, dep_winsock2, dep_log],
   install: true
 )
 
--- meson.build.orig	2020-07-04 15:52:59.700000000 +0100
+++ meson.build	2020-07-04 15:53:35.710000000 +0100
@@ -17,7 +17,7 @@
 
 conf_data = configuration_data()
 conf_data.set_quoted('URANDOM', '/dev/urandom') #TODO
-conf_data.set('_FORTIFY_SOURCE', 2,
+conf_data.set('_FORTIFY_SOURCE', 0,
               description: 'Define to 2 to get glibc warnings.')
 conf_data.set('_FILE_OFFSET_BITS', 64,
               description: 'Define to 64 for large file support.')
