--- src/reel.cc.orig	2021-01-21 20:12:48.620000000 +0000
+++ src/reel.cc	2021-01-21 20:14:04.500000000 +0000
@@ -62,39 +62,39 @@
 Reel::Reel (boost::shared_ptr<const cxml::Node> node)
 	: Object (remove_urn_uuid (node->string_child ("Id")))
 {
-	shared_ptr<cxml::Node> asset_list = node->node_child ("AssetList");
+	boost::shared_ptr<cxml::Node> asset_list = node->node_child ("AssetList");
 
-	shared_ptr<cxml::Node> main_picture = asset_list->optional_node_child ("MainPicture");
+	boost::shared_ptr<cxml::Node> main_picture = asset_list->optional_node_child ("MainPicture");
 	if (main_picture) {
 		_main_picture.reset (new ReelMonoPictureAsset (main_picture));
 	}
 
-	shared_ptr<cxml::Node> main_stereoscopic_picture = asset_list->optional_node_child ("MainStereoscopicPicture");
+	boost::shared_ptr<cxml::Node> main_stereoscopic_picture = asset_list->optional_node_child ("MainStereoscopicPicture");
 	if (main_stereoscopic_picture) {
 		_main_picture.reset (new ReelStereoPictureAsset (main_stereoscopic_picture));
 	}
 
-	shared_ptr<cxml::Node> main_sound = asset_list->optional_node_child ("MainSound");
+	boost::shared_ptr<cxml::Node> main_sound = asset_list->optional_node_child ("MainSound");
 	if (main_sound) {
 		_main_sound.reset (new ReelSoundAsset (main_sound));
 	}
 
-	shared_ptr<cxml::Node> main_subtitle = asset_list->optional_node_child ("MainSubtitle");
+	boost::shared_ptr<cxml::Node> main_subtitle = asset_list->optional_node_child ("MainSubtitle");
 	if (main_subtitle) {
 		_main_subtitle.reset (new ReelSubtitleAsset (main_subtitle));
 	}
 
 	/* XXX: it's not ideal that we silently tolerate Interop or SMPTE nodes here */
 	/* XXX: not sure if Interop supports multiple closed captions */
-	list<shared_ptr<cxml::Node> > closed_captions = asset_list->node_children ("MainClosedCaption");
+	list<boost::shared_ptr<cxml::Node> > closed_captions = asset_list->node_children ("MainClosedCaption");
 	if (closed_captions.empty()) {
 		closed_captions = asset_list->node_children ("ClosedCaption");
 	}
-	BOOST_FOREACH (shared_ptr<cxml::Node> i, closed_captions) {
+	BOOST_FOREACH (boost::shared_ptr<cxml::Node> i, closed_captions) {
 		_closed_captions.push_back (shared_ptr<ReelClosedCaptionAsset>(new ReelClosedCaptionAsset(i)));
 	}
 
-	shared_ptr<cxml::Node> atmos = asset_list->optional_node_child ("AuxData");
+	boost::shared_ptr<cxml::Node> atmos = asset_list->optional_node_child ("AuxData");
 	if (atmos) {
 		_atmos.reset (new ReelAtmosAsset (atmos));
 	}
