--- cscript.orig	2019-02-11 13:24:48.444500934 +0000
+++ cscript	2019-02-11 13:25:30.620452820 +0000
@@ -83,7 +83,7 @@
         else:
             shutil.copyfile('%s/bin/libgcc_s_seh-1.dll' % target.windows_prefix, os.path.join(d, 'libgcc_s_seh-1.dll'))
 
-        for f in ['libstdc++-6.dll', 'libboost_filesystem-mt.dll', 'libboost_system-mt.dll', 'libeay32.dll', 'libglibmm-2.4-1.dll', 'libxml++-2.6-2.dll',
+        for f in ['libstdc++-6.dll', 'libboost_filesystem-mt-x64.dll', 'libboost_system-mt-x64.dll', 'libeay32.dll', 'libglibmm-2.4-1.dll', 'libxml++-2.6-2.dll',
                   'libxml2-2.dll', 'libxmlsec1-openssl.dll', 'libxmlsec1.dll', 'zlib1.dll', 'libglib-2.0-0.dll', 'libgmodule-2.0-0.dll', 'libgobject-2.0-0.dll',
                   'libsigc-2.0-0.dll', 'libiconv-2.dll', 'liblzma-5.dll', 'libxslt-1.dll', 'libintl-8.dll', 'libpcre-1.dll', 'libffi-6.dll']:
             shutil.copyfile('%s/bin/%s' % (target.windows_prefix, f), os.path.join(d, f))
--- wscript.orig	2021-04-21 18:14:31.320000000 +0100
+++ wscript	2021-04-21 18:14:56.440000000 +0100
@@ -196,7 +196,7 @@
     			    int main() { boost::filesystem::copy_file ("a", "b"); }\n
 			    """,
                    msg='Checking for boost filesystem library',
-                   libpath='/usr/local/lib',
+                   libpath='',
                    lib=['boost_filesystem%s' % boost_lib_suffix, 'boost_system%s' % boost_lib_suffix],
                    uselib_store='BOOST_FILESYSTEM')
 
@@ -212,7 +212,7 @@
     			    int main() { boost::gregorian::day_clock::local_day(); }\n
 			    """,
                    msg='Checking for boost datetime library',
-                   libpath='/usr/local/lib',
+                   libpath='',
                    lib=['boost_date_time%s' % boost_lib_suffix, 'boost_system%s' % boost_lib_suffix],
                    uselib_store='BOOST_DATETIME')
 
