--- configure.ac.orig	2016-01-14 10:13:12.108332384 +0000
+++ configure.ac	2016-01-14 10:16:19.778332384 +0000
@@ -1,5 +1,5 @@
 AC_PREREQ(2.59)
-AC_INIT([EXIF library], [0.6.21], [libexif-devel@lists.sourceforge.net], [libexif])
+AC_INIT([EXIF library], [0.6.21.1], [libexif-devel@lists.sourceforge.net], [libexif])
 AC_CONFIG_SRCDIR([libexif/exif-data.h])
 AC_CONFIG_HEADERS([config.h])
 AC_CONFIG_MACRO_DIR([auto-m4])
@@ -77,7 +77,7 @@
 AM_PROG_LIBTOOL
 AM_CPPFLAGS="$CPPFLAGS"
 GP_CONFIG_MSG([Compiler],[${CC}])
-
+AC_SYS_LARGEFILE
 
 dnl Create a stdint.h-like file containing size-specific integer definitions
 dnl that will always be available
