Index: libgcrypt-1.2.4/configure.ac
===================================================================
--- configure.ac	2008-03-19 22:14:50.000000000 +0000
+++ configure.ac	2008-03-19 22:14:58.000000000 +0000
@@ -807,6 +807,7 @@
 doc/Makefile
 src/Makefile
 src/gcrypt.h
+src/libgcrypt.pc
 src/libgcrypt-config
 tests/Makefile
 w32-dll/Makefile
Index: libgcrypt-1.2.4/src/libgcrypt.pc.in
===================================================================
