--- src/CMakeLists.txt.orig	2018-12-26 13:04:32.828042935 +0000
+++ src/CMakeLists.txt	2018-12-26 13:08:06.890204851 +0000
@@ -60,6 +60,7 @@
 add_library(mysofa-static STATIC ${libsrc})
 target_link_libraries (mysofa-static ${MATH} ${ZLIB_LIBRARIES})
 SET_TARGET_PROPERTIES(mysofa-static PROPERTIES OUTPUT_NAME mysofa CLEAN_DIRECT_OUTPUT 1 POSITION_INDEPENDENT_CODE ON)
+target_compile_definitions(mysofa-static PUBLIC MYSOFA_STATIC_DEFINE)
 install(TARGETS mysofa-static
   ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 
@@ -73,6 +74,7 @@
   GENERATE_EXPORT_HEADER(mysofa-shared BASE_NAME mysofa EXPORT_FILE_NAME ${PROJECT_SOURCE_DIR}/src/hrtf/mysofa_export.h)
   install(TARGETS mysofa-shared 
     RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
     LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
 else()
   GENERATE_EXPORT_HEADER(mysofa-static BASE_NAME mysofa EXPORT_FILE_NAME ${PROJECT_SOURCE_DIR}/src/hrtf/mysofa_export.h)
@@ -91,6 +93,7 @@
 
   add_executable(internal tests/internal.c tests/tools.c hrtf/tools.c)
   target_link_libraries (internal m ${CUNIT_LIBRARIES})
+  target_link_libraries (internal mysofa-static m ${CUNIT_LIBRARIES})
   add_test(NAME internal WORKING_DIRECTORY ${PROJECT_SOURCE_DIR} COMMAND internal)
   
 
