--- CMakeLists.txt.orig	2018-05-21 08:31:33.654878300 +0000
+++ CMakeLists.txt	2018-05-21 08:31:54.578325500 +0000
@@ -60,7 +60,7 @@
   matroska/c/libmatroska.h
   matroska/c/libmatroska_t.h)
 
-add_library(matroska ${libmatroska_SOURCES} ${limatroska_PUBLIC_HEADERS} ${libmatroska_C_PUBLIC_HEADERS})
+add_library(matroska ${libmatroska_SOURCES} ${libmatroska_PUBLIC_HEADERS} ${libmatroska_C_PUBLIC_HEADERS})
 target_link_libraries(matroska PUBLIC ebml)
 set_target_properties(matroska PROPERTIES
   VERSION 6.0.0
