--- src/meson.build.orig	2023-06-04 17:39:23.630119460 +0100
+++ src/meson.build	2023-06-04 17:40:18.180119447 +0100
@@ -20,7 +20,7 @@
   # MinGW-w64 explicitly excludes some math functions from their ucrtbase def
   # file and replaces with own versions. To workaround the issue, generate the
   # import library and link it with UCRT versions of math functions.
-  dlltool = find_program('llvm-dlltool', 'dlltool')
+  dlltool = find_program('llvm-dlltool', 'x86_64-w64-mingw32-dlltool') 
   ucrt_math = custom_target('ucrt_math.lib',
                             output : ['ucrt_math.lib'],
                             input : 'ucrt_math.def',
