--- src/sub_time.cc.orig	2017-11-04 13:19:19.286843200 +0000
+++ src/sub_time.cc	2017-11-04 13:19:38.537440300 +0000
@@ -21,6 +21,7 @@
 #include "exceptions.h"
 #include <cmath>
 #include <iomanip>
+#include <iostream>
 
 using std::ostream;
 using std::cout;
--- src/smpte_dcp_reader.cc.orig	2017-11-04 13:20:17.062037800 +0000
+++ src/smpte_dcp_reader.cc	2017-11-04 13:20:41.402993000 +0000
@@ -26,6 +26,7 @@
 #include "dcp/smpte_load_font.h"
 #include <libcxml/cxml.h>
 #include <boost/foreach.hpp>
+#include <iostream>
 
 using std::string;
 using std::list;
--- asdcplib/src/h__Writer.cpp.orig	2017-11-04 14:33:59.940734300 +0000
+++ asdcplib/src/h__Writer.cpp	2017-11-04 14:34:21.176455700 +0000
@@ -31,6 +31,7 @@
 
 #include "AS_DCP_internal.h"
 #include "KLV.h"
+#include <iostream>
 
 using std::cout;
 using namespace ASDCP;
