--- test/wscript.orig	2019-02-11 14:12:37.094829581 +0000
+++ test/wscript	2019-02-11 14:12:50.286733512 +0000
@@ -1,7 +1,7 @@
 def configure(conf):
     boost_lib_suffix = ''
     if conf.env.TARGET_WINDOWS:
-        boost_lib_suffix = '-mt'
+        boost_lib_suffix = '-mt-x64'
 
     conf.check_cxx(fragment="""
                             #define BOOST_TEST_MODULE Config test\n
--- wscript.orig	2020-11-22 14:37:02.060000000 +0000
+++ wscript	2020-11-22 14:37:56.960000000 +0000
@@ -145,7 +145,7 @@
     			    int main() { boost::filesystem::copy_file ("a", "b"); }\n
 			    """,
                    msg='Checking for boost filesystem library',
-                   libpath='/usr/local/lib',
+                   libpath='../../x86_64-mingw32-x64/x86_64-mingw32-x64/lib',
                    lib=['boost_filesystem%s' % boost_lib_suffix, 'boost_system%s' % boost_lib_suffix],
                    uselib_store='BOOST_FILESYSTEM')
 
@@ -164,7 +164,7 @@
     			    int main() { boost::locale::conv::to_utf<char> ("a", "cp850"); }\n
 			    """,
                    msg='Checking for boost locale library',
-                   libpath='/usr/local/lib',
+                   libpath='../../x86_64-mingw32-x64/x86_64-mingw32-x64/lib',
                    lib=locale_libs,
                    uselib_store='BOOST_LOCALE')
 
@@ -173,7 +173,7 @@
     			    int main() { boost::regex re ("foo"); }\n
 			    """,
                    msg='Checking for boost regex library',
-                   libpath='/usr/local/lib',
+                   libpath='../../x86_64-mingw32-x64/x86_64-mingw32-x64/lib',
                    lib=['boost_regex%s' % boost_lib_suffix, 'boost_system%s' % boost_lib_suffix],
                    uselib_store='BOOST_REGEX')
 
 
