--- src/sub_time.cc.orig	2016-05-04 13:23:43.829000000 +0000
+++ src/sub_time.cc	2016-05-04 13:24:06.186000000 +0000
@@ -20,6 +20,7 @@
 #include "sub_time.h"
 #include "exceptions.h"
 #include <cmath>
+#include <iostream>
 #include <iomanip>
 
 using std::ostream;
--- src/stl_text_reader.cc.orig	2016-05-04 13:16:05.401000000 +0000
+++ src/stl_text_reader.cc	2016-05-04 13:16:30.007000000 +0000
@@ -21,6 +21,7 @@
 #include "compose.hpp"
 #include <boost/algorithm/string.hpp>
 #include <boost/lexical_cast.hpp>
+#include <iostream>
 #include <vector>
 
 using std::list;
--- src/dcp_reader.cc.orig	2016-05-04 12:08:22.973000000 +0000
+++ src/dcp_reader.cc	2016-05-04 12:09:00.253000000 +0000
@@ -26,6 +26,7 @@
 #include "dcp/subtitle.h"
 #include <libcxml/cxml.h>
 #include <libxml++/libxml++.h>
+#include <iostream>
 
 using std::list;
 using std::cout;
--- tools/dumpsubs.cc.orig	2016-05-04 13:54:31.712000000 +0000
+++ tools/dumpsubs.cc	2016-05-04 13:54:53.627000000 +0000
@@ -22,6 +22,7 @@
 #include "collect.h"
 #include <getopt.h>
 #include <boost/filesystem.hpp>
+#include <iostream>
 #include <map>
 
 using std::string;
--- asdcplib/src/h__Writer.cpp.orig	2016-05-04 13:59:39.346000000 +0000
+++ asdcplib/src/h__Writer.cpp	2016-05-04 14:00:05.209000000 +0000
@@ -31,6 +31,7 @@
 
 #include "AS_DCP_internal.h"
 #include "KLV.h"
+#include <iostream>
 
 using std::cout;
 using namespace ASDCP;
