--- CMakeLists.txt.orig	2020-02-12 22:20:23.157151318 +0000
+++ CMakeLists.txt	2020-02-12 22:22:19.103962385 +0000
@@ -88,13 +88,13 @@
 if(BUILD_SHARED_LIBS)
     add_library(spatialaudio-shared SHARED ${sources})
     if(MYSOFA_FOUND)
-        target_link_libraries(spatialaudio-shared ${MYSOFA_LIBRARIES})
+        target_link_libraries(spatialaudio-shared ${MYSOFA_LIBRARIES} z)
     endif(MYSOFA_FOUND)
     SET_TARGET_PROPERTIES(spatialaudio-shared PROPERTIES OUTPUT_NAME spatialaudio CLEAN_DIRECT_OUTPUT 1)
     set_property(TARGET spatialaudio-shared PROPERTY VERSION "${PACKAGE_VERSION_MAJOR}.${PACKAGE_VERSION_MINOR}.${PACKAGE_VERSION_PATCH}")
     set_property(TARGET spatialaudio-shared PROPERTY SOVERSION ${PACKAGE_VERSION_MAJOR} )
     set_property(TARGET spatialaudio-shared PROPERTY C_VISIBILITY_PRESET hidden)
-    install(TARGETS spatialaudio-shared LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
+    install(TARGETS spatialaudio-shared ARCHIVE DESTINATION lib LIBRARY DESTINATION bin)
 endif(BUILD_SHARED_LIBS)
 
 option(HAVE_MIT_HRTF "Should MIT HRTF be built-in" ON)
