--- libxml++/io/istreamparserinputbuffer.cc.orig	2023-04-10 21:27:57.747297424 +0100
+++ libxml++/io/istreamparserinputbuffer.cc	2023-04-10 21:28:37.297297289 +0100
@@ -39,6 +39,6 @@
 
   bool IStreamParserInputBuffer::do_close()
   {
-    return input_;
+    return (bool)input_;
   }
 };
--- libxml++/io/ostreamoutputbuffer.cc.orig	2023-04-10 21:29:43.987297372 +0100
+++ libxml++/io/ostreamoutputbuffer.cc	2023-04-10 21:30:06.827297364 +0100
@@ -29,13 +29,13 @@
     // here we rely on the ostream implicit conversion to boolean, to know if the stream can be used and/or if the write succeded.
     if(output_)
       output_.write(buffer, len);
-    return output_;
+    return (bool)output_;
   }
 
   bool OStreamOutputBuffer::do_close()
   {
     if(output_)
         output_.flush();
-    return output_;
+    return (bool)output_;
   }
 }
