--- src/ipc_address.hpp.orig	2020-05-16 18:42:53.266075179 +0100
+++ src/ipc_address.hpp	2020-05-16 18:45:44.212536527 +0100
@@ -34,8 +34,13 @@
 
 #include <string>
 
-#if defined _MSC_VER
-#include <afunix.h>
+#if defined(_MSC_VER) || defined(WIN32)
+#ifndef UNIX_PATH_MAX
+#define UNIX_PATH_MAX 108
+	struct sockaddr_un {
+		ADDRESS_FAMILY sun_family;
+		char sun_path[UNIX_PATH_MAX];
+	};
 #else
 #include <sys/socket.h>
 #include <sys/un.h>
--- src/ipc_address.hpp.orig	2020-05-16 19:01:00.289158000 +0100
+++ src/ipc_address.hpp	2020-05-16 19:02:17.992523793 +0100
@@ -45,7 +45,7 @@
 #include <sys/socket.h>
 #include <sys/un.h>
 #endif
-
+#endif
 #include "macros.hpp"
 
 namespace zmq
--- src/ipc_connecter.cpp.orig	2020-05-16 19:09:24.620944799 +0100
+++ src/ipc_connecter.cpp	2020-05-16 19:11:48.411714175 +0100
@@ -43,8 +43,13 @@
 #include "ipc_address.hpp"
 #include "session_base.hpp"
 
-#ifdef _MSC_VER
-#include <afunix.h>
+#if defined(_MSC_VER) || defined(WIN32)
+#ifndef UNIX_PATH_MAX
+#define UNIX_PATH_MAX 108
+	struct sockaddr_un {
+		ADDRESS_FAMILY sun_family;
+		char sun_path[UNIX_PATH_MAX];
+	};
 #else
 #include <unistd.h>
 #include <sys/types.h>
--- src/ipc_listener.cpp.orig	2020-05-16 19:12:45.495223424 +0100
+++ src/ipc_listener.cpp	2020-05-16 19:14:59.230069750 +0100
@@ -44,12 +44,17 @@
 #include "socket_base.hpp"
 #include "address.hpp"
 
-#ifdef _MSC_VER
+#if defined(_MSC_VER) || defined(WIN32)
 #ifdef ZMQ_IOTHREAD_POLLER_USE_SELECT
 #error On Windows, IPC does not work with POLLER=select, use POLLER=epoll instead, or disable IPC transport
 #endif
 
-#include <afunix.h>
+#ifndef UNIX_PATH_MAX
+#define UNIX_PATH_MAX 108
+	struct sockaddr_un {
+		ADDRESS_FAMILY sun_family;
+		char sun_path[UNIX_PATH_MAX];
+	};
 #include <direct.h>
 
 #define rmdir _rmdir
--- src/ipc_address.hpp.orig	2020-05-16 19:23:59.013377083 +0100
+++ src/ipc_address.hpp	2020-05-16 19:24:23.189166051 +0100
@@ -37,6 +37,7 @@
 #if defined(_MSC_VER) || defined(WIN32)
 #ifndef UNIX_PATH_MAX
 #define UNIX_PATH_MAX 108
+#endif
 	struct sockaddr_un {
 		ADDRESS_FAMILY sun_family;
 		char sun_path[UNIX_PATH_MAX];
@@ -45,7 +46,6 @@
 #include <sys/socket.h>
 #include <sys/un.h>
 #endif
-#endif
 #include "macros.hpp"
 
 namespace zmq
--- src/ipc_connecter.cpp.orig	2020-05-16 19:25:02.324824332 +0100
+++ src/ipc_connecter.cpp	2020-05-16 19:25:46.660101581 +0100
@@ -46,6 +46,7 @@
 #if defined(_MSC_VER) || defined(WIN32)
 #ifndef UNIX_PATH_MAX
 #define UNIX_PATH_MAX 108
+#endif
 	struct sockaddr_un {
 		ADDRESS_FAMILY sun_family;
 		char sun_path[UNIX_PATH_MAX];
