--- src/lilv_internal.h.orig	2019-03-27 21:56:51.393173502 +0000
+++ src/lilv_internal.h	2019-03-27 21:57:20.877293122 +0000
@@ -37,8 +37,8 @@
 #    include <windows.h>
 #    include <direct.h>
 #    include <stdio.h>
-#    define dlopen(path, flags) LoadLibrary(path)
-#    define dlclose(lib)        FreeLibrary((HMODULE)lib)
+#    define dlopen(path, flags) LoadLibrary(path)
+#    define dlclose(lib)        FreeLibrary((HMODULE)lib)
 #    define unlink(path)        _unlink(path)
 #    define rmdir(path)         _rmdir(path)
 #    ifdef _MSC_VER
--- src/lilv_internal.h.orig	2019-03-27 22:02:44.485132446 +0000
+++ src/lilv_internal.h	2019-03-27 22:03:49.732850428 +0000
@@ -53,7 +53,7 @@
 #ifndef NAN
 #    define NAN INFINITY - INFINITY
 #endif
-static inline const char* dlerror(void) { return "Unknown error"; }
+static inline const char* dlerror(void) { return "Unknown error"; }
 #else
 #    include <dlfcn.h>
 #    include <unistd.h>
--- wscript.orig	2019-03-27 22:14:57.575457813 +0000
+++ wscript	2019-03-27 22:15:27.763148267 +0000
@@ -226,7 +226,7 @@
     if bld.is_defined('HAVE_LIBDL'):
         lib    += ['dl']
     if bld.env.DEST_OS == 'win32':
-        lib = []
+        lib    += ['dl']
     if bld.env.MSVC_COMPILER:
         libflags = []
 
