--- scanner/inputaudio/ffmpeg/input_ffmpeg.c.orig	2017-11-11 14:28:04.138037600 +0000
+++ scanner/inputaudio/ffmpeg/input_ffmpeg.c	2017-11-11 14:28:23.249889200 +0000
@@ -13,7 +13,7 @@
 #include "ebur128.h"
 #include "input.h"
 
-#define BUFFER_SIZE (192000 + FF_INPUT_BUFFER_PADDING_SIZE)
+#define BUFFER_SIZE (192000 + AV_INPUT_BUFFER_PADDING_SIZE)
 
 static GStaticMutex ffmpeg_mutex = G_STATIC_MUTEX_INIT;
 
