--- wscript.orig	2019-03-27 21:22:44.229208706 +0000
+++ wscript	2019-03-27 21:23:06.500315457 +0000
@@ -219,7 +219,7 @@
         test_cflags    = ['']
         test_linkflags = ['']
         if bld.is_defined('HAVE_GCOV'):
-            test_lib       += ['gcov', 'rt']
+            test_lib       += ['gcov']
             test_cflags    += ['--coverage']
             test_linkflags += ['--coverage']
 
