--- include/propvarutil.h.orig	2021-07-20 20:26:10.304071900 +0100
+++ include/propvarutil.h	2021-07-20 20:27:29.934071900 +0100
@@ -178,7 +178,11 @@
 
 inline HRESULT InitPropVariantFromGUIDAsBuffer(REFGUID guid, PROPVARIANT *ppropvar)
 {
+#ifdef __cplusplus
+	return InitPropVariantFromBuffer(&guid, sizeof(GUID), ppropvar);
+#else
     return InitPropVariantFromBuffer(guid, sizeof(GUID), ppropvar);
+#endif
 }
 
 inline WINBOOL IsPropVariantVector(REFPROPVARIANT propvar)
