--- ac/compiler_flags.m4.orig	2018-10-28 08:59:24.160965800 +0000
+++ ac/compiler_flags.m4	2018-10-28 09:00:03.683617900 +0000
@@ -19,7 +19,7 @@
   AC_TRY_COMPILE([],[],[ WNO_POTENTIALLY_EVALUATED_EXPRESSION="-Wno-potentially-evaluated-expression" ],[])
 
   if check_version 3.5.0 $COMPILER_VERSION ; then
-    FSTACK_PROTECTOR="-fstack-protector-strong"
+    FSTACK_PROTECTOR=""
   fi
 
   AC_LANG_POP()
@@ -29,14 +29,14 @@
 
 else
   WNO_MAYBE_UNINITIALIZED="-Wno-maybe-uninitialized"
-  FSTACK_PROTECTOR="-fstack-protector"
+  FSTACK_PROTECTOR=""
 
   if check_version 7.0.0 $COMPILER_VERSION ; then
     WSHADOW_COMPATIBLE_LOCAL="-Wshadow-compatible-local"
   fi
 
   if check_version 4.9.0 $COMPILER_VERSION ; then
-    FSTACK_PROTECTOR="-fstack-protector-strong"
+    FSTACK_PROTECTOR=""
   fi
 
   if check_version 4.8.0 $COMPILER_VERSION ; then
