--- src/mkvtoolnix-gui/app.cpp.orig	2023-05-14 19:49:14.514512402 +0100
+++ src/mkvtoolnix-gui/app.cpp	2023-05-14 19:53:56.804513637 +0100
@@ -593,7 +593,7 @@
   static std::optional<bool> s_isWindows11OrLater;
 
   if (!s_isWindows11OrLater)
-    s_isWindows11OrLater = QOperatingSystemVersion::current() >= QOperatingSystemVersion::Windows11;
+    s_isWindows11OrLater = QOperatingSystemVersion::current() >= QOperatingSystemVersion::Windows10;
 
   return *s_isWindows11OrLater;
 }
