--- src/melt/Makefile.orig	2018-09-17 20:57:43.219898200 +0100
+++ src/melt/Makefile	2018-09-17 20:58:03.491333700 +0100
@@ -44,7 +44,7 @@
 
 install:	all
 	install -d "$(DESTDIR)$(bindir)"
-	install -c -m 755 $(meltname) "$(DESTDIR)$(bindir)"
+	install -c -m 755 $(meltname).exe "$(DESTDIR)$(bindir)"
 ifeq ($(extra_versioning), true)
 ifeq ($(melt_noversion), false)
 	ln -s $(meltname) "$(DESTDIR)$(bindir)/melt"
