diff -ur mlt.orig/src/melt/Makefile mlt/src/melt/Makefile
--- src/melt/Makefile	2016-01-10 20:14:05.036332384 +0000
+++ src/melt/Makefile	2016-01-10 20:03:21.524332384 +0000
@@ -11,8 +11,8 @@
 
 ifeq ($(targetos), MinGW)
 ifeq (, $(findstring MELT_NOSDL, $(CFLAGS)))
-CFLAGS += $(shell sdl-config --cflags)
-LDFLAGS += $(shell sdl-config --libs)
+CFLAGS += $(shell x86_64-w64-mingw32-sdl-config --cflags)
+LDFLAGS += $(shell x86_64-w64-mingw32-sdl-config --libs)
 endif
 bindir = $(prefix)
 endif
diff -ur mlt.orig/src/modules/sdl/configure mlt/src/modules/sdl/configure
--- src/modules/sdl/configure	2016-01-10 20:14:06.263332384 +0000
+++ src/modules/sdl/configure	2016-01-10 19:55:53.092332384 +0000
@@ -3,13 +3,13 @@
 if [ "$help" != "1" ]
 then
 
-	sdl-config --version > /dev/null 2>&1
+	x86_64-w64-mingw32-sdl-config --version > /dev/null 2>&1
 	disable_sdl=$?
 
 	if [ "$disable_sdl" = "0" ]
 	then
 		echo > config.mak
-		image=`sdl-config --prefix`/include/SDL/SDL_image.h
+		image=`x86_64-w64-mingw32-sdl-config --prefix`/include/SDL/SDL_image.h
 		if [ -f "$image" ]
 		then
 			echo "WITH_SDL_IMAGE=1" >> config.mak
diff -ur mlt.orig/src/modules/sdl/Makefile mlt/src/modules/sdl/Makefile
--- src/modules/sdl/Makefile	2016-01-10 20:14:05.947332384 +0000
+++ src/modules/sdl/Makefile	2016-01-10 20:07:48.421332384 +0000
@@ -21,9 +21,9 @@
 LDFLAGS += -lX11
 endif
 
-CFLAGS += $(shell sdl-config --cflags)
+CFLAGS += $(shell x86_64-w64-mingw32-sdl-config --cflags)
 
-LDFLAGS += $(shell sdl-config --libs)
+LDFLAGS += $(shell x86_64-w64-mingw32-sdl-config --libs)
 
 ifeq ($(WITH_SDL_IMAGE),1)
 OBJS += producer_sdl_image.o
