--- src/melt/Makefile.orig	2017-04-23 13:29:19.335842900 +0100
+++ src/melt/Makefile	2017-04-23 13:29:51.035300500 +0100
@@ -12,8 +12,8 @@
 ifeq ($(targetos), MinGW)
 ifeq (, $(findstring MELT_NOSDL, $(CFLAGS)))
 ifeq (, $(shell pkg-config --exists sdl && echo yes))
-CFLAGS += $(shell  sdl-config --cflags)
-LDFLAGS += $(shell sdl-config --libs)
+CFLAGS += $(shell  x86_64-w64-mingw32-sdl-config --cflags)
+LDFLAGS += $(shell x86_64-w64-mingw32-sdl-config --libs)
 else
 CFLAGS += $(shell  pkg-config --cflags sdl)
 LDFLAGS += $(shell pkg-config --libs sdl)
