--- src/modules/opencv/filter_opencv_tracker.cpp.orig	2021-05-01 12:28:46.883899600 +0100
+++ src/modules/opencv/filter_opencv_tracker.cpp	2021-05-01 12:29:00.573899600 +0100
@@ -18,7 +18,7 @@
  */
 
 #include <framework/mlt.h>
-#include <opencv2/tracking.hpp>
+#include <opencv2/video/tracking.hpp>
 #include <opencv2/core/version.hpp>
 
 #define CV_VERSION_INT (CV_VERSION_MAJOR << 16 | CV_VERSION_MINOR << 8 | CV_VERSION_REVISION)
