--- CMakeLists.txt.orig	2023-04-01 22:03:07.875634914 +0100
+++ CMakeLists.txt	2023-04-01 22:04:02.585634871 +0100
@@ -146,7 +146,7 @@
 
 if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
   # Treat warnings as errors with some exceptions
-  set(GCC_FLAGS "-Wall -Werror -Wno-deprecated-declarations")
+  set(GCC_FLAGS "-Wall -Werror -Wno-deprecated-declarations -Wno-format-overflow")
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${GCC_FLAGS} -Wno-class-memaccess -Wno-array-compare -Wno-unused-result -Wno-maybe-uninitialized")
   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${GCC_FLAGS} -Wno-discarded-qualifiers")
 endif ()
