--- video/out/gpu/d3d11_helpers.h.orig	2025-02-26 13:50:13.903851983 +0000
+++ video/out/gpu/d3d11_helpers.h	2025-02-26 13:51:08.663413583 +0000
@@ -27,9 +27,10 @@
 
 #include "video/mp_image.h"
 
-#if !HAVE_DXGI_DEBUG_D3D11
-DEFINE_GUID(DXGI_DEBUG_D3D11, 0x4b99317b, 0xac39, 0x4aa6, 0xbb, 0xb, 0xba, 0xa0, 0x47, 0x84, 0x79, 0x8f);
-#endif
+/* #if !HAVE_DXGI_DEBUG_D3D11
+ * DEFINE_GUID(DXGI_DEBUG_D3D11, 0x4b99317b, 0xac39, 0x4aa6, 0xbb, 0xb, 0xba, 0xa0, 0x47, 0x84, 0x79, 0x8f);
+ * #endif
+*/
 
 struct d3d11_device_opts {
     // Enable the debug layer (D3D11_CREATE_DEVICE_DEBUG)
