--- libdispatch/dwinpath.c.orig	2018-03-27 20:16:58.355878800 +0000
+++ libdispatch/dwinpath.c	2018-03-27 20:17:57.662931700 +0000
@@ -155,7 +155,7 @@
 makeabsolute(const char* relpath)
 {
     char* path = NULL;
-#ifdef _MSC_VER
+#if defined (_MSC_VER) || defined(__MINGW32__)
     path = _fullpath(NULL,relpath,8192);
 #else
     path = realpath(relpath, NULL);
--- libsrc/memio.c.orig	2018-05-08 22:10:36.407774800 +0000
+++ libsrc/memio.c	2018-05-08 22:10:46.468839300 +0000
@@ -17,7 +17,7 @@
 #ifdef HAVE_FCNTL_H
 #include <fcntl.h>
 #endif
-#ifdef _MSC_VER /* Microsoft Compilers */
+#if defined (_MSC_VER) || defined(__MINGW32__) /* Microsoft Compilers */
 #include <windows.h>
 #include <winbase.h>
 #include <io.h>
--- libdispatch/dwinpath.c.orig	2018-09-14 22:59:30.765549400 +0100
+++ libdispatch/dwinpath.c	2018-09-14 22:59:59.180354900 +0100
@@ -8,6 +8,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <assert.h>
+#include <errno.h>
 #ifdef HAVE_FCNTL_H
 #include <fcntl.h>
 #endif
--- libdap2/dapcvt.c.orig	2018-10-05 11:49:19.931093900 +0100
+++ libdap2/dapcvt.c	2018-10-05 11:51:30.289632900 +0100
@@ -5,7 +5,7 @@
 #include "config.h"
 #include "dapincludes.h"
 
-#ifdef _MSC_VER
+#if defined (_MSC_VER) || defined(__MINGW32__)
 #include <crtdbg.h>
 #include <math.h>
 #endif
