--- ncdump/ocprint.c.orig	2020-05-22 13:56:21.003456724 +0100
+++ ncdump/ocprint.c	2020-05-22 13:57:15.113203261 +0100
@@ -31,7 +31,7 @@
 #include "ocx.h"
 
 #ifdef _MSC_VER
-#include "XGetopt.h"
+/* #include "XGetopt.h" */
 #endif
 
 #ifndef nulldup
--- ncdump/ncdump.c.orig	2020-05-22 13:58:04.471286796 +0100
+++ ncdump/ncdump.c	2020-05-22 13:58:37.558070058 +0100
@@ -9,7 +9,7 @@
 #include <getopt.h>
 #endif
 #ifdef _MSC_VER
-#include "XGetopt.h"
+/* #include "XGetopt.h" */
 #define snprintf _snprintf
 #endif
 
--- ncdump/ncvalidator.c.orig	2020-06-03 11:55:14.462310636 +0100
+++ ncdump/ncvalidator.c	2020-06-03 11:55:33.581997658 +0100
@@ -73,8 +73,8 @@
 #include <unistd.h>     /* read() getopt() */
 #endif
 
-#ifdef _MSC_VER
-#include "XGetopt.h"
+#ifdef _WIN32
+/* #include "XGetopt.h" */
 #include <io.h>
 #define snprintf _snprintf
 #endif
