--- ncdump/ocprint.c.orig	2019-04-09 20:33:50.337293930 +0100
+++ ncdump/ocprint.c	2019-04-09 20:34:13.305100468 +0100
@@ -31,9 +31,9 @@
 #include "ocx.h"
 
 #ifdef _MSC_VER
-#include "XGetopt.h"
+/* #include "XGetopt.h"
 int opterr, optind;
-char* optarg;
+char* optarg; */
 #endif
 
 #ifndef nulldup
--- ncdump/ncdump.c.orig	2019-06-09 15:30:16.021385981 +0100
+++ ncdump/ncdump.c	2019-06-09 15:30:28.029156268 +0100
@@ -20,7 +20,7 @@
 
 #ifdef _MSC_VER
 #define snprintf _snprintf
-#include "XGetopt.h"
+/*#include "XGetopt.h"*/
 int opterr;
 int optind;
 #endif
--- ncdump/ncvalidator.c.orig	2019-06-09 15:35:07.303838094 +0100
+++ ncdump/ncvalidator.c	2019-06-09 15:35:22.079557792 +0100
@@ -76,7 +76,7 @@
 #ifdef _MSC_VER
 #include <io.h>
 #define snprintf _snprintf
-#include "XGetopt.h"
+/*#include "XGetopt.h"*/
 int opterr;
 int optind;
 #endif
