--- libdispatch/dwinpath.c.orig	2018-03-27 20:16:58.355878800 +0000
+++ libdispatch/dwinpath.c	2018-03-27 20:17:57.662931700 +0000
@@ -155,7 +155,7 @@
 makeabsolute(const char* relpath)
 {
     char* path = NULL;
-#ifdef _MSC_VER
+#if defined (_MSC_VER) || defined(__MINGW32__)
     path = _fullpath(NULL,relpath,8192);
 #else
     path = realpath(relpath, NULL);
--- libsrc/memio.c.orig	2018-05-08 22:10:36.407774800 +0000
+++ libsrc/memio.c	2018-05-08 22:10:46.468839300 +0000
@@ -17,7 +17,7 @@
 #ifdef HAVE_FCNTL_H
 #include <fcntl.h>
 #endif
-#ifdef _MSC_VER /* Microsoft Compilers */
+#if defined (_MSC_VER) || defined(__MINGW32__) /* Microsoft Compilers */
 #include <windows.h>
 #include <winbase.h>
 #include <io.h>
