--- libsrc/memio.c.orig	2019-01-31 21:46:06.756117420 +0000
+++ libsrc/memio.c	2019-01-31 21:46:54.803597550 +0000
@@ -17,7 +17,7 @@
 #ifdef HAVE_FCNTL_H
 #include <fcntl.h>
 #endif
-#ifdef _MSC_VER /* Microsoft Compilers */
+#if defined (_MSC_VER) || defined(__MINGW32__) /* Microsoft Compilers */
 #include <windows.h>
 #include <winbase.h>
 #include <io.h>
--- libdap2/dapcvt.c.orig	2019-01-31 21:52:47.567786312 +0000
+++ libdap2/dapcvt.c	2019-01-31 21:53:06.943577196 +0000
@@ -5,7 +5,7 @@
 #include "config.h"
 #include "dapincludes.h"
 
-#ifdef _MSC_VER
+#if defined (_MSC_VER) || defined(__MINGW32__)
 #include <crtdbg.h>
 #include <math.h>
 #endif
--- ncdump/ncdump.c.orig	2019-01-31 21:58:31.880072424 +0000
+++ ncdump/ncdump.c	2019-01-31 21:58:50.399872766 +0000
@@ -8,7 +8,7 @@
 #ifdef HAVE_GETOPT_H
 #include <getopt.h>
 #endif
-#ifdef _MSC_VER /* Microsoft Compilers */
+#if defined (_MSC_VER) || defined(__MINGW32__) /* Microsoft Compilers */
 #include <io.h>
 #endif
 #ifdef HAVE_UNISTD_H
--- ncdump/ncdump.c.orig	2019-01-31 22:04:24.496567033 +0000
+++ ncdump/ncdump.c	2019-01-31 22:04:55.768259369 +0000
@@ -18,11 +18,13 @@
 #include <fcntl.h>
 #endif
 
+#if 0
 #ifdef _WIN32
 #include "XGetopt.h"
 int opterr;
 int optind;
 #endif
+#endif
 
 #include <stdlib.h>
 #include <string.h>
--- ncdump/ocprint.c.orig	2019-01-31 22:09:45.949351115 +0000
+++ ncdump/ocprint.c	2019-01-31 22:10:07.909127898 +0000
@@ -30,11 +30,13 @@
 #include "oc.h"
 #include "ocx.h"
 
+#if 0
 #ifdef _WIN32
 #include "XGetopt.h"
 int opterr, optind;
 char* optarg;
 #endif
+#endif
 
 #ifndef nulldup
 #define nulldup(s) (s==NULL?NULL:strdup(s))
