--- Makefile.rule.orig	2016-12-31 18:56:37.740586100 +0000
+++ Makefile.rule	2016-12-31 18:58:34.314573600 +0000
@@ -27,8 +27,8 @@
 # Even you can specify cross compiler. Meanwhile, please set HOSTCC.
 
 # cross compiler for Windows
-# CC = x86_64-w64-mingw32-gcc
-# FC = x86_64-w64-mingw32-gfortran
+CC = x86_64-w64-mingw32-gcc
+FC = x86_64-w64-mingw32-gfortran
 
 # cross compiler for 32bit ARM
 # CC = arm-linux-gnueabihf-gcc
@@ -40,10 +40,10 @@
 
 
 # If you use the cross compiler, please set this host compiler.
-# HOSTCC = gcc
+HOSTCC = gcc
 
 # If you need 32bit binary, define BINARY=32, otherwise define BINARY=64
-# BINARY=64
+BINARY=64
 
 # About threaded BLAS. It will be automatically detected if you don't
 # specify it.
@@ -61,7 +61,7 @@
 # NUM_THREADS = 24
 
 # if you don't need to install the static library, please comment it in.
-# NO_STATIC = 1
+NO_STATIC = 1
 
 # if you don't need generate the shared library, please comment it in.
 # NO_SHARED = 1
