--- modules/objdetect/src/detection_based_tracker.cpp.orig	2016-05-10 13:51:36.777000000 +0000
+++ modules/objdetect/src/detection_based_tracker.cpp	2016-05-10 14:50:53.310000000 +0000
@@ -52,9 +52,16 @@
 #include "opencv2/core/utility.hpp"
 
 #ifdef USE_STD_THREADS
-#include <thread>
-#include <mutex>
-#include <condition_variable>
+#include <boost/thread.hpp>
+
+namespace std {
+  using boost::mutex;
+  using boost::recursive_mutex;
+  using boost::lock_guard;
+  using boost::condition_variable;
+  using boost::unique_lock;
+  using boost::thread;
+}
 #else
 #include <pthread.h>
 #endif
