--- toolchains/win32.cmake.orig	2015-03-12 23:48:09.108988100 +0000
+++ toolchains/win32.cmake	2015-03-12 23:49:09.286328300 +0000
@@ -11,9 +11,9 @@
 SET(LIB_DIR_SYS ${SYSROOT_PATH}/lib)
 
 IF(TARGET_ARCH STREQUAL "i686")
-    SET(CMAKE_RC_COMPILER /usr/bin/i686-w64-mingw32-windres) 
+    SET(CMAKE_RC_COMPILER ${SYSROOT_PATH}/../bin/i686-w64-mingw32-windres) 
 ELSE()
-    SET(CMAKE_RC_COMPILER /usr/bin/x86_64-w64-mingw32-windres) 
+    SET(CMAKE_RC_COMPILER ${SYSROOT_PATH}/../bin/x86_64-w64-mingw32-windres) 
 ENDIF()
 
 INCLUDE_DIRECTORIES(${SYSROOT_PATH}/include/libxml2)
