--- toolchains/win32.cmake.orig	2015-03-12 22:57:05.432293800 +0000
+++ toolchains/win32.cmake	2015-03-12 22:59:04.492202900 +0000
@@ -39,6 +39,7 @@
 SET(LIBS ${LIBS} ${LIB_DIR_SYS}/libtiff.a ${LIB_DIR_SYS}/libjpeg.a)
 SET(LIBS ${LIBS} -L${LIB_DIR_SYS} -lcrypto -lssl)
 SET(LIBS ${LIBS} -L${LIB_DIR_SYS} -lz -lws2_32)
+SET(LIBS ${LIBS} -L${LIB_DIR_SYS} -lltdl -liconv -lcrypt32)
 #-------------------------------------------------------------------------------
 
 IF(TARGET_ARCH STREQUAL x86_64)
