--- src/opentime/CMakeLists.txt.orig	2020-02-04 23:46:33.336496967 +0000
+++ src/opentime/CMakeLists.txt	2020-02-04 23:47:23.127980433 +0000
@@ -14,7 +14,9 @@
 target_include_directories(opentime PUBLIC "${PROJECT_SOURCE_DIR}/src")
 set_target_properties(opentime PROPERTIES INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib"
 			MACOSX_RPATH true WINDOWS_EXPORT_ALL_SYMBOLS true)
-install(TARGETS opentime LIBRARY DESTINATION lib)
+install(TARGETS opentime LIBRARY DESTINATION lib
+	RUNTIME DESTINATION bin
+	ARCHIVE DESTINATION lib)
 if (NOT OTIO_CXX_NOINSTALL)
     install(FILES ${OPENTIME_HEADER_FILES} DESTINATION include/opentime)
 endif (NOT OTIO_CXX_NOINSTALL)
--- src/opentimelineio/CMakeLists.txt.orig	2020-02-04 23:51:09.605638257 +0000
+++ src/opentimelineio/CMakeLists.txt	2020-02-04 23:51:46.125261492 +0000
@@ -77,7 +77,9 @@
 target_link_libraries(opentimelineio PUBLIC opentime)
 set_target_properties(opentimelineio PROPERTIES INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib"
 			MACOSX_RPATH true WINDOWS_EXPORT_ALL_SYMBOLS true)
-install(TARGETS opentimelineio LIBRARY DESTINATION lib)
+install(TARGETS opentimelineio LIBRARY DESTINATION lib
+	ARCHIVE DESTINATION lib
+	RUNTIME DESTINATION bin)
 
 if (NOT OTIO_CXX_NOINSTALL)
     install(FILES ${OPENTIMELINEIO_HEADER_FILES} DESTINATION include/opentimelineio)
