--- modules/dnn/CMakeLists.txt.orig	2018-06-16 17:17:55.128800800 +0100
+++ modules/dnn/CMakeLists.txt	2018-06-16 17:18:39.314850000 +0100
@@ -31,6 +31,7 @@
                                        /wd4702 /wd4456 /wd4457 /wd4065 /wd4310 /wd4661 /wd4506
   )
 else()
+  add_definitions( -DNO_STRICT)
   ocv_warnings_disable(CMAKE_CXX_FLAGS -Wno-deprecated -Wmissing-prototypes -Wmissing-declarations -Wshadow
                                        -Wunused-parameter -Wunused-local-typedefs -Wsign-compare -Wsign-promo
                                        -Wundef -Wtautological-undefined-compare -Wignored-qualifiers -Wextra
