--- libopendcp/CMakeLists.txt.orig	2016-05-16 10:19:45.926000000 +0000
+++ libopendcp/CMakeLists.txt	2016-05-16 10:21:01.156000000 +0000
@@ -51,7 +51,7 @@
 
 IF(BUILD_SHARED)
     ADD_LIBRARY(opendcp-lib-shared SHARED ${OPENDCP_SRC_FILES})
-    SET_TARGET_PROPERTIES(opendcp-lib-shared PROPERTIES OUTPUT_NAME "opendcp")
+    SET_TARGET_PROPERTIES(opendcp-lib-shared PROPERTIES OUTPUT_NAME "opendcp-lib")
     SET_TARGET_PROPERTIES(opendcp-lib-shared PROPERTIES PREFIX "lib")
     TARGET_LINK_LIBRARIES(opendcp-lib-shared ${ASDCP_SHARED_LIBRARIES} ${LIBS} ${LIB_RAGNAROK})
     IF(INSTALL_LIB)
@@ -60,7 +60,7 @@
 ENDIF()
 IF(BUILD_STATIC)
     ADD_LIBRARY(opendcp-lib STATIC ${OPENDCP_SRC_FILES})
-    SET_TARGET_PROPERTIES(opendcp-lib PROPERTIES OUTPUT_NAME "opendcp")
+    SET_TARGET_PROPERTIES(opendcp-lib PROPERTIES OUTPUT_NAME "opendcp-lib")
     SET_TARGET_PROPERTIES(opendcp-lib PROPERTIES PREFIX "lib")
     TARGET_LINK_LIBRARIES(opendcp-lib ${ASDCP_LIBRARIES} ${LIBS} ${LIB_RAGNAROK})
     IF(INSTALL_LIB)
@@ -72,7 +72,7 @@
 #--install header---------------------------------------------------------------
 IF(INSTALL_LIB)
     INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/include/opendcp.h DESTINATION include)
-    INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/libopendcp/codecs/opendcp_encoders.h DESTINATION include)
-    INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/libopendcp/codecs/opendcp_decoders.h DESTINATION include)
+    INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/codecs/opendcp_encoder.h DESTINATION include)
+    INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/codecs/opendcp_decoder.h DESTINATION include)
 ENDIF()
 #-------------------------------------------------------------------------------
