--- CMakeLists.txt.orig	2017-03-29 11:48:08.404188200 +0100
+++ CMakeLists.txt	2017-03-29 11:48:52.313515900 +0100
@@ -290,7 +290,7 @@
 
 if(WIN32)
   if(DEFINED OpenCV_RUNTIME AND DEFINED OpenCV_ARCH)
-    ocv_update(OpenCV_INSTALL_BINARIES_PREFIX "${OpenCV_ARCH}/${OpenCV_RUNTIME}/")
+    ocv_update(OpenCV_INSTALL_BINARIES_PREFIX "")
   else()
     message(STATUS "Can't detect runtime and/or arch")
     set(OpenCV_INSTALL_BINARIES_PREFIX "")
--- CMakeLists.txt.orig	2017-03-29 12:56:25.862770600 +0100
+++ CMakeLists.txt	2017-03-29 12:57:25.036277400 +0100
@@ -360,7 +360,7 @@
 
 if(WIN32)
   # Postfix of DLLs:
-  set(OPENCV_DLLVERSION "${OPENCV_VERSION_MAJOR}${OPENCV_VERSION_MINOR}${OPENCV_VERSION_PATCH}")
+  set(OPENCV_DLLVERSION "")
   set(OPENCV_DEBUG_POSTFIX d)
 else()
   # Postfix of so's:
