--- CMakeLists.txt.orig	2020-01-04 15:51:47.745907445 +0000
+++ CMakeLists.txt	2020-01-04 15:52:52.557004211 +0000
@@ -22,9 +22,11 @@
     pkg_search_module(NCURSESW REQUIRED ncursesw)
     link_libraries(${NCURSESW_LDFLAGS})
     add_definitions(-DFEAT_UNICODE)
+    message("Adding ncursesw")
 ELSE ()
     pkg_search_module(NCURSES REQUIRED ncurses)
     link_libraries(${NCURSES_LDFLAGS})
+    message("NOT adding ncursesw")
 ENDIF ()
 
 add_executable(pamix ${pamix_SRC})
--- include/pamix.hpp.orig	2020-01-04 15:57:11.829679013 +0000
+++ include/pamix.hpp	2020-01-04 15:57:33.457418241 +0000
@@ -3,6 +3,7 @@
 #include <../config.hpp>
 #include <painterface.hpp>
 #include "pamix_ui.hpp"
+#include <ncursesw/curses.h>
 
 struct UpdateData {
 	bool redrawAll;
--- include/pamix.hpp.orig	2020-01-04 16:27:34.918469516 +0000
+++ include/pamix.hpp	2020-01-04 16:28:09.286103033 +0000
@@ -3,6 +3,7 @@
 #include <../config.hpp>
 #include <painterface.hpp>
 #include "pamix_ui.hpp"
+#define _XOPEN_SOURCE_EXTENDED 1 
 #include <ncursesw/curses.h>
 
 struct UpdateData {
--- include/pamix.hpp.orig	2020-01-04 16:38:09.455885265 +0000
+++ include/pamix.hpp	2020-01-04 16:38:45.687517062 +0000
@@ -3,7 +3,8 @@
 #include <../config.hpp>
 #include <painterface.hpp>
 #include "pamix_ui.hpp"
-#define _XOPEN_SOURCE_EXTENDED 1 
+#define _XOPEN_SOURCE_EXTENDED 1
+#define NCURSES_WIDECHAR 1
 #include <ncursesw/curses.h>
 
 struct UpdateData {
--- config.hpp.in.orig	2020-01-04 16:51:37.119756138 +0000
+++ config.hpp.in	2020-01-04 16:52:12.455402745 +0000
@@ -1,5 +1,7 @@
 #pragma once
 #ifdef FEAT_UNICODE
+#define _XOPEN_SOURCE 1000
+#define _XOPEN_SOURCE_EXTENDED 1
 #include <${NCURSESW_H_INCLUDE}>
 #else
 #include <ncurses.h>
--- src/pamix.cpp.orig	2020-01-04 17:07:36.633779525 +0000
+++ src/pamix.cpp	2020-01-04 17:08:04.381492329 +0000
@@ -215,8 +215,9 @@
 
 	signal(SIGABRT, sig_handle);
 	signal(SIGSEGV, sig_handle);
+#if defined SIGWINCH
 	signal(SIGWINCH, sig_handle_resize);
-
+#endif
 	PAInterface pai("pamix");
 	pamix_ui pamixUi(&pai);
 	if (configuration.has(CONFIGURATION_AUTOSPAWN_PULSE))
--- CMakeLists.txt.orig	2020-01-04 17:15:46.280756968 +0000
+++ CMakeLists.txt	2020-01-04 17:16:32.448286993 +0000
@@ -30,7 +30,7 @@
 ENDIF ()
 
 add_executable(pamix ${pamix_SRC})
-install(FILES pamix.conf DESTINATION /etc/xdg)
+install(FILES pamix.conf DESTINATION etc/xdg)
 install(TARGETS pamix DESTINATION bin)
 install(FILES man/pamix.1 DESTINATION share/man/man1)
 
