--- src/hostapi/wasapi/pa_win_wasapi.c.orig	2016-07-15 13:50:56.245000000 +0000
+++ src/hostapi/wasapi/pa_win_wasapi.c	2016-07-15 13:53:16.523000000 +0000
@@ -49,7 +49,7 @@
 #include <assert.h>
 #include <mmsystem.h>
 #include <mmreg.h>  // must be before other Wasapi headers
-#if defined(_MSC_VER) && (_MSC_VER >= 1400)
+#if defined((_MSC_VER) && (_MSC_VER >= 1400)) || defined (_WIN32)
 	#include <Avrt.h>
 	#define COBJMACROS
 	#include <Audioclient.h>
