--- CMakeLists.txt.orig	2021-01-20 21:13:13.110000000 +0000
+++ CMakeLists.txt	2021-01-20 21:16:33.150000000 +0000
@@ -237,6 +237,23 @@
 
   SET(PA_EXTRA_SHARED_SOURCES ${CMAKE_CURRENT_BINARY_DIR}/portaudio_cmake.def)
 
+  FIND_PACKAGE(Jack)
+  IF(JACK_FOUND)
+    OPTION(PA_USE_JACK "Enable support for Jack" ON)
+  ELSE()
+    OPTION(PA_USE_JACK "Enable support for Jack" OFF)
+  ENDIF()
+  IF(PA_USE_JACK)
+    SET(PA_PRIVATE_INCLUDE_PATHS ${PA_PRIVATE_INCLUDE_PATHS} ${JACK_INCLUDE_DIRS})
+    SET(PA_JACK_SOURCES src/hostapi/jack/pa_jack.c)
+    SOURCE_GROUP("hostapi\\JACK" FILES ${PA_JACK_SOURCES})
+    SET(PA_PUBLIC_INCLUDES ${PA_PUBLIC_INCLUDES} include/pa_jack.h)
+    SET(PA_SOURCES ${PA_SOURCES} ${PA_JACK_SOURCES})
+    SET(PA_PRIVATE_COMPILE_DEFINITIONS ${PA_PRIVATE_COMPILE_DEFINITIONS} PA_USE_JACK)
+    SET(PA_LIBRARY_DEPENDENCIES ${PA_LIBRARY_DEPENDENCIES} ${JACK_LIBRARIES} regex)
+    SET(PA_PKGCONFIG_LDFLAGS "${PA_PKGCONFIG_LDFLAGS} -ljack64")
+  ENDIF()
+
 ELSE()
 
   SET(PA_PRIVATE_INCLUDE_PATHS ${PA_PRIVATE_INCLUDE_PATHS} src/os/unix)
--- src/hostapi/jack/pa_jack.c.orig	2021-01-20 21:20:07.840000000 +0000
+++ src/hostapi/jack/pa_jack.c	2021-01-20 21:20:27.440000000 +0000
@@ -57,6 +57,7 @@
 #include <errno.h>  /* EBUSY */
 #include <signal.h> /* sig_atomic_t */
 #include <math.h>
+#include <pthread.h>
 #include <semaphore.h>
 
 #include <jack/types.h>
