--- configure.in.orig	2016-07-15 15:38:06.464000000 +0000
+++ configure.in	2016-07-15 15:40:30.885000000 +0000
@@ -10,7 +10,7 @@
 dnl Init autoconf and make sure configure is being called
 dnl from the right directory
 AC_INIT([include/portaudio.h])
-
+AM_INIT_AUTOMAKE($PACKAGE, $VERSION)
 dnl Define build, build_cpu, build_vendor, build_os
 AC_CANONICAL_BUILD
 dnl Define host, host_cpu, host_vendor, host_os
--- configure.in.orig	2016-07-15 16:15:18.063000000 +0000
+++ configure.in	2016-07-15 16:15:42.444000000 +0000
@@ -97,6 +97,7 @@
 dnl Checks for programs.
 
 AC_PROG_CC
+AC_PROG_CXX
 dnl ASIO and CXX bindings need a C++ compiler
 if [[ "$with_asio" = "yes" ] || [ "$enable_cxx" = "yes" ]] ; then
        AC_PROG_CXX
