--- configure.orig	2019-09-11 14:41:22.335648247 +0100
+++ configure	2019-09-11 14:41:35.879517512 +0100
@@ -9872,7 +9872,7 @@
     # Warnings to be aware of that appear with -D_FORTIFY_SOURCE=2 but without -U_FORTIFY_SOURCE:
     # On Fedora 20 with -O0: #warning _FORTIFY_SOURCE requires compiling with optimization (-O) [-Wcpp]
     # On Gentoo with -O2:    "_FORTIFY_SOURCE" redefined [enabled by default]
-    as_fn_append CPPFLAGS " -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2"
+    # as_fn_append CPPFLAGS " -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2"
  ;;
 esac
 
