--- src/qjackctlMainForm.cpp.orig	2016-07-14 20:33:45.799000000 +0000
+++ src/qjackctlMainForm.cpp	2016-07-14 20:39:25.788000000 +0000
@@ -54,8 +54,10 @@
 #include <QCloseEvent>
 
 #if defined(WIN32)
+#if 0
 #include <QPlastiqueStyle>
 #endif
+#endif
 
 
 #if QT_VERSION < 0x040500
@@ -317,7 +319,7 @@
 	: QWidget(pParent, wflags)
 {
 #if defined(WIN32)
-	QApplication::setStyle(new QPlastiqueStyle());
+//	QApplication::setStyle(new QPlastiqueStyle());
 #endif
 
 	// Setup UI struct...
