--- src/qjackctlMainForm.cpp.orig	2017-11-06 22:27:54.371148100 +0000
+++ src/qjackctlMainForm.cpp	2017-11-06 22:29:49.664060300 +0000
@@ -54,9 +54,10 @@
 #include <QCloseEvent>
 
 #if defined(__WIN32__) || defined(_WIN32) || defined(WIN32)
+#if 0
 #include <QPlastiqueStyle>
 #endif
-
+#endif
 
 #if QT_VERSION < 0x040500
 namespace Qt {
@@ -317,8 +318,10 @@
 	: QWidget(pParent, wflags)
 {
 #if defined(__WIN32__) || defined(_WIN32) || defined(WIN32)
+#if 0
 	QApplication::setStyle(new QPlastiqueStyle());
 #endif
+#endif
 
 	// Setup UI struct...
 	m_ui.setupUi(this);
