--- src/CMakeLists.txt.orig	2021-03-28 19:05:15.380000000 +0100
+++ src/CMakeLists.txt	2021-03-28 19:05:34.030000000 +0100
@@ -102,11 +102,7 @@
 	translations/qjackctl_uk.ts
 )
 
-if (QT_VERSION VERSION_LESS 5.15.0)
-  qt5_add_translation (QM_FILES ${TRANSLATIONS})
-else ()
-  qt_add_translation (QM_FILES ${TRANSLATIONS})
-endif ()
+qt_add_translation (QM_FILES ${TRANSLATIONS})
 
 add_custom_target (translations ALL DEPENDS ${QM_FILES})
 
--- src/CMakeLists.txt.orig	2021-03-28 19:28:18.180000000 +0100
+++ src/CMakeLists.txt	2021-03-28 19:29:04.480000000 +0100
@@ -102,6 +102,7 @@
 	translations/qjackctl_uk.ts
 )
 
+find_package (Qt5 5.15 COMPONENTS LinguistTools REQUIRED)
 qt_add_translation (QM_FILES ${TRANSLATIONS})
 
 add_custom_target (translations ALL DEPENDS ${QM_FILES})
--- src/CMakeLists.txt.orig	2021-03-28 19:42:50.900000000 +0100
+++ src/CMakeLists.txt	2021-03-28 19:43:29.040000000 +0100
@@ -101,7 +101,7 @@
 	translations/qjackctl_sk.ts
 	translations/qjackctl_uk.ts
 )
-
+find_package (Qt5 5.15 COMPONENTS Core REQUIRED)
 find_package (Qt5 5.15 COMPONENTS LinguistTools REQUIRED)
 qt_add_translation (QM_FILES ${TRANSLATIONS})
 
--- src/src.pro.orig	2021-03-28 20:07:45.290000000 +0100
+++ src/src.pro	2021-03-28 20:08:00.290000000 +0100
@@ -139,6 +139,3 @@
 # XML/DOM support
 QT += widgets xml
 
-win32 {
-	CONFIG += static
-}
