--- src/qjackctlMainForm.cpp.orig	2020-06-27 23:00:28.460000000 +0100
+++ src/qjackctlMainForm.cpp	2020-06-27 23:00:39.200000000 +0100
@@ -2655,9 +2655,6 @@
 		jackFinished();
 		break;
 	case QProcess::Crashed:
-	#if defined(__WIN32__) || defined(_WIN32) || defined(WIN32)
-		if (!m_bJackStopped)
-	#endif
 		appendMessagesColor(tr("JACK has crashed."), "#cc3366");
 		break;
 	case QProcess::Timedout:
