--- qtdeclarative/src/imports/imports.pro.orig	2020-12-31 20:01:14.340000000 +0000
+++ qtdeclarative/src/imports/imports.pro	2020-12-31 20:02:18.700000000 +0000
@@ -4,12 +4,11 @@
 SUBDIRS += \
     builtins \
     qtqml \
-    models \
-    labsmodels
+    models 
 
-qtConfig(qml-itemmodel): SUBDIRS += folderlistmodel
+qtConfig(qml-itemmodel): SUBDIRS += 
 qtConfig(qml-worker-script): SUBDIRS += workerscript
-qtHaveModule(sql): SUBDIRS += localstorage
+qtHaveModule(sql): SUBDIRS += 
 qtConfig(settings): SUBDIRS += settings
 qtConfig(statemachine): SUBDIRS += statemachine
 
--- qtdeclarative/src/imports/imports.pro.orig	2020-12-31 20:28:42.630000000 +0000
+++ qtdeclarative/src/imports/imports.pro	2020-12-31 20:30:28.170000000 +0000
@@ -9,15 +9,13 @@
 qtConfig(qml-itemmodel): SUBDIRS += 
 qtConfig(qml-worker-script): SUBDIRS += workerscript
 qtHaveModule(sql): SUBDIRS += 
-qtConfig(settings): SUBDIRS += settings
-qtConfig(statemachine): SUBDIRS += statemachine
+qtConfig(settings): SUBDIRS += 
+qtConfig(statemachine): SUBDIRS += 
 
 qtHaveModule(quick) {
     QT_FOR_CONFIG += quick-private
 
     SUBDIRS += \
-        labsanimation \
-        layouts \
         qtquick2 \
         window
 
