--- qtbase/mkspecs/win32-g++/qmake.conf.orig	2020-08-09 11:16:25.870000000 +0100
+++ qtbase/mkspecs/win32-g++/qmake.conf	2020-08-09 11:17:22.250000000 +0100
@@ -13,11 +13,11 @@
 # modifications to g++-win32.conf
 
 QMAKE_CC                = $${CROSS_COMPILE}gcc
-QMAKE_CFLAGS           += -fno-keep-inline-dllexport
+QMAKE_CFLAGS           += -fno-keep-inline-dllexport -fpermissive
 QMAKE_CFLAGS_WARN_ON   += -Wextra
 
 QMAKE_CXX               = $${CROSS_COMPILE}g++
-QMAKE_CXXFLAGS         += -fno-keep-inline-dllexport
+QMAKE_CXXFLAGS         += -fno-keep-inline-dllexport -fpermissive
 QMAKE_CXXFLAGS_WARN_ON  = $$QMAKE_CFLAGS_WARN_ON
 
 QMAKE_LINK              = $${CROSS_COMPILE}g++
