--- qtmultimedia/src/plugins/wmf/player/mfplayercontrol.cpp.orig	2021-01-26 21:16:57.370000000 +0000
+++ qtmultimedia/src/plugins/wmf/player/mfplayercontrol.cpp	2021-01-26 21:17:26.670000000 +0000
@@ -38,7 +38,7 @@
 ****************************************************************************/
 
 #include "mfplayercontrol.h"
-#include <qtcore/qdebug.h>
+#include <QtCore/qdebug.h>
 
 //#define DEBUG_MEDIAFOUNDATION
 
--- qtmultimedia/src/plugins/wmf/player/mfvideorenderercontrol.cpp.orig	2021-03-26 16:02:31.840000000 +0000
+++ qtmultimedia/src/plugins/wmf/player/mfvideorenderercontrol.cpp	2021-03-26 16:03:04.410000000 +0000
@@ -44,13 +44,13 @@
 
 #include <qabstractvideosurface.h>
 #include <qvideosurfaceformat.h>
-#include <qtcore/qtimer.h>
-#include <qtcore/qmutex.h>
-#include <qtcore/qcoreevent.h>
-#include <qtcore/qcoreapplication.h>
-#include <qtcore/qthread.h>
+#include <QtCore/qtimer.h>
+#include <QtCore/qmutex.h>
+#include <QtCore/qcoreevent.h>
+#include <QtCore/qcoreapplication.h>
+#include <QtCore/qthread.h>
 #include "guiddef.h"
-#include <qtcore/qdebug.h>
+#include <QtCore/qdebug.h>
 #include <QtMultimedia/private/qmediaopenglhelper_p.h>
 
 //#define DEBUG_MEDIAFOUNDATION
