--- qtdeclarative/src/qmldevtools/qmldevtools.pro.orig	2021-03-21 17:57:19.370000000 +0000
+++ qtdeclarative/src/qmldevtools/qmldevtools.pro	2021-03-21 17:58:18.610000000 +0000
@@ -2,6 +2,10 @@
 TARGET     = QtQmlDevTools
 QT         = core-private
 CONFIG    += minimal_syncqt internal_module generated_privates
+win32:!contains(CONFIG, static) {
+	CONFIG += shared
+	CONFIG -= static staticlib
+	}
 
 MODULE_INCNAME = QtQml
 INCLUDEPATH += $$OUT_PWD/../qml
