--- qtdeclarative/src/qml/parser/qqmljsglobal_p.h.orig	2021-03-21 18:02:59.090000000 +0000
+++ qtdeclarative/src/qml/parser/qqmljsglobal_p.h	2021-03-21 18:04:54.140000000 +0000
@@ -54,9 +54,9 @@
 
 #ifdef QT_CREATOR
 
-#  ifdef QDECLARATIVEJS_BUILD_DIR
+#  if defined(QDECLARATIVEJS_BUILD_DIR) && !defined(QT_STATIC)
 #    define QML_PARSER_EXPORT Q_DECL_EXPORT
-#  elif QML_BUILD_STATIC_LIB
+#  elif defined(QML_BUILD_STATIC_LIB) || defined(QT_STATIC)
 #    define QML_PARSER_EXPORT
 #  else
 #    define QML_PARSER_EXPORT Q_DECL_IMPORT
@@ -64,7 +64,7 @@
 
 #else // !QT_CREATOR
 #  ifndef QT_STATIC
-#    if defined(QT_BUILD_QMLDEVTOOLS_LIB) || defined(QT_QMLDEVTOOLS_LIB)
+#    if defined(QT_BUILD_QMLDEVTOOLS_LIB) || defined(QT_QMLDEVTOOLS_LIB) || defined(QT_STATIC)
        // QmlDevTools is a static library
 #      define QML_PARSER_EXPORT
 #    elif defined(QT_BUILD_QML_LIB)
