diff -rupN --no-dereference qtbase-everywhere-src-6.2.1/cmake/QtToolHelpers.cmake qtbase-everywhere-src-6.2.1-new/cmake/QtToolHelpers.cmake
--- qtbase-everywhere-src-6.2.1/cmake/QtToolHelpers.cmake	2021-10-20 13:54:23.000000000 +0200
+++ qtbase-everywhere-src-6.2.1-new/cmake/QtToolHelpers.cmake	2021-11-01 12:28:16.688394428 +0100
@@ -82,7 +82,7 @@ function(qt_internal_add_tool target_nam
         # instead of PATHS option, because any find_dependency call inside a Tools package would
         # not get the proper prefix when using PATHS.
         set(BACKUP_CMAKE_PREFIX_PATH ${CMAKE_PREFIX_PATH})
-        set(CMAKE_PREFIX_PATH "${QT_HOST_PATH}")
+        set(CMAKE_PREFIX_PATH "${QT_HOST_PATH_CMAKE_DIR}")
 
         # Search both with sysroots prepended as well as in the host system. When cross compiling
         # the mode_package might be set to ONLY only, and the Qt6 tools packages are actually
