--- colors.c	2013-03-20 11:19:08.000000000 -0400
+++ colors.c	2015-07-20 12:44:31.821014500 -0400
@@ -37,6 +37,10 @@
 #include "posixstat.h" // stat related macros (S_ISREG, ...)
 #include <fcntl.h> // S_ISUID
 
+#ifndef S_ISDIR
+#define	S_ISDIR(m)	(((m) & S_IFMT) == S_IFDIR)
+#endif
+
 // strlen()
 #if defined (HAVE_STRING_H)
 #  include <string.h>
@@ -151,12 +155,17 @@
       if (S_ISREG (mode))
         {
           colored_filetype = C_FILE;
-
+#ifdef S_ISUID
           if ((mode & S_ISUID) != 0 && is_colored (C_SETUID))
             colored_filetype = C_SETUID;
-          else if ((mode & S_ISGID) != 0 && is_colored (C_SETGID))
+		else
+#endif
+#ifdef S_ISGID
+          if ((mode & S_ISGID) != 0 && is_colored (C_SETGID))
             colored_filetype = C_SETGID;
-          else if (is_colored (C_CAP) && 0) //f->has_capability)
+          else
+#endif
+		  if (is_colored (C_CAP) && 0) //f->has_capability)
             colored_filetype = C_CAP;
           else if ((mode & S_IXUGO) != 0 && is_colored (C_EXEC))
             colored_filetype = C_EXEC;
--- colors.c.orig	2016-10-04 11:13:20.199758400 +0100
+++ colors.c	2016-10-04 11:13:48.698813300 +0100
@@ -186,12 +186,12 @@
       if (S_ISREG (mode))
         {
           colored_filetype = C_FILE;
-#ifdef S_ISUID
+#ifndef _WIN32
           if ((mode & S_ISUID) != 0 && is_colored (C_SETUID))
             colored_filetype = C_SETUID;
 		else
 #endif
-#ifdef S_ISGID
+#ifndef _WIN32
           if ((mode & S_ISGID) != 0 && is_colored (C_SETGID))
             colored_filetype = C_SETGID;
           else
--- histfile.c.orig	2016-10-04 11:22:09.873794900 +0100
+++ histfile.c	2016-10-04 11:22:51.678317000 +0100
@@ -610,8 +610,10 @@
      user is running this, it's a no-op.  If the shell is running after sudo
      with a shared history file, we don't want to leave the history file
      owned by root. */
+#ifndef _WIN32
   if (rv == 0 && exists)
     r = chown (filename, finfo.st_uid, finfo.st_gid);
+#endif
 
   xfree (filename);
   FREE (tempname);
--- histfile.c.orig	2016-10-04 11:25:51.081688700 +0100
+++ histfile.c	2016-10-04 11:26:21.036152700 +0100
@@ -759,8 +759,10 @@
      user is running this, it's a no-op.  If the shell is running after sudo
      with a shared history file, we don't want to leave the history file
      owned by root. */
+#ifndef _WIN32
   if (rv == 0 && exists)
     mode = chown (histname, finfo.st_uid, finfo.st_gid);
+#endif
 
   FREE (histname);
   FREE (tempname);
--- colors.c.orig	2016-10-04 11:28:58.535955500 +0100
+++ colors.c	2016-10-04 11:31:29.993417300 +0100
@@ -157,6 +157,7 @@
     {
       mode = astat.st_mode;
 #if defined (HAVE_LSTAT)
+#ifndef _WIN32
       if (S_ISLNK (mode))
 	{
 	  linkok = stat (name, &linkstat) == 0;
@@ -164,6 +165,7 @@
 	    mode = linkstat.st_mode;
 	}
       else
+#endif /* _WIN32 */
 #endif
 	linkok = 1;
     }
@@ -224,8 +226,10 @@
         colored_filetype = C_LINK;
       else if (S_ISFIFO (mode))
         colored_filetype = C_FIFO;
+#ifndef _WIN32
       else if (S_ISSOCK (mode))
         colored_filetype = C_SOCK;
+#endif
       else if (S_ISBLK (mode))
         colored_filetype = C_BLK;
       else if (S_ISCHR (mode))
--- colors.c.orig	2016-10-04 11:33:44.086870600 +0100
+++ colors.c	2016-10-04 11:34:49.550204800 +0100
@@ -222,8 +222,10 @@
             colored_filetype = C_STICKY;
 #endif
         }
+#ifndef _WIN32
       else if (S_ISLNK (mode))
         colored_filetype = C_LINK;
+#endif
       else if (S_ISFIFO (mode))
         colored_filetype = C_FIFO;
 #ifndef _WIN32
