--- rtaudio.pc.in.orig	2017-12-10 10:54:53.837857707 +0000
+++ rtaudio.pc.in	2017-12-10 10:56:17.573049223 +0000
@@ -7,6 +7,6 @@
 Description: RtAudio - a set of C++ classes that provide a common API for realtime audio input/output
 Version: @PACKAGE_VERSION@
 Requires: @req@ 
-Libs: -L${libdir} -lrtaudio
+Libs: -L${libdir} -lpthread -lrtaudio -lole32 -loleaut32 -lwinmm -lksuser -luuid 
 Libs.private: -lpthread
 Cflags: -pthread -I${includedir} @api@
