--- configure.ac.orig	2017-09-06 17:09:37.184503900 +0100
+++ configure.ac	2017-09-06 17:10:28.896612500 +0100
@@ -29,13 +29,13 @@
 AC_SUBST(SO_VERSION)
 
 # Check version number coherency between RtAudio.h and configure.ac
-AC_MSG_CHECKING([that version numbers are coherent])
-AC_RUN_IFELSE(
-   [AC_LANG_PROGRAM([#include <string.h>
-                     `grep "define RTAUDIO_VERSION" $srcdir/RtAudio.h`],
-                    [return strcmp(RTAUDIO_VERSION, PACKAGE_VERSION);])],
-   [AC_MSG_RESULT([yes])],
-   [AC_MSG_FAILURE([testing RTAUDIO_VERSION==PACKAGE_VERSION failed, check that RtAudio.h defines RTAUDIO_VERSION as "$PACKAGE_VERSION" or that the first line of configure.ac has been updated.])])
+#AC_MSG_CHECKING([that version numbers are coherent])
+#AC_RUN_IFELSE(
+#   [AC_LANG_PROGRAM([#include <string.h>
+#                     `grep "define RTAUDIO_VERSION" $srcdir/RtAudio.h`],
+#                    [return strcmp(RTAUDIO_VERSION, PACKAGE_VERSION);])],
+#   [AC_MSG_RESULT([yes])],
+#   [AC_MSG_FAILURE([testing RTAUDIO_VERSION==PACKAGE_VERSION failed, check that RtAudio.h defines RTAUDIO_VERSION as "$PACKAGE_VERSION" or that the first line of configure.ac has been updated.])])
 
 # Enable some nice automake features if they are available
 m4_ifdef([AM_MAINTAINER_MODE], [AM_MAINTAINER_MODE])
