--- build.ninja.orig	2026-05-09 10:11:27.324156677 +0100
+++ build.ninja	2026-05-09 10:15:10.678120993 +0100
@@ -504,10 +504,10 @@
 # Link the shared library libshaderc/libshaderc_shared.dll
 
 build libshaderc/libshaderc_shared.dll libshaderc/libshaderc_shared.dll.a: CXX_SHARED_LIBRARY_LINKER__shaderc_shared_Debug libshaderc/CMakeFiles/shaderc_shared.dir/src/shaderc.cc.obj | libshaderc_util/libshaderc_util.a || libshaderc_util/libshaderc_util.a
-  CONFIG = Debug
+  CONFIG = Release
   LANGUAGE_COMPILE_FLAGS = -Wimplicit-fallthrough -g
   LINK_FLAGS = -shared -static -static-libgcc -static-libstdc++
-  LINK_LIBRARIES = -lglslang  -lSPIRV-Tools-opt  -lOSDependent  -lOGLCompiler  -lglslang  libshaderc_util/libshaderc_util.a  -lSPIRV  -lSPIRV-Tools  -lHLSL  -lglslang  -lOSDependent  -lOGLCompiler  -lglslang  -lOSDependent  -lOGLCompiler  -lSPIRV-Tools-opt  -lSPIRV  -lkernel32 -luser32 -lgdi32 -lwinspool -lshell32 -lole32 -loleaut32 -luuid -lcomdlg32 -ladvapi32
+  LINK_LIBRARIES = -lSPIRV-Tools -lHLSL -lglslang -lSPIRV-Tools-opt -lOSDependent -lOGLCompiler -lglslang -lpthread libshaderc_util/libshaderc_util.a -lSPIRV -lSPIRV-Tools -lHLSL -lglslang -lOSDependent -lOGLCompiler -lglslang -lOSDependent -lOGLCompiler -lSPIRV-Tools-opt -lpthread -lSPIRV -lSPIRV-Tools -lHLSL -lkernel32 -luser32 -lgdi32 -lwinspool -lshell32 -lole32 -loleaut32 -luuid -lcomdlg32 -ladvapi32
   OBJECT_DIR = libshaderc/CMakeFiles/shaderc_shared.dir
   POST_BUILD = :
   PRE_LINK = :
@@ -744,10 +744,10 @@
 # Link the executable glslc/glslc.exe
 
 build glslc/glslc.exe: CXX_EXECUTABLE_LINKER__glslc_exe_Debug glslc/CMakeFiles/glslc_exe.dir/src/main.cc.obj | glslc/libglslc.a libshaderc_util/libshaderc_util.a libshaderc/libshaderc.a libshaderc_util/libshaderc_util.a || build-version glslc/libglslc.a libshaderc/libshaderc.a libshaderc_util/libshaderc_util.a
-  CONFIG = Debug
+  CONFIG = Release
   FLAGS = -Wimplicit-fallthrough -g
   LINK_FLAGS = -static -static-libgcc -static-libstdc++
-  LINK_LIBRARIES = -lHLSL  glslc/libglslc.a  libshaderc_util/libshaderc_util.a  libshaderc/libshaderc.a  -lHLSL  libshaderc_util/libshaderc_util.a  -lHLSL  -lSPIRV-Tools-opt  -lSPIRV-Tools  -lglslang  -lOSDependent  -lOGLCompiler  -lglslang  -lOSDependent  -lOGLCompiler  -lSPIRV  -lkernel32 -luser32 -lgdi32 -lwinspool -lshell32 -lole32 -loleaut32 -luuid -lcomdlg32 -ladvapi32
+  LINK_LIBRARIES = -lHLSL glslc/libglslc.a libshaderc_util/libshaderc_util.a libshaderc/libshaderc.a -lHLSL libshaderc_util/libshaderc_util.a -lHLSL -lSPIRV-Tools-opt -lSPIRV-Tools -lglslang -lOSDependent -lOGLCompiler -lglslang -lHLSL -lOSDependent -lOGLCompiler -lSPIRV -lHLSL -lpthread -lkernel32 -luser32 -lgdi32 -lwinspool -lshell32 -lole32 -loleaut32 -luuid -lcomdlg32 -ladvapi32
   OBJECT_DIR = glslc/CMakeFiles/glslc_exe.dir
   POST_BUILD = :
   PRE_LINK = :
