--- build.ninja.orig	2020-02-02 12:11:55.720990867 +0000
+++ build.ninja	2020-02-02 12:16:06.974090841 +0000
@@ -426,7 +426,7 @@
 build libshaderc/libshaderc_shared.dll libshaderc/libshaderc_shared.dll.a: CXX_SHARED_LIBRARY_LINKER__shaderc_shared libshaderc/CMakeFiles/shaderc_shared.dir/src/shaderc.cc.obj | libshaderc_util/libshaderc_util.a || libshaderc_util/libshaderc_util.a
   LANGUAGE_COMPILE_FLAGS = -Wimplicit-fallthrough -g
   LINK_FLAGS = -static -static-libgcc -static-libstdc++
-  LINK_LIBRARIES = -lglslang  -lSPIRV-Tools-opt  -lOSDependent  -lOGLCompiler  -lglslang  libshaderc_util/libshaderc_util.a  -lSPIRV  -lSPIRV-Tools  -lHLSL  -lglslang  -lOSDependent  -lOGLCompiler  -lglslang  -lOSDependent  -lOGLCompiler  -lSPIRV-Tools-opt  -lSPIRV  -lkernel32 -luser32 -lgdi32 -lwinspool -lshell32 -lole32 -loleaut32 -luuid -lcomdlg32 -ladvapi32
+  LINK_LIBRARIES = -lSPIRV-Tools -lHLSL -lglslang -lSPIRV-Tools-opt -lOSDependent -lOGLCompiler -lglslang -lpthread libshaderc_util/libshaderc_util.a -lSPIRV -lSPIRV-Tools -lHLSL -lglslang -lOSDependent -lOGLCompiler -lglslang -lOSDependent -lOGLCompiler -lSPIRV-Tools-opt -lpthread -lSPIRV -lSPIRV-Tools -lHLSL -lkernel32 -luser32 -lgdi32 -lwinspool -lshell32 -lole32 -loleaut32 -luuid -lcomdlg32 -ladvapi32
   OBJECT_DIR = libshaderc/CMakeFiles/shaderc_shared.dir
   POST_BUILD = :
   PRE_LINK = :
@@ -670,7 +670,7 @@
 build glslc/glslc.exe: CXX_EXECUTABLE_LINKER__glslc_exe glslc/CMakeFiles/glslc_exe.dir/src/main.cc.obj | glslc/libglslc.a libshaderc_util/libshaderc_util.a libshaderc/libshaderc.a libshaderc_util/libshaderc_util.a || build-version glslc/libglslc.a libshaderc/libshaderc.a libshaderc_util/libshaderc_util.a
   FLAGS = -Wimplicit-fallthrough -g
   LINK_FLAGS = -static -static-libgcc -static-libstdc++
-  LINK_LIBRARIES = -lHLSL  glslc/libglslc.a  libshaderc_util/libshaderc_util.a  libshaderc/libshaderc.a  -lHLSL  libshaderc_util/libshaderc_util.a  -lHLSL  -lSPIRV-Tools-opt  -lSPIRV-Tools  -lglslang  -lOSDependent  -lOGLCompiler  -lglslang  -lOSDependent  -lOGLCompiler  -lSPIRV  -lkernel32 -luser32 -lgdi32 -lwinspool -lshell32 -lole32 -loleaut32 -luuid -lcomdlg32 -ladvapi32
+  LINK_LIBRARIES = -lHLSL glslc/libglslc.a libshaderc_util/libshaderc_util.a libshaderc/libshaderc.a -lHLSL libshaderc_util/libshaderc_util.a -lHLSL -lSPIRV-Tools-opt -lSPIRV-Tools -lglslang -lOSDependent -lOGLCompiler -lglslang -lHLSL -lOSDependent -lOGLCompiler -lSPIRV -lHLSL -lpthread -lkernel32 -luser32 -lgdi32 -lwinspool -lshell32 -lole32 -loleaut32 -luuid -lcomdlg32 -ladvapi32
   OBJECT_DIR = glslc/CMakeFiles/glslc_exe.dir
   POST_BUILD = :
   PRE_LINK = :
