--- configure.ac.orig	2016-01-24 21:43:16.331766659 +0000
+++ configure.ac	2016-01-24 21:44:21.006766659 +0000
@@ -13,6 +13,8 @@
 # These are flags passed to automake (though they look like gcc flags!)
 AM_INIT_AUTOMAKE([-Wall])
 
+AM_PROG_AR
+
 LT_INIT
 AC_SUBST([LIBTOOL_DEPS])
 AC_PROG_CXX
--- Makefile.am.orig	2016-01-24 21:48:41.428766659 +0000
+++ Makefile.am	2016-01-24 21:49:09.437766659 +0000
@@ -3,7 +3,7 @@
 # Library.
 lib_LTLIBRARIES = libsnappy.la
 libsnappy_la_SOURCES = snappy.cc snappy-sinksource.cc snappy-stubs-internal.cc snappy-c.cc
-libsnappy_la_LDFLAGS = -version-info $(SNAPPY_LTVERSION)
+libsnappy_la_LDFLAGS = -no-undefined -version-info $(SNAPPY_LTVERSION)
 
 include_HEADERS = snappy.h snappy-sinksource.h snappy-stubs-public.h snappy-c.h
 noinst_HEADERS = snappy-internal.h snappy-stubs-internal.h snappy-test.h
