--- CMakeLists.txt.orig	2018-06-04 20:56:55.854029300 +0100
+++ CMakeLists.txt	2018-06-04 20:58:02.399678900 +0100
@@ -142,12 +142,12 @@
 
   # For MinGW cross compile, statically link to the C++ runtime.
   # But it still depends on MSVCRT.dll.
-  if (${CMAKE_SYSTEM_NAME} MATCHES "Windows")
-    if (${CMAKE_CXX_COMPILER_ID} MATCHES "GNU")
-      set_target_properties(${TARGET} PROPERTIES
-        LINK_FLAGS -static -static-libgcc -static-libstdc++)
-    endif()
-  endif()
+  #if (${CMAKE_SYSTEM_NAME} MATCHES "Windows")
+  #  if (${CMAKE_CXX_COMPILER_ID} MATCHES "GNU")
+  #    set_target_properties(${TARGET} PROPERTIES
+  #      LINK_FLAGS -static -static-libgcc -static-libstdc++)
+  #  endif()
+  #endif()
 endfunction()
 
 if(NOT COMMAND find_host_package)
