--- Makefile.am.orig	2022-11-29 15:52:51.206082848 +0000
+++ Makefile.am	2022-11-29 15:56:10.686082590 +0000
@@ -15,20 +15,7 @@
 endif
 
 if COMPILE_WINDOWS
-synaesthesia_LDFLAGS = -static
-
-if COMPILE_WINICON
-synaesthesia_SOURCES += syn-res.rc
-synaesthesia_DEPENDENCIES = synaesthesia.ico
-
-.rc.o:
-	$(WINDRES) -o $@ $<
-
-syn-res.o: synaesthesia.ico
-
-synaesthesia.ico: synaesthesia.png
-	icotool -c -o $@ $^
-endif
+synaesthesia_LDFLAGS = -shared -Wl,-no-undefined
 
 if COMPILE_WINAMP
 synaesthesia_SOURCES += winamp.cc
@@ -36,16 +23,6 @@
 endif
 endif COMPILE_WINDOWS
 
-if COMPILE_SDLICON
-syn_icon.rgb: synaesthesia.png
-	convert $< -resize "64x64!" $@
-
-syn_icon_rgb.cc: syn_icon.rgb
-	xxd -i $< > $@
-
-nodist_synaesthesia_SOURCES = syn_icon_rgb.cc
-endif COMPILE_SDLICON
-
 if COMPILE_AUDACIOUS
 synaesthesia_SOURCES += audacious.cc
 synaesthesia_LDFLAGS = -shared -Wl,-no-undefined
@@ -69,12 +46,3 @@
 CLEANFILES = synaesthesia.ico syn_icon.rgb syn_icon_rgb.cc \
              synaesthesia.js synaesthesia.html.mem a.out.js
 
-install-data-local:
-	-xdg-icon-resource install --novendor --context apps \
-	                           --size 256 synaesthesia.png synaesthesia
-	-xdg-desktop-menu install --novendor synaesthesia.desktop
-
-uninstall-local:
-	-xdg-icon-resource uninstall --novendor --context apps \
-	                             --size 256 synaesthesia
-	-xdg-desktop-menu uninstall --novendor synaesthesia.desktop
--- configure.ac.orig	2022-11-29 15:56:27.096082572 +0000
+++ configure.ac	2022-11-29 15:56:55.716082634 +0000
@@ -165,32 +165,5 @@
        AC_DEFINE([HAVE_SDLAUDIO], [1])])
 AM_CONDITIONAL([COMPILE_SDLAUDIO], [test "x$enable_sdlaudio" = "xyes"])
 
-dnl Use icon if present
-AH_TEMPLATE([HAVE_ICON], [Set to 1 if window icon is compiled in and set by code.])
-AC_CHECK_FILE([synaesthesia.png],
-              [AS_IF([test "x$windows_host" = "xyes"],
-                     [AC_CHECK_PROG([has_icotool],[icotool],[yes],[no])
-                      AS_IF([test "x$has_icotool" = "xyes"],
-                            [AC_CHECK_TOOL([WINDRES], [windres], [:])],
-                            [WINDRES=:])
-                      AS_IF([test "x$WINDRES" = "x:"],
-                            [windows_icon=no
-                             AC_MSG_WARN([Icon found but need icotool from icoutils and windres to use it])],
-                            [windows_icon=yes])],
-                      [windows_icon=0])
-               AS_IF([test "x$enable_emscripten" = "xyes"], [sdl_icon=no],
-                     [test "x$windows_host" = "xyes" -a "x$with_sdl2" != "xno"], [sdl_icon=no],
-                     [AC_CHECK_PROG([has_convert],[convert],[yes],[no])
-                      AS_IF([test "x$has_convert" = "xyes"],
-                            [AC_CHECK_PROG([has_xxd],[xxd],[yes],[no])],
-                            [use_xxd=no])
-                      AS_IF([test "x$has_xxd" = "xyes"],
-                            [AC_DEFINE([HAVE_ICON], [1])
-                             sdl_icon=yes],
-                            [AC_MSG_WARN([Icon found but need convert from ImageMagick and xxd to use it])
-                             sdl_icon=no])])])
-AM_CONDITIONAL([COMPILE_WINICON], [test "x$windows_icon" = "xyes"])
-AM_CONDITIONAL([COMPILE_SDLICON], [test "x$sdl_icon" = "xyes"])
-
 AC_CONFIG_FILES([Makefile])
 AC_OUTPUT
