--- ccstruct/imagedata.cpp.orig	2016-11-27 17:12:34.479035300 +0000
+++ ccstruct/imagedata.cpp	2016-11-27 17:13:39.536213700 +0000
@@ -30,7 +30,7 @@
 #include "helpers.h"
 #include "tprintf.h"
 
-#include <thread>
+#include <boost/thread.hpp>
 
 // Number of documents to read ahead while training. Doesn't need to be very
 // large.
@@ -449,7 +449,7 @@
     if (needs_loading) LoadPageInBackground(index);
     // We can't directly load the page, or the background load will delete it
     // while the caller is using it, so give it a chance to work.
-    std::this_thread::sleep_for(std::chrono::seconds(1));
+    boost::this_thread::sleep(boost::posix_time::seconds(1));
   }
   return page;
 }
