--- src/traverso/CMakeLists.txt.orig	2018-07-29 09:13:02.496580500 +0100
+++ src/traverso/CMakeLists.txt	2018-07-29 09:13:42.036477600 +0100
@@ -193,11 +193,17 @@
     TARGET_LINK_LIBRARIES(traverso
         sndfile-1
         fftw3-3
+        speex
+        speexdsp
+        ogg
     )
 ELSE(WIN32)
     TARGET_LINK_LIBRARIES(traverso
         sndfile
         fftw3
+        speex
+        speexdsp
+        ogg
     )
 ENDIF(WIN32)
 
--- src/traverso/CMakeLists.txt.orig	2018-07-29 09:26:37.695630400 +0100
+++ src/traverso/CMakeLists.txt	2018-07-29 09:28:46.715713000 +0100
@@ -196,6 +196,8 @@
         speex
         speexdsp
         ogg
+        Qt5Widgets
+        Qt5Xml
     )
 ELSE(WIN32)
     TARGET_LINK_LIBRARIES(traverso
@@ -204,6 +206,8 @@
         speex
         speexdsp
         ogg
+        Qt5Widgets
+        Qt5Xml
     )
 ENDIF(WIN32)
 
