--- src/vamp-sdk/PluginAdapter.cpp.orig	2019-10-27 15:33:50.990000000 +0000
+++ src/vamp-sdk/PluginAdapter.cpp	2019-10-27 15:36:32.710000000 +0000
@@ -39,7 +39,8 @@
 #include <cstring>
 #include <cstdlib>
 
-#include <mutex>
+/* #include <mutex> */
+#include <boost/thread/mutex.hpp>
 
 #if ( VAMP_SDK_MAJOR_VERSION != 2 || VAMP_SDK_MINOR_VERSION != 9 )
 #error Unexpected version of Vamp SDK header included
@@ -50,8 +51,8 @@
 using std::string;
 using std::cerr;
 using std::endl;
-using std::mutex;
-using std::lock_guard;
+using boost::mutex;
+using boost::lock_guard;
 
 //#define DEBUG_PLUGIN_ADAPTER 1
 
--- src/vamp-sdk/PluginAdapter.cpp.orig	2019-10-27 15:41:56.040000000 +0000
+++ src/vamp-sdk/PluginAdapter.cpp	2019-10-27 15:42:36.640000000 +0000
@@ -41,7 +41,7 @@
 
 /* #include <mutex> */
 #include <boost/thread/mutex.hpp>
-
+#include <boost/thread.hpp>
 #if ( VAMP_SDK_MAJOR_VERSION != 2 || VAMP_SDK_MINOR_VERSION != 9 )
 #error Unexpected version of Vamp SDK header included
 #endif
