--- src/GvimExt/gvimext.h.orig	2019-09-01 16:34:45.251857476 +0100
+++ src/GvimExt/gvimext.h	2019-09-01 16:36:08.347069273 +0100
@@ -58,7 +58,12 @@
 #define INITGUID
 #include <initguid.h>
 #include <shlguid.h>
-
+// mingw-w64 recently, Aug 2019, defines the below in libuuid.
+// We don't want them here.
+//
+#undef FMTID_ImageSummaryInformation
+#undef FMTID_MediaFileSummaryInformation
+#undef FMTID_AudioSummaryInformation
 //
 // The class ID of this Shell extension class.
 //
