--- modules/video_output/opengl/egl.c.orig	2020-11-01 19:49:01.610000000 +0000
+++ modules/video_output/opengl/egl.c	2020-11-01 19:50:11.380000000 +0000
@@ -87,7 +87,7 @@
 # define Resize (NULL)
 #endif
 
-static void SwapBuffers (vlc_gl_t *gl)
+static void vlc_eglSwapBuffers (vlc_gl_t *gl)
 {
     vlc_gl_sys_t *sys = gl->sys;
 
@@ -380,7 +380,7 @@
     gl->make_current = MakeCurrent;
     gl->release_current = ReleaseCurrent;
     gl->resize = Resize;
-    gl->swap = SwapBuffers;
+    gl->swap = vlc_eglSwapBuffers;
     gl->get_proc_address = GetSymbol;
     gl->destroy = Close;
     gl->egl.queryString = QueryString;
