--- modules/video_output/win32/direct3d11.c.orig	2017-11-11 15:37:38.385681700 +0000
+++ modules/video_output/win32/direct3d11.c	2017-11-11 15:38:55.949137300 +0000
@@ -40,11 +40,10 @@
 #define COBJMACROS
 #include <initguid.h>
 #include <d3d11.h>
-#ifdef HAVE_DXGI1_6_H
-# include <dxgi1_6.h>
-#else
-# include <dxgi1_5.h>
-#endif
+#include <dxgi1_2.h>
+#include <dxgi1_4.h>
+#include <dxgi1_5.h>
+#include <dxgi1_6.h>
 
 /* avoided until we can pass ISwapchainPanel without c++/cx mode
 # include <windows.ui.xaml.media.dxinterop.h> */
