diff --git lib/Makefile.am lib/Makefile.am
index 892d9b14f1..5252d59e6a 100644
--- lib/Makefile.am
+++ lib/Makefile.am
@@ -65,7 +65,7 @@ libvlc_la_LDFLAGS = \
 libvlc_la_DEPENDENCIES = libvlc.sym
 if HAVE_WIN32
 libvlc_la_DEPENDENCIES += ../src/libvlc_win32_rc.$(OBJEXT)
-libvlc_la_LDFLAGS += -Wl,../src/libvlc_win32_rc.$(OBJEXT) -avoid-version -Wc,-static
+libvlc_la_LDFLAGS += -Wl,../src/libvlc_win32_rc.$(OBJEXT) -avoid-version
 endif
 if HAVE_OS2
 libvlc_la_LDFLAGS += -avoid-version
diff --git modules/common.am modules/common.am
index 1b9117c98d..30e30ca4ec 100644
--- modules/common.am
+++ modules/common.am
@@ -33,7 +33,7 @@ AM_LDFLAGS = \
 	-no-undefined \
 	$(top_builddir)/compat/libcompat.la $(LTLIBVLCCORE) 
 if HAVE_WIN32
-AM_LDFLAGS += $(top_builddir)/modules/module.rc.lo -Wc,-static
+AM_LDFLAGS += $(top_builddir)/modules/module.rc.lo
 endif
 AM_YFLAGS = -d
 
diff --git src/Makefile.am src/Makefile.am
index 044169156c..e961f7734d 100644
--- src/Makefile.am
+++ src/Makefile.am
@@ -472,7 +472,7 @@ libvlccore_la_LDFLAGS = \
 libvlccore_la_DEPENDENCIES = libvlccore.sym
 if HAVE_WIN32
 libvlccore_la_DEPENDENCIES += libvlc_win32_rc.$(OBJEXT)
-libvlccore_la_LDFLAGS += -Wl,libvlc_win32_rc.$(OBJEXT) -avoid-version -Wc,-static $(LIBCOM)
+libvlccore_la_LDFLAGS += -Wl,libvlc_win32_rc.$(OBJEXT) -avoid-version $(LIBCOM)
 endif
 if HAVE_OS2
 libvlccore_la_LDFLAGS += -avoid-version
-- 
2.15.0
