--- Makefile.am.orig	2017-07-22 21:14:44.180563100 +0100
+++ Makefile.am	2017-07-22 21:16:19.048542900 +0100
@@ -107,7 +107,7 @@
 	cd lib && $(MAKE) $(AM_MAKEFLAGS) libvlc.la
 
 core: libvlc vlc$(EXEEXT)
-	cd bin && $(MAKE) $(AM_MAKEFLAGS) vlc$(EXEEXT) vlc-static$(EXEEXT)
+	cd bin && $(MAKE) $(AM_MAKEFLAGS) vlc$(EXEEXT) 
 
 doc:
 	cd doc && $(MAKE) $(AM_MAKEFLAGS) doc
--- configure.ac.orig	2017-07-22 21:16:38.256910900 +0100
+++ configure.ac	2017-07-22 21:17:20.622198300 +0100
@@ -860,7 +860,7 @@
 dnl On some OS we need static linking
 AS_IF([test -n "${PKG_CONFIG}" ],[
     AS_IF([test "${SYS}" = "mingw32" -o "${SYS}" = "darwin" -o "${SYS}" = "os2" ],[
-        PKG_CONFIG="${PKG_CONFIG} --static"
+        PKG_CONFIG="${PKG_CONFIG} "
     ])
 ])
 
 
--- modules/gui/qt/maininterface/main_interface_win32.cpp.orig	2020-01-03 17:19:25.827161445 +0000
+++ modules/gui/qt/maininterface/main_interface_win32.cpp	2020-01-03 17:20:10.762708682 +0000
@@ -37,7 +37,7 @@
 #include <assert.h>
 
 #include <QWindow>
-#include <qpa/qplatformnativeinterface.h>
+#include <5.14.2/QtGui/qpa/qplatformnativeinterface.h>
 
 #define WM_APPCOMMAND 0x0319
 
--- modules/gui/qt/widgets/native/interface_widgets.cpp.orig	2020-01-03 17:43:24.632965847 +0000
+++ modules/gui/qt/widgets/native/interface_widgets.cpp	2020-01-03 17:43:47.524748332 +0000
@@ -61,7 +61,7 @@
 
 #if defined(_WIN32)
 #include <QWindow>
-#include <qpa/qplatformnativeinterface.h>
+#include <5.14.2/QtGui/qpa/qplatformnativeinterface.h>
 #endif
 
 #include <math.h>
