--- configure.ac.orig	2022-02-01 18:43:18.192336600 +0000
+++ configure.ac	2022-02-01 18:43:28.852336600 +0000
@@ -46,7 +46,7 @@
 AC_PROG_CC
 AC_USE_SYSTEM_EXTENSIONS
 VLC_PROG_CC_C11
-AC_DEFINE([_FORTIFY_SOURCE], 2, [Define to 2 to get glibc warnings.])
+AC_DEFINE([_FORTIFY_SOURCE], 0, [Define to 2 to get glibc warnings.])
 AC_DEFINE([_FILE_OFFSET_BITS], 64, [Define to 64 for large files support.])
 AH_TOP([
 #ifndef _REENTRANT
