--- modules/gui/qt/Makefile.am
+++ modules/gui/qt/Makefile.am
@@ -45,7 +45,7 @@
 libqt_plugin_la_LIBADD += $(WAYLAND_CLIENT_LIBS)
 endif
 if HAVE_WIN32
+libqt_plugin_la_LIBADD += $(LIBCOM) -lcomctl32 -luuid
-libqt_plugin_la_LIBADD += $(LIBCOM) -lcomctl32 -luuid -ld3d11
 endif
 if UPDATE_CHECK
 libqt_plugin_la_CPPFLAGS += -DUPDATE_CHECK
--- modules/gui/qt/Makefile.am.orig	2020-08-27 21:40:07.120000000 +0100
+++ modules/gui/qt/Makefile.am	2020-08-27 21:40:20.200000000 +0100
@@ -228,14 +228,6 @@
 	gui/qt/maininterface/compositor_win7.cpp \
 	gui/qt/maininterface/compositor_win7.hpp
 
-if HAVE_DCOMP
-libqt_plugin_la_SOURCES += \
-	gui/qt/maininterface/compositor_dcomp.cpp \
-	gui/qt/maininterface/compositor_dcomp.hpp \
-	gui/qt/maininterface/compositor_dcomp_error.hpp \
-	gui/qt/maininterface/compositor_dcomp_uisurface.cpp \
-	gui/qt/maininterface/compositor_dcomp_uisurface.hpp
-endif
 endif
 
 # Meta-object compilation
--- modules/gui/qt/Makefile.am.orig	2020-08-27 21:54:47.030000000 +0100
+++ modules/gui/qt/Makefile.am	2020-08-27 21:54:58.530000000 +0100
@@ -348,12 +348,6 @@
 nodist_libqt_plugin_la_SOURCES += gui/qt/maininterface/main_interface_win32.moc.cpp \
 	gui/qt/maininterface/compositor_win7.moc.cpp
 
-if HAVE_DCOMP
-nodist_libqt_plugin_la_SOURCES += \
-	gui/qt/maininterface/compositor_dcomp.moc.cpp \
-	gui/qt/maininterface/compositor_dcomp_uisurface.moc.cpp
-endif
-
 endif
 
 nodist_libqt_plugin_la_SOURCES += \
--- modules/gui/qt/Makefile.am.orig	2020-11-01 20:01:03.040000000 +0000
+++ modules/gui/qt/Makefile.am	2020-11-01 20:02:28.410000000 +0000
@@ -232,9 +232,7 @@
 if HAVE_WIN32
 libqt_plugin_la_SOURCES += \
 	gui/qt/maininterface/main_interface_win32.cpp \
-	gui/qt/maininterface/main_interface_win32.hpp \
-	gui/qt/maininterface/compositor_win7.cpp \
-	gui/qt/maininterface/compositor_win7.hpp
+	gui/qt/maininterface/main_interface_win32.hpp 
 
 endif
 
--- modules/gui/qt/Makefile.am.orig	2020-11-01 20:17:01.280000000 +0000
+++ modules/gui/qt/Makefile.am	2020-11-01 20:17:17.230000000 +0000
@@ -347,8 +347,7 @@
 	gui/qt/widgets/native/searchlineedit.moc.cpp
 
 if HAVE_WIN32
-nodist_libqt_plugin_la_SOURCES += gui/qt/maininterface/main_interface_win32.moc.cpp \
-	gui/qt/maininterface/compositor_win7.moc.cpp
+nodist_libqt_plugin_la_SOURCES += gui/qt/maininterface/main_interface_win32.moc.cpp 
 
 endif
 
--- modules/gui/qt/maininterface/compositor.cpp.orig	2021-08-10 20:48:36.736213500 +0100
+++ modules/gui/qt/maininterface/compositor.cpp	2021-08-10 20:49:01.426213500 +0100
@@ -19,13 +19,6 @@
 #include "compositor.hpp"
 #include "compositor_dummy.hpp"
 
-#ifdef _WIN32
-#ifdef HAVE_DCOMP_H
-#  include "compositor_dcomp.hpp"
-#endif
-#  include "compositor_win7.hpp"
-#endif
-
 namespace vlc {
 
 template<typename T>
--- modules/gui/qt/maininterface/compositor.cpp.orig	2021-08-11 17:09:20.846213500 +0100
+++ modules/gui/qt/maininterface/compositor.cpp	2021-08-11 17:10:44.466213500 +0100
@@ -36,12 +36,6 @@
     Compositor* (*instanciate)(qt_intf_t *p_intf);
     bool (*preInit)(qt_intf_t *p_intf);
 } static compositorList[] = {
-#ifdef _WIN32
-#ifdef HAVE_DCOMP_H
-    {"dcomp", &instanciateCompositor<CompositorDirectComposition>, &preInit<CompositorDirectComposition> },
-#endif
-    {"win7", &instanciateCompositor<CompositorWin7>, &preInit<CompositorWin7> },
-#endif
     {"dummy", &instanciateCompositor<CompositorDummy>, &preInit<CompositorDummy> }
 };
 
