--- bin/winvlc.c.orig	2017-08-03 23:50:29.070393600 +0100
+++ bin/winvlc.c	2017-08-03 23:51:09.427915300 +0100
@@ -156,7 +156,7 @@
      * The LoadLibrary* calls from the modules and the 3rd party code
      * will search in SYSTEM32 only
      * */
-    SetDefaultDllDirectories(LOAD_LIBRARY_SEARCH_SYSTEM32);
+    SetDefaultDllDirectories(LOAD_LIBRARY_SEARCH_DEFAULT_DIRS);
     /***
      * Load DLLs from system32 before any other folder (when possible)
      */
--- src/win32/plugin.c.orig	2017-12-17 20:30:15.347086546 +0000
+++ src/win32/plugin.c	2017-12-17 20:30:41.214782502 +0000
@@ -62,7 +62,7 @@
     DWORD mode;
     if (SetThreadErrorMode (SEM_FAILCRITICALERRORS, &mode) != 0)
     {
-        handle = LoadLibraryExW(wfile, NULL, LOAD_LIBRARY_SEARCH_SYSTEM32);
+        handle = LoadLibraryExW(wfile, NULL, LOAD_LIBRARY_SEARCH_DEFAULT_DIRS);
         SetThreadErrorMode (mode, NULL);
     }
 #else
