--- modules/codec/Makefile.am.orig	2022-02-01 21:16:24.892336600 +0000
+++ modules/codec/Makefile.am	2022-02-01 21:16:36.932336600 +0000
@@ -269,7 +269,7 @@
 libzvbi_plugin_la_LDFLAGS = $(AM_LDFLAGS) -rpath '$(codecdir)'
 libzvbi_plugin_la_LIBADD = $(ZVBI_LIBS) $(SOCKET_LIBS)
 if HAVE_WIN32
-libzvbi_plugin_la_LIBADD += -lpthreadGC2
+libzvbi_plugin_la_LIBADD += -lpthread
 endif
 EXTRA_LTLIBRARIES += libzvbi_plugin.la
 codec_LTLIBRARIES += $(LTLIBzvbi)
