--- modules/gui/qt/Makefile.am
+++ modules/gui/qt/Makefile.am
@@ -45,7 +45,7 @@
 libqt_plugin_la_LIBADD += $(WAYLAND_CLIENT_LIBS)
 endif
 if HAVE_WIN32
+libqt_plugin_la_LIBADD += $(LIBCOM) -lcomctl32 -luuid
-libqt_plugin_la_LIBADD += $(LIBCOM) -lcomctl32 -luuid -ld3d11
 endif
 if UPDATE_CHECK
 libqt_plugin_la_CPPFLAGS += -DUPDATE_CHECK
--- modules/gui/qt/Makefile.am.orig	2020-08-27 21:40:07.120000000 +0100
+++ modules/gui/qt/Makefile.am	2020-08-27 21:40:20.200000000 +0100
@@ -228,14 +228,6 @@
 	gui/qt/maininterface/compositor_win7.cpp \
 	gui/qt/maininterface/compositor_win7.hpp
 
-if HAVE_DCOMP
-libqt_plugin_la_SOURCES += \
-	gui/qt/maininterface/compositor_dcomp.cpp \
-	gui/qt/maininterface/compositor_dcomp.hpp \
-	gui/qt/maininterface/compositor_dcomp_error.hpp \
-	gui/qt/maininterface/compositor_dcomp_uisurface.cpp \
-	gui/qt/maininterface/compositor_dcomp_uisurface.hpp
-endif
 endif
 
 # Meta-object compilation
--- modules/gui/qt/Makefile.am.orig	2020-08-27 21:54:47.030000000 +0100
+++ modules/gui/qt/Makefile.am	2020-08-27 21:54:58.530000000 +0100
@@ -348,12 +348,6 @@
 nodist_libqt_plugin_la_SOURCES += gui/qt/maininterface/main_interface_win32.moc.cpp \
 	gui/qt/maininterface/compositor_win7.moc.cpp
 
-if HAVE_DCOMP
-nodist_libqt_plugin_la_SOURCES += \
-	gui/qt/maininterface/compositor_dcomp.moc.cpp \
-	gui/qt/maininterface/compositor_dcomp_uisurface.moc.cpp
-endif
-
 endif
 
 nodist_libqt_plugin_la_SOURCES += \
--- modules/gui/qt/maininterface/compositor.cpp.orig	2020-08-27 22:04:05.800000000 +0100
+++ modules/gui/qt/maininterface/compositor.cpp	2020-08-27 22:04:28.540000000 +0100
@@ -19,34 +19,10 @@
 #include "compositor.hpp"
 #include "compositor_dummy.hpp"
 
-#ifdef _WIN32
-#ifdef HAVE_DCOMP_H
-#  include "compositor_dcomp.hpp"
-#endif
-#  include "compositor_win7.hpp"
-#endif
-
 namespace vlc {
 
 Compositor* Compositor::createCompositor(intf_thread_t *p_intf)
 {
-    bool ret;
-    VLC_UNUSED(ret);
-#ifdef _WIN32
-#ifdef HAVE_DCOMP_H
-    CompositorDirectComposition* dcomp_compositor = new CompositorDirectComposition(p_intf);
-    ret = dcomp_compositor->init();
-    if (ret)
-        return dcomp_compositor;
-    delete dcomp_compositor;
-    msg_Dbg(p_intf, "failed to create DirectComposition backend, use fallback");
-#endif
-    CompositorWin7* win7_compositor = new CompositorWin7(p_intf);
-    if (win7_compositor->init())
-        return win7_compositor;
-    delete win7_compositor;
-    msg_Dbg(p_intf, "failed to create Win7 compositor backend, use fallback");
-#endif
     return new CompositorDummy(p_intf);
 }
 

--- modules/gui/qt/Makefile.am.orig	2020-11-01 20:01:03.040000000 +0000
+++ modules/gui/qt/Makefile.am	2020-11-01 20:02:28.410000000 +0000
@@ -232,9 +232,7 @@
 if HAVE_WIN32
 libqt_plugin_la_SOURCES += \
 	gui/qt/maininterface/main_interface_win32.cpp \
-	gui/qt/maininterface/main_interface_win32.hpp \
-	gui/qt/maininterface/compositor_win7.cpp \
-	gui/qt/maininterface/compositor_win7.hpp
+	gui/qt/maininterface/main_interface_win32.hpp 
 
 endif
 
--- modules/gui/qt/Makefile.am.orig	2020-11-01 20:17:01.280000000 +0000
+++ modules/gui/qt/Makefile.am	2020-11-01 20:17:17.230000000 +0000
@@ -347,8 +347,7 @@
 	gui/qt/widgets/native/searchlineedit.moc.cpp
 
 if HAVE_WIN32
-nodist_libqt_plugin_la_SOURCES += gui/qt/maininterface/main_interface_win32.moc.cpp \
-	gui/qt/maininterface/compositor_win7.moc.cpp
+nodist_libqt_plugin_la_SOURCES += gui/qt/maininterface/main_interface_win32.moc.cpp 
 
 endif
 
