--- source/CMakeLists.txt.orig	2026-05-08 21:35:39.401718716 +0100
+++ source/CMakeLists.txt	2026-05-08 21:35:53.913589564 +0100
@@ -6,15 +6,15 @@
         FORCE)
 endif()
 message(STATUS "cmake version ${CMAKE_VERSION}")
-if(POLICY CMP0025)
-    cmake_policy(SET CMP0025 OLD) # report Apple's Clang as just Clang
-endif()
+#if(POLICY CMP0025)
+#    cmake_policy(SET CMP0025 OLD) # report Apple's Clang as just Clang
+#endif()
 if(POLICY CMP0042)
     cmake_policy(SET CMP0042 NEW) # MACOSX_RPATH
 endif()
-if(POLICY CMP0054)
-    cmake_policy(SET CMP0054 OLD) # Only interpret if() arguments as variables or keywords when unquoted
-endif()
+#if(POLICY CMP0054)
+#    cmake_policy(SET CMP0054 OLD) # Only interpret if() arguments as variables or keywords when unquoted
+#endif()
 
 project (x265)
 cmake_minimum_required (VERSION 2.8.8) # OBJECT libraries require 2.8.8
@@ -299,16 +299,16 @@
         list(APPEND LINKER_OPTIONS "-static-libgcc")
     endif(STATIC_LINK_CRT)
     check_cxx_compiler_flag(-Wno-strict-overflow CC_HAS_NO_STRICT_OVERFLOW)
-    check_cxx_compiler_flag(-Wno-narrowing CC_HAS_NO_NARROWING) 
-    check_cxx_compiler_flag(-Wno-array-bounds CC_HAS_NO_ARRAY_BOUNDS) 
+    check_cxx_compiler_flag(-Wno-narrowing CC_HAS_NO_NARROWING)
+    check_cxx_compiler_flag(-Wno-array-bounds CC_HAS_NO_ARRAY_BOUNDS)
     if (CC_HAS_NO_ARRAY_BOUNDS)
         add_definitions(-Wno-array-bounds) # these are unhelpful
     endif()
-    check_cxx_compiler_flag(-ffast-math CC_HAS_FAST_MATH) 
+    check_cxx_compiler_flag(-ffast-math CC_HAS_FAST_MATH)
     if (CC_HAS_FAST_MATH)
         add_definitions(-ffast-math)
     endif()
-    check_cxx_compiler_flag(-mstackrealign CC_HAS_STACK_REALIGN) 
+    check_cxx_compiler_flag(-mstackrealign CC_HAS_STACK_REALIGN)
     if (CC_HAS_STACK_REALIGN)
         add_definitions(-mstackrealign)
     endif()
