--- src/openssl/x509vfy.c.orig	2017-10-31 11:07:44.507701700 +0000
+++ src/openssl/x509vfy.c	2017-10-31 11:08:14.335692200 +0000
@@ -12,7 +12,7 @@
 #include "globals.h"
 
 #ifndef XMLSEC_NO_X509
-
+#define WIN32_LEAN_AND_MEAN
 #include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
--- src/openssl/x509.c.orig	2017-10-31 11:15:04.346942100 +0000
+++ src/openssl/x509.c	2017-10-31 11:15:19.791457700 +0000
@@ -12,7 +12,7 @@
 #include "globals.h"
 
 #ifndef XMLSEC_NO_X509
-
+#define WIN32_LEAN_AND_MEAN
 #include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
