--- include/yaml-cpp/node/detail/iterator.h.orig	2020-01-19 17:01:21.508012718 +0000
+++ include/yaml-cpp/node/detail/iterator.h	2020-01-19 17:01:42.363804653 +0000
@@ -11,6 +11,7 @@
 #include "yaml-cpp/node/ptr.h"
 #include "yaml-cpp/node/detail/node_iterator.h"
 #include <boost/iterator/iterator_adaptor.hpp>
+#include <boost/next_prior.hpp>
 #include <boost/utility.hpp>
 
 namespace YAML {
--- CMakeLists.txt.orig	2020-01-19 17:05:12.553717876 +0000
+++ CMakeLists.txt	2020-01-19 17:06:05.425195384 +0000
@@ -328,7 +328,7 @@
 ### Extras
 ###
 if(YAML_CPP_BUILD_TOOLS)
-	add_subdirectory(test)
+	#	add_subdirectory(test)
 	add_subdirectory(util)
 endif()
 
--- CMakeLists.txt.orig	2020-01-22 21:34:19.367909975 +0000
+++ CMakeLists.txt	2020-01-22 21:34:37.199735807 +0000
@@ -317,11 +317,9 @@
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/yaml-cpp-config-version.cmake.in
 	"${PROJECT_BINARY_DIR}/yaml-cpp-config-version.cmake" @ONLY)
 
-if(UNIX)
-	set(PC_FILE ${CMAKE_BINARY_DIR}/yaml-cpp.pc)
-	configure_file("yaml-cpp.pc.cmake" ${PC_FILE} @ONLY)
-	install(FILES ${PC_FILE} DESTINATION ${LIB_INSTALL_DIR}/pkgconfig)
-endif()
+set(PC_FILE ${CMAKE_BINARY_DIR}/yaml-cpp.pc)
+configure_file("yaml-cpp.pc.cmake" ${PC_FILE} @ONLY)
+install(FILES ${PC_FILE} DESTINATION ${LIB_INSTALL_DIR}/pkgconfig)
 
 
 ###
