--- Project/CMake/CMakeLists.txt.orig	2025-05-02 10:01:48.029183695 +0100
+++ Project/CMake/CMakeLists.txt	2025-05-02 10:02:25.636830157 +0100
@@ -7,12 +7,6 @@
 set(ZenLib_PATCH_VERSION 41)
 set(ZenLib_VERSION ${ZenLib_MAJOR_VERSION}.${ZenLib_MINOR_VERSION}.${ZenLib_PATCH_VERSION})
 
-if(WIN32)
-  set(BUILD_SHARED_LIBS OFF)
-elseif(NOT DEFINED BUILD_SHARED_LIBS)
-  option(BUILD_SHARED_LIBS "Build shared libs" ON)
-endif()
-
 option(ENABLE_UNICODE "Enable unicode support" ON)
 option(LARGE_FILES "Enable large files support" ON)
 
